{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: PATH=/soft/EB_repo/bio/structure/programs/goolf/1.7.20/Chimera/1.11.2/bin:/soft/system/easybuild/software/Miniconda3/4.7.10:/soft/system/easybuild/software/Miniconda3/4.7.10/bin:/soft/EB_repo/bio/structure/programs/foss/2016b/VMD/1.9.4a9/bin:/soft/system/lua/bin:/soft/system/luarocks/bin:/home/daranda/miniconda3/bin:/home/daranda/miniconda3/condabin:/usr/lib64/qt-3.3/bin:/soft/system/lua/bin:/soft/system/luarocks/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:/opt/puppetlabs/bin:/home/daranda/bin:/home/daranda/software/getcontacts-master:/usr/local/dowser/bin:/usr/local/dowser/bin/linux:/home/daranda/.local/bin:/home/daranda/bin:/home/daranda/bin:/home/daranda/software/getcontacts-master:/gpcr/users/daranda/doctorat/GPCR_simulations/fake_slurm/:/gpcr/users/daranda/doctorat/GPCR_simulations/fake_slurm/\n"
     ]
    }
   ],
   "source": [
    "####################\n",
    "## Initial stuff\n",
    "####################\n",
    "\n",
    "# Load functions, modules and global variables required in our pipeline\n",
    "from simulate_structures_functions import *\n",
    "from importlib import reload  \n",
    "from htmd.builder.charmm import _recoverProtonations\n",
    "\n",
    "# Are we running curated structures??\n",
    "curated = True\n",
    "\n",
    "# PDB codes of the GPCRs to be simulated. \n",
    "# If no codes are provided, all avalible structures in GPCRdb will be used (except the ones already simulated)\n",
    "noncomplex = {\"3C9L\",\"3C9M\",\"6ZDR\",\"6ZDV\",\"6TPK\", \"6S0Q\", \"6V9S\", \"6WJC\", \"6W25\", \"6OBA\", \"6LW5\", \"6KP6\", \"6LUQ\", \"6LI1\", \"6LI0\", \"6LI2\", \"6KPC\", \"6LRY\", \"6KNM\", \"6TOT\", \"6TOS\", \"6TO7\", \"6TP6\", \"6TQ7\", \"6TOD\", \"6TP3\", \"6TQ4\", \"6TQ9\", \"6TP4\", \"6TQ6\", \"6TPN\", \"6TPG\", \"6TPJ\", \"6OL9\", \"6RZ6\", \"6RZ9\", \"6RZ7\", \"6RZ8\", \"6PT2\", \"6PT3\", \"6KUX\", \"6KUY\", \"6KUW\", \"6IQL\", \"6PS7\", \"6PS0\", \"6PS3\", \"6PS5\", \"6PS1\", \"6PS4\", \"6PRZ\", \"6PS2\", \"6PS6\", \"6KK1\", \"6KK7\", \"6KJV\", \"6PS8\", \"6JZH\", \"6RZ4\", \"6RZ5\", \"6KQI\", \"6K1Q\", \"6GT3\", \"6MH8\", \"6ME2\", \"6ME3\", \"6ME4\", \"6ME5\", \"6ME6\", \"6ME7\", \"6ME8\", \"6ME9\", \"6J21\", \"6J20\", \"6A94\", \"6A93\", \"5ZTY\", \"6HLP\", \"6HLO\", \"6HLL\", \"6GPX\", \"6GPS\", \"6IIV\", \"6IIU\", \"6E59\", \"6M9T\", \"6AK3\", \"5ZHP\", \"5ZKC\", \"5ZK3\", \"5YC8\", \"5ZK8\", \"5ZKB\", \"6IGK\", \"6IGL\", \"6FJ3\", \"6AKX\", \"6AKY\", \"6D27\", \"6D26\", \"6DRX\", \"6DS0\", \"6DRY\", \"6DRZ\", \"6BD4\", \"5ZKQ\", \"5ZKP\", \"6C1R\", \"6C1Q\", \"6D32\", \"6D35\", \"5KW2\", \"5ZBH\", \"5ZBQ\", \"6FK7\", \"6FKA\",  \"6FK6\", \"6FK9\", \"6FKC\", \"6FK8\", \"6FKB\", \"6CM4\", \"6FFI\", \"6FFH\", \"5WF5\", \"5WF6\", \"5V54\", \"5OLH\", \"5OLZ\", \"5OLO\", \"5OM1\", \"5OLG\", \"5OLV\", \"5OM4\", \"5YQZ\", \"6AQF\", \"5O9H\", \"5X33\", \"5VRA\", \"5WS3\", \"5WQC\", \"5WIV\", \"5WIU\", \"5NM4\", \"5NM2\", \"5NLX\", \"5X7D\", \"5X93\", \"5XPR\", \"5XSZ\", \"5N2S\", \"5MZP\", \"5MZJ\", \"5N2R\", \"5XRA\", \"5XR8\", \"5UIW\", \"5NX2\", \"5TZY\", \"5TZR\", \"5JTB\", \"5VBL\", \"5UVI\", \"5VEW\", \"5V56\", \"5V57\", \"5VEX\", \"5NDD\", \"5NDZ\", \"5UNH\", \"5UNF\", \"5UNG\", \"5TE3\", \"5TE5\", \"5UEN\", \"5UIG\", \"5TVN\", \"5T04\", \"5T1A\", \"5U09\", \"5TGZ\", \"5K2C\", \"5K2B\", \"5K2A\", \"5K2D\", \"5GLI\", \"5GLH\", \"5D6L\", \"5DYS\", \"5L7D\", \"5L7I\", \"5IU7\", \"5IUB\", \"5IU8\", \"5IU4\", \"5IUA\", \"4Z9G\", \"5EE7\", \"5DSG\", \"5CXV\", \"4ZJ8\", \"4ZJC\", \"5F8U\", \"5DHG\", \"5DHH\", \"4ZUD\", \"5A8E\", \"5CGC\", \"5CGD\", \"4XEE\", \"4XES\", \"4Z35\", \"4Z34\", \"4Z36\", \"4YAY\", \"4UHR\", \"4XNW\", \"4XNV\", \"4XT3\", \"4RWS\", \"4RWD\", \"4S0V\", \"4U15\", \"4U16\", \"4QIM\", \"4QIN\", \"4PHU\", \"4OO9\", \"4PXZ\", \"4PY0\", \"4BVN\", \"4NTJ\", \"4OR2\", \"4O9R\", \"4BUO\", \"4N4W\", \"4N6H\", \"4NC3\", \"4MBS\", \"4L6R\", \"4K5Y\", \"4JKV\", \"3ZPR\", \"3ZPQ\", \"4IAQ\", \"4IAR\", \"4IB4\", \"4GPO\", \"3VW7\", \"4GBR\", \"4GRV\", \"4EIY\", \"4AMJ\", \"4AMI\", \"4EJ4\", \"4EA3\", \"3UZA\", \"3UZC\", \"4DJH\", \"4DKL\", \"3V2Y\", \"3UON\", \"3REY\", \"3RFM\", \"3RZE\", \"2YCZ\", \"2YCW\", \"2YDV\", \"2YDO\", \"3QAK\", \"2Y04\", \"2Y00\", \"2Y03\", \"2Y02\", \"3PDS\", \"3PBL\", \"3ODU\", \"3OE0\", \"3NY9\", \"3NY8\", \"3NYA\", \"3D4S\", \"2Z73\", \"2RH1\", \"1U19\", \"1GZM\"}\n",
    "noncomplex_nonGPCRmd = {\"3C9L\",\"3C9M\",\"6ZDR\",\"6ZDV\",'6RZ8', '6FFI', '6KJV', '6TQ6', '6DRY', '5WIV', '5V57', '6LI2', '6MH8', '5MZJ', '5KW2', '5X33', '5ZHP', '6PS6', '6FJ3', '6D35', '4NTJ', '6A94', '6AK3', '5UVI', '6FFH', '5WIU', '6KQI', '5XR8', '6RZ9', '5VEX', '6LW5', '6FKA', '6PS7', '6M9T', '6KNM', '6ME6', '6GPX', '5XSZ', '5K2A', '5YQZ', '6PS0', '5K2C', '6ME7', '2Z73', '6D32', '6RZ4', '5YC8', '5T1A', '6AKY', '4N4W', '5WS3', '6LRY', '6J21', '5TE5', '4O9R', '6FK6', '1GZM', '6DRX', '6TQ7', '5NDZ', '6IGL', '5VRA', '5ZBQ', '6TPN', '5ZTY', '6DS0', '5NLX', '5X93', '6HLP', '5OLG', '5OM4', '5O9H', '5XPR', '6A93', '5GLI', '6OBA', '6D27', '6HLO', '6RZ5', '5TE3', '6IIV', '6KP6', '6TQ4', '6TPJ', '5V56', '5UNG', '6FK9', '6PS8', '2YCZ', '6JZH', '5XRA', '5ZKB', '4JKV', '5MZP', '5K2D', '5OLV', '5X7D', '5T04', '5OLO', '5DYS', '6FK7', '6J20', '6K1Q', '5EE7', '6D26', '6PS1', '6DRZ', '4GBR', '6FKC', '6TP6', '6LUQ', '5ZKQ', '5UNH', '6PRZ', '5UNF', '5NX2', '6LI0', '5NM2', '5D6L', '6AQF', '6KK1', '6C1Q', '4Z9G', '6LI1', '5WF5', '6FKB',  '5OLH', '4XT3', '5ZK3', '5VEW', '4XES', '6HLL', '6TOT', '5N2R', '5WF6', '5K2B', '6TPG', '6PS4', '5F8U', '6ME4', '6GT3', '6RZ6', '6ME8', '6ME2', '5VBL', '5N2S', '5ZBH', '6PS3', '6IIU', '4NC3', '5JTB', '5ZKP', '6TQ9', '6GPS', '6TP3', '6RZ7', '6ME3', '5ZK8', '6C1R', '5V54', '5UIG', '5TVN', '5OM1', '6AKX', '5ZKC', '5NM4', '5TZY', '6TOD', '4GPO', '6TO7', '6CM4', '6TOS', '6PS5', '6ME9', '5NDD', '6KPC', '6KK7', '4BUO', '6OL9', '6PS2', '6FK8', '6ME5', '6TP4', '4QIM', '6IGK', '5WQC', '5UIW', '5TZR', '5OLZ', '4EJ4', '6E59'}\n",
    "noncomplex_nonGPCRmd_agonist = {'5T04', '6ME9', '4BUO', '6ME6', '5VBL', '5TZR', '6D35', '6LW5', '5YQZ', '6ME3', '5WF6', '6D32', '5TE5', '4XES', '6AK3', '5DYS', '5KW2', '5WF5', '6FK6', '6ME7', '4NC3', '6ME2', '6FKA', '6DRY', '6FK7', '5TVN', '6KPC', '6KQI', '6LRY', '6IGL', '5NX2', '6M9T', '6FKC', '6FKB', '6FJ3', '6IGK', '6KNM', '6FK9', '6ME8', '6ME5', '6ME4', '5XRA', '5N2S', '5TZY', '6FK8', '6LI0', '5XR8'}\n",
    "pdb_set = noncomplex_nonGPCRmd\n",
    "\n",
    "# Our main path\n",
    "basepath = basepath = os.getcwd()+'/'\n",
    "# Other Paths\n",
    "strucpath = basepath + 'data_structures/'\n",
    "resultspath = basepath + 'simulation_output/'\n",
    "membranepdb = basepath + 'membrane/popc36_box_renumbered.pdb'\n",
    "topparpath = basepath + 'toppar/TOP_PARAMS_ACE3/'#toppar= topology+parameters\n",
    "ligandsdict_path = basepath + 'ligands.json'\n",
    "modres_path = basepath + 'modified_residues.json'\n",
    "slurmpath = basepath+'fake_slurm/'\n",
    "# Path to slurm queing system binaries\n",
    "# In our case, Ismael designed a bunch of small bash scripts (fake_slurm) which do ssh to Hydra and execute slurm there\n",
    "path= os.environ['PATH']\n",
    "\n",
    "# Modify path to include fake slurm\n",
    "%env PATH=$path:$slurmpath\n",
    "\n",
    "# Load topology, parameter and stream files with our current basepath\n",
    "topos = [os.path.join(topparpath,file) for file in toposfilenames] \n",
    "params = [os.path.join(topparpath,file) for file in paramsfilenames]\n",
    "streams = [os.path.join(topparpath,file) for file in streamsfilenames]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4Z9G/4Z9G_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6C1Q/6C1Q_refined.pdb was not found.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 15:20:09,821 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=\"all\" option when reading the file.\n",
      "2022-04-05 15:20:09,867 - moleculekit.molecule - INFO - Removed 1 atoms. 2996 atoms remaining in the molecule.\n",
      "2022-04-05 15:20:16,343 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=\"all\" option when reading the file.\n",
      "2022-04-05 15:20:16,346 - moleculekit.molecule - INFO - Removed 1 atoms. 3005 atoms remaining in the molecule.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FFI/6FFI_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6LUQ/6LUQ_refined.pdb was not found.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 15:20:22,119 - moleculekit.readers - INFO - Attempting PDB query for 6TPJ\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZKP/5ZKP_refined.pdb was not found.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 15:20:24,976 - moleculekit.molecule - INFO - Removed 8607 atoms. 37 atoms remaining in the molecule.\n",
      "2022-04-05 15:20:25,021 - moleculekit.molecule - INFO - Removed 0 atoms. 37 atoms remaining in the molecule.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/6TPJ_YCM.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/6TPJ_YCM.mol2 does not exist. Skipping....\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/6TPJ_legacy_toppar.str' -> '/gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/pre_6TPJ_legacy_toppar.str'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-97d753411bef>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     13\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0;31m# Get topology-parameter files for modified residues\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m \u001b[0mget_modres_toppar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmodresdict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbasepath\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0musername\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpassword\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/gpcr/users/daranda/doctorat/GPCR_simulations/simulate_structures_functions.py\u001b[0m in \u001b[0;36mget_modres_toppar\u001b[0;34m(modresdict, basepath, username, password, pdbfiles)\u001b[0m\n\u001b[1;32m    792\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mmodres\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mmodresdict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpdbcode\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    793\u001b[0m             \u001b[0mmodrespath\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbasepath\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'toppar/mod_residues/'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mmodres\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'/'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 794\u001b[0;31m             \u001b[0mmodres_covlig_toppar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpdbcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmodres\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmodrespath\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpdbfiles\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    795\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    796\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/gpcr/users/daranda/doctorat/GPCR_simulations/simulate_structures_functions.py\u001b[0m in \u001b[0;36mmodres_covlig_toppar\u001b[0;34m(pdbcode, modres, modrespath, covlig, pdbfiles)\u001b[0m\n\u001b[1;32m    930\u001b[0m         \u001b[0;31m#Rename output files from paramchem\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    931\u001b[0m         \u001b[0mpre_file\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodrespath\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'pre_'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mtopparname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 932\u001b[0;31m         \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmodrespath\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mtopparname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpre_file\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    933\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    934\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/6TPJ_legacy_toppar.str' -> '/gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/pre_6TPJ_legacy_toppar.str'"
     ]
    }
   ],
   "source": [
    "################################################\n",
    "# Part 1: Download data and prepare dictionaries\n",
    "################################################\n",
    "\n",
    "# Download and store refined structures from GPCRdb\n",
    "# download_GPCRdb_structures(pdb_set, strucpath)\n",
    "\n",
    "#Create or moidfy the ligands dictionary\n",
    "(ligandsdict, ligandsset, modresdict) = ligand_dictionary(pdb_set, ligandsdict_path, modres_path, basepath, blacklist)\n",
    "\n",
    "# Get topology-parameter files for ligands\n",
    "get_lig_toppar(ligandsdict, basepath, username, password)\n",
    "\n",
    "# Get topology-parameter files for modified residues\n",
    "get_modres_toppar(modresdict, basepath, username, password)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:38:20,763 - moleculekit.molecule - INFO - Removed 0 atoms. 4888 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:20,798 - moleculekit.molecule - INFO - Removed 0 atoms. 4888 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:23,069 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=\"all\" option when reading the file.\n",
      "2021-12-30 09:38:23,070 - moleculekit.molecule - INFO - Removed 15 atoms. 3684 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:23,323 - moleculekit.molecule - INFO - Removed 0 atoms. 4888 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:23,354 - moleculekit.molecule - INFO - Removed 2 atoms. 4886 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:23,450 - moleculekit.molecule - INFO - Removed 0 atoms. 4886 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:26,818 - moleculekit.molecule - INFO - Removed 0 atoms. 4886 atoms remaining in the molecule.\n",
      "/home/david/miniconda3/lib/python3.6/site-packages/Bio/SubsMat/__init__.py:131: BiopythonDeprecationWarning: Bio.SubsMat has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.substitution_matrices as a replacement, and contact the Biopython developers if you still need the Bio.SubsMat module.\n",
      "  BiopythonDeprecationWarning,\n",
      "2021-12-30 09:38:29,716 - moleculekit.tools.sequencestructuralalignment - INFO - No segment was specified by the user for `mol`. Alignment will be done on all protein segments.\n",
      "2021-12-30 09:38:29,756 - moleculekit.tools.sequencestructuralalignment - INFO - No segment was specified by the user for `ref`. Alignment will be done on all protein segments.\n",
      "2021-12-30 09:38:29,933 - moleculekit.tools.sequencestructuralalignment - WARNING - 12 alignments found. Limiting to 1 as specified in the `maxalignments` argument.\n",
      "2021-12-30 09:38:29,935 - moleculekit.tools.sequencestructuralalignment - INFO - Alignment #0 was done on 69 residues: mol segid P resid 154-222\n",
      "/home/david/miniconda3/lib/python3.6/site-packages/moleculekit/align.py:16: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, A), array(float32, 2d, A))\n",
      "  covariance = np.dot(P.T, Q)\n",
      "/home/david/miniconda3/lib/python3.6/site-packages/moleculekit/align.py:54: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, C), array(float32, 2d, A))\n",
      "  all1 = np.dot(all1, rot.T)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adding membrane...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:38:33,738 - htmd.builder.builder - INFO - Replicating Membrane 3x3\n",
      "Replicating Membrane: 100%|██████████| 9/9 [00:02<00:00,  4.08it/s]\n",
      "2021-12-30 09:39:57,781 - moleculekit.molecule - INFO - Removed 2066 atoms. 42862 atoms remaining in the molecule.\n",
      "2021-12-30 09:39:58,915 - moleculekit.molecule - INFO - Removed 139 residues from appended Molecule due to collisions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Solvating...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:39:59,620 - htmd.builder.solvate - INFO - Using water pdb file at: /home/david/miniconda3/lib/python3.6/site-packages/htmd/share/solvate/wat.pdb\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wataerbox Max and min:  [51.999847 52.89102  28.060892] [-47.308155 -46.37798  -27.734108]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:40:00,394 - htmd.builder.solvate - INFO - Replicating 8 water segments, 2 by 2 by 2\n",
      "Solvating: 100%|██████████| 8/8 [00:08<00:00,  1.05s/it]\n",
      "2021-12-30 09:40:10,433 - htmd.builder.solvate - INFO - 14165 water molecules were added to the system.\n",
      "2021-12-30 09:40:16,562 - moleculekit.molecule - INFO - Removed 300 atoms. 84548 atoms remaining in the molecule.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pre-build...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:40:22,384 - htmd.builder.charmm - INFO - Writing out segments.\n",
      "2021-12-30 09:40:28,612 - htmd.builder.builder - INFO - One disulfide bond was added\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Disulfide Bond between: UniqueResidueID<resname: 'CYX', chain: 'P', resid: 174, insertion: '', segid: 'P'>\n",
      "                   and: UniqueResidueID<resname: 'CYX', chain: 'P', resid: 179, insertion: '', segid: 'P'>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:40:30,554 - htmd.builder.charmm - INFO - Starting the build.\n",
      "2021-12-30 09:40:37,826 - htmd.builder.charmm - WARNING - Failed to set coordinates for 2 atoms.\n",
      "2021-12-30 09:40:37,828 - htmd.builder.charmm - WARNING - Failed to guess coordinates for 1 atoms due to bad angles.\n",
      "2021-12-30 09:40:37,828 - htmd.builder.charmm - WARNING - Poorly guessed coordinates for 225 atoms.\n",
      "2021-12-30 09:40:37,828 - htmd.builder.charmm - WARNING - 2 undefined warnings were produced during building.\n",
      "2021-12-30 09:40:37,828 - htmd.builder.charmm - WARNING - Please check /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/pre-build/5ZTY/log.txt for further information.\n",
      "2021-12-30 09:40:37,829 - htmd.builder.charmm - INFO - Finished building.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking aromatic insertions...\n",
      "WARNING: removed 0 atoms within 1.5 of a protein aromatic ring\n",
      "Renumbering...\n",
      "Ionizing...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:42:11,896 - htmd.builder.charmm - INFO - Writing out segments.\n",
      "2021-12-30 09:42:18,160 - htmd.builder.builder - INFO - One disulfide bond was added\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Disulfide Bond between: UniqueResidueID<resname: 'CYS', chain: 'P', resid: 174, insertion: '', segid: 'P'>\n",
      "                   and: UniqueResidueID<resname: 'CYS', chain: 'P', resid: 179, insertion: '', segid: 'P'>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:42:19,361 - htmd.builder.charmm - INFO - Starting the build.\n",
      "2021-12-30 09:42:26,693 - htmd.builder.charmm - WARNING - Failed to set coordinates for 1 atoms.\n",
      "2021-12-30 09:42:26,696 - htmd.builder.charmm - WARNING - Poorly guessed coordinates for 1 atoms.\n",
      "2021-12-30 09:42:26,697 - htmd.builder.charmm - WARNING - 2 undefined warnings were produced during building.\n",
      "2021-12-30 09:42:26,699 - htmd.builder.charmm - WARNING - Please check /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/ionize/5ZTY/log.txt for further information.\n",
      "2021-12-30 09:42:26,702 - htmd.builder.charmm - INFO - Finished building.\n",
      "2021-12-30 09:42:32,556 - htmd.builder.ionize - INFO - Adding 17 anions + 0 cations for neutralizing and 102 ions for the given salt concentration.\n",
      "2021-12-30 09:43:07,077 - htmd.builder.charmm - INFO - Writing out segments.\n",
      "2021-12-30 09:43:14,896 - htmd.builder.charmm - INFO - Starting the build.\n",
      "2021-12-30 09:43:21,976 - htmd.builder.charmm - WARNING - Failed to set coordinates for 1 atoms.\n",
      "2021-12-30 09:43:21,979 - htmd.builder.charmm - WARNING - Poorly guessed coordinates for 1 atoms.\n",
      "2021-12-30 09:43:21,982 - htmd.builder.charmm - WARNING - 2 undefined warnings were produced during building.\n",
      "2021-12-30 09:43:21,984 - htmd.builder.charmm - WARNING - Please check /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/ionize/5ZTY/log.txt for further information.\n",
      "2021-12-30 09:43:21,986 - htmd.builder.charmm - INFO - Finished building.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:44:13,843 - htmd.builder.charmm - INFO - Writing out segments.\n",
      "2021-12-30 09:44:20,582 - htmd.builder.builder - INFO - One disulfide bond was added\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Disulfide Bond between: UniqueResidueID<resname: 'CYS', chain: 'P', resid: 174, insertion: '', segid: 'P'>\n",
      "                   and: UniqueResidueID<resname: 'CYS', chain: 'P', resid: 179, insertion: '', segid: 'P'>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:44:21,674 - htmd.builder.charmm - INFO - Starting the build.\n",
      "2021-12-30 09:44:31,029 - htmd.builder.charmm - WARNING - Failed to set coordinates for 1 atoms.\n",
      "2021-12-30 09:44:31,030 - htmd.builder.charmm - WARNING - Poorly guessed coordinates for 1 atoms.\n",
      "2021-12-30 09:44:31,031 - htmd.builder.charmm - WARNING - 2 undefined warnings were produced during building.\n",
      "2021-12-30 09:44:31,032 - htmd.builder.charmm - WARNING - Please check /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/5ZTY/log.txt for further information.\n",
      "2021-12-30 09:44:31,033 - htmd.builder.charmm - INFO - Finished building.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "End of 5ZTY after 394.39238953590393 seconds\n",
      "\n",
      "Build model for 5ZTY_apo already exists. Skipping...\n"
     ]
    }
   ],
   "source": [
    "###########################\n",
    "## Part 2: Build the models \n",
    "###########################\n",
    "# Iterate by GPCRdb structures to simulate\n",
    "pdbs_number = len(pdb_set)\n",
    "i = 0\n",
    "for pdbcode in {\"5ZTY\"}:\n",
    "# for pdbcode in bad_set:\n",
    "    try:\n",
    "        for apo in [False, True]:\n",
    "            \n",
    "            #Starting simulation\n",
    "            start_time = time.time()        \n",
    "            i += 1\n",
    "            sysname = pdbcode+'_apo' if apo else pdbcode\n",
    "            mystrucpath = strucpath+pdbcode+'/'\n",
    "\n",
    "            # Skip if there is already a model build for this\n",
    "            if os.path.exists(resultspath+'build/'+sysname+'/structure.pdb'):\n",
    "                print('Build model for '+sysname+' already exists. Skipping...')\n",
    "                continue\n",
    "\n",
    "            # Check if simulation is aminergic\n",
    "            aminergic = gpcrdb_dict[pdbcode]['family'].startswith('001_001')\n",
    "            adenosine = gpcrdb_dict[pdbcode]['family'].startswith('001_006_001')\n",
    "\n",
    "            ## Load curated structures\n",
    "            if curated:\n",
    "                apo_name = \"apo\" if apo else \"complex\"\n",
    "                curated_filename = str(\"%sreceptor2curate_output/%s/%s_%s_curated.pdb\"%(basepath, pdbcode, pdbcode, apo_name))\n",
    "                gpcrdb_mol = Molecule(curated_filename)\n",
    "                # Remove ligand hidrogens in case curators put any\n",
    "                gpcrdb_mol.remove(\"not (protein or water or lipid) and element H\")\n",
    "            ## Load non-curated structure\n",
    "            else:\n",
    "                (sod2x50, watered_filename) = internal_waters(mystrucpath, pdbcode, gpcrdb_dict, apo)\n",
    "                gpcrdb_mol = Molecule(watered_filename)\n",
    "            \n",
    "            # Add peptide ligand to structure if necessary\n",
    "            if any([ True for lig in gpcrdb_dict[pdbcode]['ligands'] if lig['type'] in {'protein', 'peptide'} ]) and not (apo):\n",
    "                add_peplig(watered_filename, pdbcode)\n",
    "            \n",
    "            # If the pipeline is running in 'apoform mode', remove any non-protein, non-ion, non-water thing on the system      \n",
    "            # Delete also sod2x50 (we don't want them here)\n",
    "            if apo:\n",
    "                gpcrdb_mol.remove('not (protein or water or ion) or element Na')\n",
    "                \n",
    "            # Remove unnecessary ligand molecules: mostly crystalization detergents, quelants, buffers,\n",
    "            # or post-traductional glicosilations\n",
    "            gpcrdb_mol.remove('resname '+' '.join(blacklist))\n",
    "            \n",
    "            # Remove 2x50Sodium from non-A-class GPCRs\n",
    "            if not gpcrdb_dict[pdbcode]['family'].startswith('001'):\n",
    "                gpcrdb_mol.remove('element NA')\n",
    "\n",
    "            # Get aligned OPM structure\n",
    "            thickness,opm_mol = get_opm(pdbcode)\n",
    "\n",
    "            # Ismael's function to add labels (segid) for 'ligand' and 'protein' parts of the system\n",
    "            gpcrdb_mol_fixed,prot_segids = fix_and_prepare_input(gpcrdb_mol,pdbcode,modresdict,new_pdb_chain)\n",
    "            \n",
    "            # If there's any, parameterize and rename covalent-bound ligands\n",
    "            if not apo:\n",
    "                (gpcrdb_mol_fixed, covligs) = covalent_ligands(gpcrdb_mol_fixed, pdbcode, ligandsdict)\n",
    "            else:\n",
    "                covligs = []\n",
    "            \n",
    "            # write file to remember to which chain in the original structure belongs each segment\n",
    "            segchain_json(gpcrdb_mol_fixed, pdbcode, basepath, prot_segids)\n",
    "            \n",
    "            # Align structrues using sequences, and take first one\n",
    "            alignment_results = sequenceStructureAlignment(gpcrdb_mol_fixed, opm_mol, maxalignments = 1)\n",
    "            mol_aligned = alignment_results[0] \n",
    "            \n",
    "            #Center to receptor XY\n",
    "            center = np.mean(mol_aligned.get('coords',sel='chain P'),axis=0)\n",
    "            mol_aligned.moveBy([-center[0],-center[1],0])\n",
    "            \n",
    "            # Prepare protein: asign titration states, flipping side chains of HIS, ASN and GLN; rotate some sidechains, optimize waters, etc.\n",
    "            # Most of this is done with a HTMD function called proteinPrepare()\n",
    "            # Skip step if we are working with curators structures\n",
    "            prepared_mol = mol_aligned if curated else prepare_system(mol_aligned, pdbcode, thickness, sod2x50, aminergic, adenosine)\n",
    "            \n",
    "            #Add membrane\n",
    "            print('Adding membrane...')\n",
    "            membranemol = Molecule(membranepdb)\n",
    "            mol_membraned, membrane_resnames, membrane_segids, xreps, yreps = add_membrane(prepared_mol, membranemol,prot_segids,membrane_distance)\n",
    "            \n",
    "            # Needed later for equilibration\n",
    "            with open(mystrucpath+\"const_sel.txt\",'w') as out: \n",
    "                const_sel = 'segid '+' '.join(prot_segids)+' and name C CA N O or not (segid ' + \\\n",
    "                  ' '.join(prot_segids)+' or lipid or water or ions ) and noh or segid ION WAT and noh'\n",
    "                out.write(const_sel)\n",
    "\n",
    "            #Solvate\n",
    "            print('Solvating...')\n",
    "            mol_solvated = solvate_pdbmol(mol_membraned,membrane_segids,water_thickness,water_margin,buffer=buffer,coldist=coldist,prefix='WT')\n",
    "\n",
    "            # Check if system has lone-pair hallogen atoms. If it does, use legacy CGenFF parameters\n",
    "            has_halo = bool(len(gpcrdb_mol_fixed.get('name', 'element Br Cl I and not ion')))\n",
    "            if has_halo:\n",
    "                cgenff_par = [topparpath+'David_top_params/parameters/legacy_par_all36_cgenff.prm']\n",
    "                cgenff_top = [topparpath+'David_top_params/topologies/legacy_top_all36_cgenff.rtf']\n",
    "            else: \n",
    "                cgenff_par = [topparpath+'General_top_params/parameters/par_all36_cgenff.prm']\n",
    "                cgenff_top = [topparpath+'General_top_params/topologies/top_all36_cgenff.rtf']\n",
    "            \n",
    "            #Obtain extra parameters for ligands and modified residues \n",
    "            ligstreams=extra_parameters(pdbcode, ligandsdict, modresdict, blacklist, covligs, basepath, has_halo)\n",
    "            \n",
    "            # Assignign terminology for cap atoms of protein chain, depending if it is the receptor protein or not\n",
    "            caps = get_caps(prot_segids, mol_solvated)\n",
    "            #{'P0': ['first ACE', 'last CT3'], 'P1': ['first ACE', 'last CT3']}\n",
    "\n",
    "            #Pre-build model\n",
    "            print('Pre-build...')\n",
    "            prebuildmol = charmm.build(mol_solvated, \n",
    "                                       topo=topos+cgenff_top, \n",
    "                                       param=params+cgenff_par,\n",
    "                                       stream=streams+ligstreams,\n",
    "                                       caps=caps,\n",
    "                                       outdir=resultspath+'/pre-build/'+sysname,\n",
    "                                       ionize=False)\n",
    "\n",
    "            # Save prebuild model topologies in files, and  store prebuild model in molecule object\n",
    "            prebuild_psffile = prebuildmol.topoloc\n",
    "            prebuild_pdbfile = os.path.splitext(prebuildmol.topoloc)[0]+'.pdb'\n",
    "            prebuildmol = Molecule(prebuild_pdbfile)\n",
    "            _recoverProtonations(prebuildmol)\n",
    "\n",
    "            # Checking of aromatic insertions (takes quite a lot fo time)\n",
    "            print('Checking aromatic insertions...')\n",
    "            mol_removed,removed_indexes = remove_aromatic_insertions(prebuildmol,prot_segids, outpdb=resultspath+'/pre-build/'+sysname+'/aromatic_check.pdb')\n",
    "\n",
    "            # Checking of water/lipid ratio\n",
    "            lipid_num = len(set(prebuildmol.get('resid',sel='segid '+membrane_lipid_segid)))\n",
    "            solv_num = len(mol_removed.get('index',sel='resname TIP3 and name OH2'))\n",
    "            if float(solv_num) / lipid_num < 35:\n",
    "                raise ValueError('Water/lipid ratio lower than 35.')\n",
    "\n",
    "            #Renumber residues\n",
    "            print('Renumbering...')\n",
    "            mol_renumbered = renumber_resid_vmd(mol_removed,'segid '+' '.join(membrane_segids),by=2)\n",
    "\n",
    "            # Ionizing system\n",
    "            print('Ionizing...')\n",
    "            molbuilt = charmm.build(mol_removed,\n",
    "                                    topo=topos+cgenff_top, \n",
    "                                    param=params+cgenff_par,\n",
    "                                    stream=streams+ligstreams,                        \n",
    "                                    outdir=resultspath+'/ionize/'+sysname,\n",
    "                                    saltconc=0.15,\n",
    "                                    caps=caps)\n",
    "            build_psffile = molbuilt.topoloc\n",
    "            build_pdbfile = os.path.splitext(molbuilt.topoloc)[0]+'.pdb'\n",
    "            molbuilt = Molecule(build_pdbfile)\n",
    "            _recoverProtonations(molbuilt)\n",
    "\n",
    "            # Remove clashing rebuild waters\n",
    "            molbuilt.remove(\"same residue as (water and within 1 of protein)\")\n",
    "            \n",
    "            #Building system\n",
    "            print('Building...')\n",
    "            molbuilt = renumber_resid_vmd(molbuilt,'segid \"WT.*\" or segid I',by=2)\n",
    "            molbuilt = charmm.build(molbuilt, \n",
    "                                    topo=topos+cgenff_top, \n",
    "                                    param=params+cgenff_par,\n",
    "                                    stream=streams+ligstreams,                        \n",
    "                                    outdir=resultspath+'/build/'+sysname,\n",
    "                                    caps=caps,ionize=False)\n",
    "\n",
    "            print('End of %s after %s seconds\\n' % (sysname, time.time() - start_time))\n",
    "\n",
    "    except Exception as e:\n",
    "        print(\"model \"+pdbcode+\" could not be build because \",e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "model 6LI1 could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/6LI1. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 6LI1 already has been equilibrated\n",
      " structure 6ME6 already has been equilibrated\n",
      " structure 6ME6 already has been equilibrated\n",
      " structure 5OM1 already has been equilibrated\n",
      " structure 5OM1 already has been equilibrated\n",
      " structure 6LUQ already has been equilibrated\n",
      " structure 6LUQ already has been equilibrated\n",
      " structure 3C9M already has been equilibrated\n",
      " structure 3C9M already has been equilibrated\n",
      " structure 6D35 already has been equilibrated\n",
      " structure 6D35 already has been equilibrated\n",
      " structure 6HLO already has been equilibrated\n",
      " structure 6HLO already has been equilibrated\n",
      " structure 6ME2 already has been equilibrated\n",
      " structure 6ME2 already has been equilibrated\n",
      " structure 5V54 already has been equilibrated\n",
      " structure 5V54 already has been equilibrated\n",
      " structure 3C9L already has been equilibrated\n",
      " structure 3C9L already has been equilibrated\n",
      " structure 6TPJ already has been equilibrated\n",
      " structure 6TPJ already has been equilibrated\n",
      " structure 6TO7 already has been equilibrated\n",
      " structure 6TO7 already has been equilibrated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-18 15:57:09,453 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 15:57:09,454 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 15:58:21,365 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/6TP3/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " structure 6TP3 already has been equilibrated\n",
      " structure 6D26 already has been equilibrated\n",
      " structure 6D26 already has been equilibrated\n",
      " structure 5TZR already has been equilibrated\n",
      " structure 5TZR already has been equilibrated\n",
      " structure 6LW5 already has been equilibrated\n",
      " structure 6LW5 already has been equilibrated\n",
      " structure 6TP6 already has been equilibrated\n",
      " structure 6TP6 already has been equilibrated\n",
      " structure 5DYS already has been equilibrated\n",
      " structure 5DYS already has been equilibrated\n",
      " structure 4BUO already has been equilibrated\n",
      " structure 4BUO already has been equilibrated\n",
      " structure 5OLG already has been equilibrated\n",
      " structure 5OLG already has been equilibrated\n",
      " structure 4XT3 already has been equilibrated\n",
      " structure 4XT3 already has been equilibrated\n",
      " structure 6MH8 already has been equilibrated\n",
      " structure 6MH8 already has been equilibrated\n",
      " structure 4NC3 already has been equilibrated\n",
      " structure 4NC3 already has been equilibrated\n",
      "model 4GPO could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/4GPO. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 4GPO already has been equilibrated\n",
      " structure 5D6L already has been equilibrated\n",
      " structure 5D6L already has been equilibrated\n",
      " structure 5NM2 already has been equilibrated\n",
      " structure 5NM2 already has been equilibrated\n",
      " structure 5ZK3 already has been equilibrated\n",
      " structure 5ZK3 already has been equilibrated\n",
      " structure 4NTJ already has been equilibrated\n",
      " structure 4NTJ already has been equilibrated\n",
      " structure 6FJ3 already has been equilibrated\n",
      " structure 6FJ3 already has been equilibrated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-18 15:58:52,430 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 15:58:52,431 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 16:00:08,888 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/5XR8/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " structure 5XR8 already has been equilibrated\n",
      " structure 6RZ4 already has been equilibrated\n",
      " structure 6RZ4 already has been equilibrated\n",
      " structure 6ME8 already has been equilibrated\n",
      " structure 6ME8 already has been equilibrated\n",
      " structure 6PS6 already has been equilibrated\n",
      " structure 6PS6 already has been equilibrated\n",
      " structure 5UNG already has been equilibrated\n",
      " structure 5UNG already has been equilibrated\n",
      " structure 5V56 already has been equilibrated\n",
      " structure 5V56 already has been equilibrated\n",
      " structure 5F8U already has been equilibrated\n",
      " structure 5F8U already has been equilibrated\n",
      " structure 5UIG already has been equilibrated\n",
      " structure 5UIG already has been equilibrated\n",
      " structure 5OLZ already has been equilibrated\n",
      " structure 5OLZ already has been equilibrated\n",
      " structure 6E59 already has been equilibrated\n",
      " structure 6E59 already has been equilibrated\n",
      " structure 5MZJ already has been equilibrated\n",
      " structure 5MZJ already has been equilibrated\n",
      " structure 5NDZ already has been equilibrated\n",
      " structure 5NDZ already has been equilibrated\n",
      " structure 6FFI already has been equilibrated\n",
      " structure 6FFI already has been equilibrated\n",
      " structure 5YC8 already has been equilibrated\n",
      " structure 5YC8 already has been equilibrated\n",
      " structure 6J20 already has been equilibrated\n",
      " structure 6J20 already has been equilibrated\n",
      " structure 6FKB already has been equilibrated\n",
      " structure 6FKB already has been equilibrated\n",
      " structure 5EE7 already has been equilibrated\n",
      " structure 5EE7 already has been equilibrated\n",
      " structure 6PS1 already has been equilibrated\n",
      " structure 6PS1 already has been equilibrated\n",
      " structure 2Z73 already has been equilibrated\n",
      " structure 2Z73 already has been equilibrated\n",
      " structure 4N4W already has been equilibrated\n",
      " structure 4N4W already has been equilibrated\n",
      " structure 5ZKQ already has been equilibrated\n",
      " structure 5ZKQ already has been equilibrated\n",
      " structure 6TQ4 already has been equilibrated\n",
      " structure 6TQ4 already has been equilibrated\n",
      " structure 5VRA already has been equilibrated\n",
      " structure 5VRA already has been equilibrated\n",
      " structure 6FKA already has been equilibrated\n",
      " structure 6FKA already has been equilibrated\n",
      " structure 5NX2 already has been equilibrated\n",
      " structure 5NX2 already has been equilibrated\n",
      " structure 1GZM already has been equilibrated\n",
      " structure 1GZM already has been equilibrated\n",
      " structure 5WS3 already has been equilibrated\n",
      " structure 5WS3 already has been equilibrated\n",
      " structure 5ZBH already has been equilibrated\n",
      " structure 5ZBH already has been equilibrated\n",
      " structure 6KPC already has been equilibrated\n",
      " structure 6KPC already has been equilibrated\n",
      " structure 5YQZ already has been equilibrated\n",
      " structure 5YQZ already has been equilibrated\n",
      " structure 6C1R already has been equilibrated\n",
      " structure 6C1R already has been equilibrated\n",
      " structure 6FKC already has been equilibrated\n",
      " structure 6FKC already has been equilibrated\n",
      " structure 2YCZ already has been equilibrated\n",
      " structure 2YCZ already has been equilibrated\n",
      " structure 6OBA already has been equilibrated\n",
      " structure 6OBA already has been equilibrated\n",
      " structure 5OM4 already has been equilibrated\n",
      " structure 5OM4 already has been equilibrated\n",
      " structure 5UVI already has been equilibrated\n",
      " structure 5UVI already has been equilibrated\n",
      " structure 6PS0 already has been equilibrated\n",
      " structure 6PS0 already has been equilibrated\n",
      " structure 6GPX already has been equilibrated\n",
      " structure 6GPX already has been equilibrated\n",
      " structure 5XRA already has been equilibrated\n",
      " structure 5XRA already has been equilibrated\n",
      " structure 5TVN already has been equilibrated\n",
      " structure 5TVN already has been equilibrated\n",
      " structure 6RZ7 already has been equilibrated\n",
      " structure 6RZ7 already has been equilibrated\n",
      " structure 4XES already has been equilibrated\n",
      " structure 4XES already has been equilibrated\n",
      " structure 6LRY already has been equilibrated\n",
      " structure 6LRY already has been equilibrated\n",
      " structure 6DRY already has been equilibrated\n",
      " structure 6DRY already has been equilibrated\n",
      " structure 5TE5 already has been equilibrated\n",
      " structure 5TE5 already has been equilibrated\n",
      " structure 5WIU already has been equilibrated\n",
      " structure 5WIU already has been equilibrated\n",
      " structure 6IIV already has been equilibrated\n",
      " structure 6IIV already has been equilibrated\n",
      " structure 5V57 already has been equilibrated\n",
      " structure 5V57 already has been equilibrated\n",
      " structure 5X93 already has been equilibrated\n",
      " structure 5X93 already has been equilibrated\n",
      " structure 5OLV already has been equilibrated\n",
      " structure 5OLV already has been equilibrated\n",
      " structure 6PS2 already has been equilibrated\n",
      " structure 6PS2 already has been equilibrated\n",
      " structure 5VEW already has been equilibrated\n",
      " structure 5VEW already has been equilibrated\n",
      " structure 4QIM already has been equilibrated\n",
      " structure 4QIM already has been equilibrated\n",
      " structure 6AKY already has been equilibrated\n",
      " structure 6AKY already has been equilibrated\n",
      "model 5TE3 could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/5TE3. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 5TE3 already has been equilibrated\n",
      " structure 5ZK8 already has been equilibrated\n",
      " structure 5ZK8 already has been equilibrated\n",
      " structure 6D27 already has been equilibrated\n",
      " structure 6D27 already has been equilibrated\n",
      " structure 4JKV already has been equilibrated\n",
      " structure 4JKV already has been equilibrated\n",
      " structure 6TPG already has been equilibrated\n",
      " structure 6TPG already has been equilibrated\n",
      " structure 6RZ5 already has been equilibrated\n",
      " structure 6RZ5 already has been equilibrated\n",
      " structure 6TQ6 already has been equilibrated\n",
      " structure 6TQ6 already has been equilibrated\n",
      " structure 6TOT already has been equilibrated\n",
      " structure 6TOT already has been equilibrated\n",
      " structure 6PS8 already has been equilibrated\n",
      " structure 6PS8 already has been equilibrated\n",
      " structure 6K1Q already has been equilibrated\n",
      " structure 6K1Q already has been equilibrated\n",
      " structure 5WQC already has been equilibrated\n",
      " structure 5WQC already has been equilibrated\n",
      " structure 6ME9 already has been equilibrated\n",
      " structure 6ME9 already has been equilibrated\n",
      " structure 6CM4 already has been equilibrated\n",
      " structure 6CM4 already has been equilibrated\n",
      " structure 5T1A already has been equilibrated\n",
      " structure 5T1A already has been equilibrated\n",
      " structure 5OLO already has been equilibrated\n",
      " structure 5OLO already has been equilibrated\n",
      " structure 6AQF already has been equilibrated\n",
      " structure 6AQF already has been equilibrated\n",
      " structure 6OL9 already has been equilibrated\n",
      " structure 6OL9 already has been equilibrated\n",
      " structure 4EJ4 already has been equilibrated\n",
      " structure 4EJ4 already has been equilibrated\n",
      " structure 6AKX already has been equilibrated\n",
      " structure 6AKX already has been equilibrated\n",
      " structure 6FK8 already has been equilibrated\n",
      " structure 6FK8 already has been equilibrated\n",
      " structure 5N2R already has been equilibrated\n",
      " structure 5N2R already has been equilibrated\n",
      " structure 6C1Q already has been equilibrated\n",
      " structure 6C1Q already has been equilibrated\n",
      " structure 5O9H already has been equilibrated\n",
      " structure 5O9H already has been equilibrated\n",
      " structure 5K2A already has been equilibrated\n",
      " structure 5K2A already has been equilibrated\n",
      " structure 4GBR already has been equilibrated\n",
      " structure 4GBR already has been equilibrated\n",
      " structure 6KQI already has been equilibrated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-18 16:01:28,280 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 16:01:28,280 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 16:02:22,617 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/6KQI_apo/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " structure 6TOS already has been equilibrated\n",
      " structure 6TOS already has been equilibrated\n",
      " structure 6J21 already has been equilibrated\n",
      " structure 6J21 already has been equilibrated\n",
      " structure 5OLH already has been equilibrated\n",
      " structure 5OLH already has been equilibrated\n",
      " structure 6DRZ already has been equilibrated\n",
      " structure 6DRZ already has been equilibrated\n",
      " structure 5XPR already has been equilibrated\n",
      " structure 5XPR already has been equilibrated\n",
      " structure 6LI0 already has been equilibrated\n",
      " structure 6LI0 already has been equilibrated\n",
      " structure 6PS7 already has been equilibrated\n",
      " structure 6PS7 already has been equilibrated\n",
      " structure 5VEX already has been equilibrated\n",
      " structure 5VEX already has been equilibrated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-18 16:02:56,171 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 16:02:56,171 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 16:04:25,677 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/5ZTY/\n",
      "2022-01-18 16:04:55,847 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 16:04:55,848 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 16:06:02,014 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/5ZTY_apo/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " structure 6GT3 already has been equilibrated\n",
      " structure 6GT3 already has been equilibrated\n",
      " structure 6RZ6 already has been equilibrated\n",
      " structure 6RZ6 already has been equilibrated\n",
      " structure 6PS3 already has been equilibrated\n",
      " structure 6PS3 already has been equilibrated\n",
      " structure 6A93 already has been equilibrated\n",
      " structure 6A93 already has been equilibrated\n",
      " structure 6ZDV already has been equilibrated\n",
      " structure 6ZDV already has been equilibrated\n",
      " structure 6ME4 already has been equilibrated\n",
      " structure 6ME4 already has been equilibrated\n",
      " structure 5WIV already has been equilibrated\n",
      " structure 5WIV already has been equilibrated\n",
      " structure 6TQ7 already has been equilibrated\n",
      " structure 6TQ7 already has been equilibrated\n",
      " structure 6AK3 already has been equilibrated\n",
      " structure 6AK3 already has been equilibrated\n",
      " structure 5UNF already has been equilibrated\n",
      " structure 5UNF already has been equilibrated\n",
      " structure 5N2S already has been equilibrated\n",
      " structure 5N2S already has been equilibrated\n",
      " structure 5JTB already has been equilibrated\n",
      " structure 5JTB already has been equilibrated\n",
      " structure 6RZ8 already has been equilibrated\n",
      " structure 6RZ8 already has been equilibrated\n",
      " structure 6TPN already has been equilibrated\n",
      " structure 6TPN already has been equilibrated\n",
      " structure 6M9T already has been equilibrated\n",
      " structure 6M9T already has been equilibrated\n",
      " structure 5VBL already has been equilibrated\n",
      " structure 5VBL already has been equilibrated\n",
      " structure 6FK6 already has been equilibrated\n",
      " structure 6FK6 already has been equilibrated\n",
      " structure 5ZKP already has been equilibrated\n",
      " structure 5ZKP already has been equilibrated\n",
      " structure 6PS4 already has been equilibrated\n",
      " structure 6PS4 already has been equilibrated\n",
      " structure 5TZY already has been equilibrated\n",
      " structure 5TZY already has been equilibrated\n",
      " structure 6FK9 already has been equilibrated\n",
      " structure 6FK9 already has been equilibrated\n",
      " structure 5T04 already has been equilibrated\n",
      " structure 5T04 already has been equilibrated\n",
      " structure 6TOD already has been equilibrated\n",
      " structure 6TOD already has been equilibrated\n",
      " structure 6A94 already has been equilibrated\n",
      " structure 6A94 already has been equilibrated\n",
      " structure 5NLX already has been equilibrated\n",
      " structure 5NLX already has been equilibrated\n",
      " structure 6KJV already has been equilibrated\n",
      " structure 6KJV already has been equilibrated\n",
      " structure 6D32 already has been equilibrated\n",
      " structure 6D32 already has been equilibrated\n",
      " structure 5WF6 already has been equilibrated\n",
      " structure 5WF6 already has been equilibrated\n",
      " structure 6ME5 already has been equilibrated\n",
      " structure 6ME5 already has been equilibrated\n",
      " structure 6TQ9 already has been equilibrated\n",
      " structure 6TQ9 already has been equilibrated\n",
      " structure 6IGK already has been equilibrated\n",
      " structure 6IGK already has been equilibrated\n",
      " structure 5X33 already has been equilibrated\n",
      " structure 5X33 already has been equilibrated\n",
      " structure 4Z9G already has been equilibrated\n",
      " structure 4Z9G already has been equilibrated\n",
      " structure 6JZH already has been equilibrated\n",
      " structure 6JZH already has been equilibrated\n",
      " structure 6IGL already has been equilibrated\n",
      " structure 6IGL already has been equilibrated\n",
      " structure 6FK7 already has been equilibrated\n",
      " structure 6FK7 already has been equilibrated\n",
      " structure 6KNM already has been equilibrated\n",
      " structure 6KNM already has been equilibrated\n",
      " structure 5MZP already has been equilibrated\n",
      " structure 5MZP already has been equilibrated\n",
      " structure 5K2B already has been equilibrated\n",
      " structure 5K2B already has been equilibrated\n",
      " structure 6IIU already has been equilibrated\n",
      " structure 6IIU already has been equilibrated\n",
      " structure 6DS0 already has been equilibrated\n",
      " structure 6DS0 already has been equilibrated\n",
      " structure 6PS5 already has been equilibrated\n",
      " structure 6PS5 already has been equilibrated\n",
      " structure 5NDD already has been equilibrated\n",
      " structure 5NDD already has been equilibrated\n",
      " structure 5K2C already has been equilibrated\n",
      " structure 5K2C already has been equilibrated\n",
      " structure 6KK7 already has been equilibrated\n",
      " structure 6KK7 already has been equilibrated\n",
      " structure 5UIW already has been equilibrated\n",
      " structure 5UIW already has been equilibrated\n",
      " structure 5UNH already has been equilibrated\n",
      " structure 5UNH already has been equilibrated\n",
      " structure 6ME7 already has been equilibrated\n",
      " structure 6ME7 already has been equilibrated\n",
      " structure 6KK1 already has been equilibrated\n",
      " structure 6KK1 already has been equilibrated\n",
      " structure 6ZDR already has been equilibrated\n",
      " structure 6ZDR already has been equilibrated\n",
      " structure 6DRX already has been equilibrated\n",
      " structure 6DRX already has been equilibrated\n",
      " structure 6RZ9 already has been equilibrated\n",
      " structure 6RZ9 already has been equilibrated\n",
      "model 6LI2 could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/6LI2. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 6LI2 already has been equilibrated\n",
      " structure 5ZBQ already has been equilibrated\n",
      " structure 5ZBQ already has been equilibrated\n",
      " structure 6HLP already has been equilibrated\n",
      " structure 6HLP already has been equilibrated\n",
      " structure 6HLL already has been equilibrated\n",
      " structure 6HLL already has been equilibrated\n",
      " structure 6TP4 already has been equilibrated\n",
      " structure 6TP4 already has been equilibrated\n",
      "model 5XSZ could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/5XSZ. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 5XSZ already has been equilibrated\n",
      " structure 4O9R already has been equilibrated\n",
      " structure 4O9R already has been equilibrated\n",
      " structure 5NM4 already has been equilibrated\n",
      " structure 5NM4 already has been equilibrated\n",
      "model 5GLI could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/5GLI. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 5GLI already has been equilibrated\n",
      " structure 6PRZ already has been equilibrated\n",
      " structure 6PRZ already has been equilibrated\n",
      " structure 5WF5 already has been equilibrated\n",
      " structure 5WF5 already has been equilibrated\n",
      " structure 5ZKC already has been equilibrated\n",
      " structure 5ZKC already has been equilibrated\n",
      " structure 5ZKB already has been equilibrated\n",
      " structure 5ZKB already has been equilibrated\n",
      " structure 5X7D already has been equilibrated\n",
      " structure 5X7D already has been equilibrated\n",
      " structure 5K2D already has been equilibrated\n",
      " structure 5K2D already has been equilibrated\n",
      " structure 6ME3 already has been equilibrated\n",
      " structure 6ME3 already has been equilibrated\n",
      "model 6KP6 could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/6KP6. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 6KP6 already has been equilibrated\n",
      " structure 5ZHP already has been equilibrated\n",
      " structure 5ZHP already has been equilibrated\n",
      " structure 6FFH already has been equilibrated\n",
      " structure 6FFH already has been equilibrated\n",
      " structure 5KW2 already has been equilibrated\n",
      " structure 5KW2 already has been equilibrated\n",
      " structure 6GPS already has been equilibrated\n",
      " structure 6GPS already has been equilibrated\n"
     ]
    }
   ],
   "source": [
    "#########################\n",
    "## Part 3: Equillibration\n",
    "#########################\n",
    "\n",
    "for pdbcode in pdb_set:\n",
    "# for pdbcode in ['5XR8']:\n",
    "    for apo in [False, True]:\n",
    "        try:\n",
    "            modelname = pdbcode+'_apo' if apo else pdbcode\n",
    "            equildir = resultspath+'equil/'+modelname+'/'\n",
    "            if os.path.exists(equildir+'output.xtc') or os.path.exists(equildir+'simrunning'):\n",
    "                print(\" structure %s already has been equilibrated\" % pdbcode)\n",
    "                continue\n",
    "\n",
    "            if not os.path.exists(equildir):\n",
    "                os.makedirs(equildir)\n",
    "\n",
    "            # Taking vmd selection line\n",
    "            const_sel = []\n",
    "            with open(strucpath+pdbcode+'/const_sel.txt', 'r') as outfile:\n",
    "                const_sel += outfile.readlines()\n",
    "\n",
    "            md = define_equilibration(const_sel)\n",
    "            md.write(resultspath+'build/'+modelname,equildir)\n",
    "\n",
    "            #Substitute run.sh generated by HTMD by a different one, adapted to the specified path of ACEMD\n",
    "            with open(equildir + 'run.sh', 'w') as f:\n",
    "                f.write('#!/bin/bash\\n%s > %slog.txt 2>&1' % (acemd_path, equildir))\n",
    "        \n",
    "            #Prepare slurm job  \n",
    "            sq = SlurmQueue()\n",
    "            sq.envvars = acemd_license\n",
    "            sq.jobname = 'eql_'+pdbcode\n",
    "            sq.datadir = None\n",
    "            sq.partition = 'gpcr_gpu'\n",
    "            sq.ngpu = 1\n",
    "            sq.ncpu = 1\n",
    "            sq.prerun = job_commands(equildir, '/home/daranda/'+modelname+'/')\n",
    "            sq.memory = 2000\n",
    "#             sq.exclude = ['arwen','aragorn','bifur']\n",
    "            sq.nodelist = ['arwen']\n",
    "    \n",
    "            # Submit\n",
    "            sq.submit(equildir)\n",
    "            \n",
    "        except Exception as e:\n",
    "            print(\"model \"+modelname+\" could not be send to equilibrate because of \",e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "replicate 1 of structure 4EJ4 already has been simulated\n",
      "replicate 2 of structure 4EJ4 already has been simulated\n",
      "replicate 3 of structure 4EJ4 already has been simulated\n",
      "submitting replicate 1 of 4EJ4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:11:09,787 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (SER 204 CA P P) (SER 204 C P P) (PRO 205 N P P) (PRO 205 CA P P) \" with indexes [2602, 2609, 2611, 2615]\n",
      "2022-02-10 14:11:10,254 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:12:01,055 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/4EJ4/rep_1/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "submitting replicate 2 of 4EJ4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:12:19,914 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (SER 204 CA P P) (SER 204 C P P) (PRO 205 N P P) (PRO 205 CA P P) \" with indexes [2602, 2609, 2611, 2615]\n",
      "2022-02-10 14:12:20,372 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:13:09,531 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/4EJ4/rep_2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "submitting replicate 3 of 4EJ4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:13:27,607 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (SER 204 CA P P) (SER 204 C P P) (PRO 205 N P P) (PRO 205 CA P P) \" with indexes [2602, 2609, 2611, 2615]\n",
      "2022-02-10 14:13:28,053 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:14:16,931 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/4EJ4/rep_3/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "replicate 1 of structure 6LRY already has been simulated\n",
      "replicate 2 of structure 6LRY already has been simulated\n",
      "replicate 3 of structure 6LRY already has been simulated\n",
      "submitting replicate 1 of 6LRY\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:14:48,405 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (ILE 212 CA P P) (ILE 212 C P P) (GLY 213 N P P) (GLY 213 CA P P) \" with indexes [2020, 2035, 2037, 2039]\n",
      "2022-02-10 14:14:48,901 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:15:40,890 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/6LRY/rep_1/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "submitting replicate 2 of 6LRY\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:16:00,232 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (ILE 212 CA P P) (ILE 212 C P P) (GLY 213 N P P) (GLY 213 CA P P) \" with indexes [2020, 2035, 2037, 2039]\n",
      "2022-02-10 14:16:00,697 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:16:44,572 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/6LRY/rep_2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "submitting replicate 3 of 6LRY\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:17:03,929 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (ILE 212 CA P P) (ILE 212 C P P) (GLY 213 N P P) (GLY 213 CA P P) \" with indexes [2020, 2035, 2037, 2039]\n",
      "2022-02-10 14:17:04,387 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:17:48,813 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/6LRY/rep_3/\n"
     ]
    }
   ],
   "source": [
    "#####################\n",
    "## Part 4: Production\n",
    "#####################\n",
    "\n",
    "# Production protocol\n",
    "md = define_production(timestep, trajperiod)\n",
    "\n",
    "# If some model should be skipped, put its name here\n",
    "modelname_skip = {}\n",
    "\n",
    "# For each PDB \n",
    "for pdbcode in ['4EJ4','6LRY']:\n",
    "# for pdbcode in pdb_set:\n",
    "#     for apo in [False]:\n",
    "    for apo in [True, False]:\n",
    "        # must match with equildir in equilibration launcher code and contain input and output of equilibration.\n",
    "        modelname = pdbcode+'_apo' if apo else pdbcode\n",
    "        equildir = '%s/equil/%s/' % (resultspath, modelname)\n",
    "        # IF equillibration is not avalible\n",
    "        if not os.path.exists(equildir+'output.xtc'):\n",
    "            print(\"replicate %d of structure %s has no avalible equillibration\" %(rep, pdbcode))\n",
    "            continue\n",
    "        for rep in range(1,repnum+1):\n",
    "            \n",
    "            try: \n",
    "                    \n",
    "                # If simulation for this PDB has already been run\n",
    "                proddir='%sproduction/%s/rep_%d/' % (resultspath, modelname, rep)\n",
    "                if os.path.exists(proddir+'/output.xtc') or os.path.exists(proddir+'simrunning'):\n",
    "                    print(\"replicate %d of structure %s already has been simulated\" %(rep, pdbcode))\n",
    "                    continue\n",
    "\n",
    "                print('submitting replicate %d of %s' % (rep, pdbcode))\n",
    "                # directory copy output of equilibration to production input (initial working directory for run_prod.sh).\n",
    "                md.write(equildir,proddir)\n",
    "\n",
    "                sq = SlurmQueue()\n",
    "                sq.envvars = acemd_license\n",
    "                sq.jobname = modelname+'_pr'+str(rep)\n",
    "                sq.datadir = None\n",
    "                sq.partition = 'gpcr_gpu'\n",
    "                sq.prerun = job_commands(proddir, '/home/daranda/%s_pr_%d/'%(modelname,rep))\n",
    "                sq.ngpu = 1\n",
    "                sq.ncpu = 1\n",
    "                sq.exclude = ['aragorn','arwen','bifur']\n",
    "\n",
    "                #Substitute run.sh generated by HTMD by a different one, adapted to the specified path of ACEMD\n",
    "                with open(proddir + 'run.sh', 'w') as f:\n",
    "                    f.write('#!/bin/bash\\n%s >%slog.txt 2>&1' % (acemd_path,proddir))\n",
    "\n",
    "                sq.submit(proddir)\n",
    "                \n",
    "            except Exception as e:\n",
    "                print(\"model \"+modelname+\" could not be send to equilibrate because of \",e)            \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#####################\n",
    "## EXTRA: ligand RMSD\n",
    "#####################\n",
    "### Assess quality of systems by calculating the RMSD of the ligand/s molecule/s\n",
    "\n",
    "# Set paths and files\n",
    "outpath = resultspath+'rmsd_lig.tsv'\n",
    "if os.path(outpath):\n",
    "    print('RMSDs already computed. SKipping...')\n",
    "else:\n",
    "    # Iterate over pdbcodes \n",
    "    results = []\n",
    "    for pdbcode in pdbcodes: \n",
    "\n",
    "        # Iterate over production replicates\n",
    "        for mytrajid in [\"1\",\"2\",\"3\"]:\n",
    "\n",
    "            try:\n",
    "                sysname = \"%s_rep%s\"%(pdbcode,mytrajid)\n",
    "\n",
    "                # Input files of simulation\n",
    "                files_path = resultspath+'production/%s/rep_%s/'%(name, mytrajid)\n",
    "                mypdbpath = files_path+'structure.pdb'\n",
    "                mypsfpath = files_path+'structure.psf'\n",
    "                mytrajpath = files_path+'output_wrapped.xtc'\n",
    "\n",
    "                # Skip if no trajectory\n",
    "                if not os.path.exists(mytrajpath):\n",
    "                    print(\"no trajectory replicate %s for system %s avalible. Skipping...\"%(mytrajid, sysname))\n",
    "                    continue\n",
    "\n",
    "                print('computing ligand RMSD for trajectory %s of system %s' % (mytrajid, sysname))\n",
    "\n",
    "                # Load trajectory and topology into MDA universe, and select protein atoms\n",
    "                u = mda.Universe(mypsfpath, mytrajpath)\n",
    "                ligsel = u.select_atoms(\"segid LIG and not resname CLR\")\n",
    "\n",
    "                # Compute rmsd, extract its values and put them in corresponding lists and dicts\n",
    "                R = rms.RMSD(ligsel)\n",
    "                R.run()\n",
    "                rmsd = np.mean([ a[2] for a in R.rmsd ])\n",
    "                results.append((sysname,rmsd)) \n",
    "\n",
    "            except Exception as e:\n",
    "                print(\"error: system %s failed becasue %s\"%(sysname,e))\n",
    "\n",
    "    # Once everything is done, sort and write RMSD results\n",
    "    out = open(outpath,'w')\n",
    "    out.write(\"Simulated_system\\tRMSD_ligand\\n\")\n",
    "    results_sorted = sorted(results, key=lambda tup: tup[0])\n",
    "    for line in results_sorted:\n",
    "        out.write(\"%s\\t%f\\n\"%(line[0],line[1])) \n",
    "    out.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wrapping replicate 1 of 4EJ4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/moleculekit/align.py:16: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, A), array(float32, 2d, A))\n",
      "  covariance = np.dot(P.T, Q)\n",
      "/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/moleculekit/align.py:54: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, C), array(float32, 2d, A))\n",
      "  all1 = np.dot(all1, rot.T)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wrapping replicate 2 of 4EJ4\n",
      "wrapping replicate 3 of 4EJ4\n",
      "End of 4EJ4 after 193.5178680419922 seconds\n",
      "\n",
      "wrapping replicate 1 of 6LRY\n",
      "wrapping replicate 2 of 6LRY\n",
      "wrapping replicate 3 of 6LRY\n",
      "End of 6LRY after 198.5682544708252 seconds\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "##########################\n",
    "## Part 5: Wrap Structures\n",
    "##########################\n",
    "\n",
    "# Wrap trajectories obtained during production with an htmd command\n",
    "gpcr_sel = \"protein and chain \"+new_pdb_chain\n",
    "# for pdbcode in pdb_set:\n",
    "for pdbcode in ['4EJ4','6LRY']:\n",
    "# for pdbcode in testset:\n",
    "    for apo in [False]:\n",
    "#     for apo in [True, False]:\n",
    "        try:\n",
    "            modelname = pdbcode+'_apo' if apo else pdbcode\n",
    "            mymol_pdb = Molecule('%sproduction/%s/rep_1/structure.pdb' % (resultspath, modelname))\n",
    "            for rep in range(1,repnum+1):\n",
    "            #for rep in [3]:\n",
    "                start_time = time.time()        \n",
    "                print('wrapping replicate %d of %s' % (rep, modelname))\n",
    "                proddir='%sproduction/%s/rep_%d/' % (resultspath, modelname, rep)\n",
    "                rep = str(rep)\n",
    "\n",
    "                # To avoid repeating wrapping in Trajectories already wrapped, check the existance of this file\n",
    "                outname = proddir+'output_wrapped.xtc'\n",
    "                if os.path.exists(outname):\n",
    "                    print('replicate already wrapped. Skipping...')\n",
    "                    continue\n",
    "                \n",
    "                # Wrap system\n",
    "                mymol = Molecule('%sproduction/%s/rep_1/structure.psf' % (resultspath, modelname))\n",
    "                mymol.read(proddir+'output.xtc')\n",
    "                mymol.wrap(gpcr_sel)\n",
    "\n",
    "                # Align frames and write\n",
    "                mymol.align('all', refmol=mymol_pdb)\n",
    "                mymol.write(outname)\n",
    "\n",
    "            print('End of %s after %s seconds\\n' % (modelname, time.time() - start_time))\n",
    "        except Exception as e:\n",
    "            print(\"model \"+modelname+\" could not be wrapped because of \",e)            \n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "dynids={\n",
    "\"6PS0_apo\" : 684,\n",
    "\"6PS0\" : 685,\n",
    "\"6RZ7_apo\" : 805,\n",
    "\"6RZ7\" : 806,\n",
    "\"5VEX_apo\" : 710,\n",
    "\"5VEX\" : 711,\n",
    "\"5N2S_apo\" : 775,\n",
    "\"5N2S\" : 1097,\n",
    "\"5ZK3_apo\" : 989,\n",
    "\"5ZK3\" : 990,\n",
    "\"5MZJ_apo\" : 995,\n",
    "\"5MZJ\" : 1098,\n",
    "\"6LRY_apo\" : 996,\n",
    "\"6LRY\" : 997,\n",
    "\"6TPN_apo\" : 902,\n",
    "\"6TPN\" : 903,\n",
    "\"5UIG_apo\" : 773,\n",
    "\"5UIG\" : 774,\n",
    "\"6FKA_apo\" : 732,\n",
    "\"6FKA\" : 733,\n",
    "\"6KNM_apo\" : 1041,\n",
    "\"6KNM\" : 1042,\n",
    "\"6TP4_apo\" : 759,\n",
    "\"6TP4\" : 760,\n",
    "\"4XT3_apo\" : 878,\n",
    "\"4XT3\" : 879,\n",
    "\"6PS7_apo\" : 1013,\n",
    "\"6PS7\" : 1014,\n",
    "\"5F8U_apo\" : 728,\n",
    "\"5F8U\" : 729,\n",
    "\"5ZKP_apo\" : 1024,\n",
    "\"5ZKP\" : 1025,\n",
    "\"5XSZ_apo\" : 871,\n",
    "\"5T04_apo\" : 722,\n",
    "\"5T04\" : 723,\n",
    "\"6LW5\" : 1081,\n",
    "\"6LW5\" : 1082,\n",
    "\"5NM4_apo\" : 953,\n",
    "\"5NM4\" : 954,\n",
    "\"6DRY_apo\" : 918,\n",
    "\"6DRY\" : 919,\n",
    "\"6KJV_apo\" : 746,\n",
    "\"6KJV\" : 1039,\n",
    "\"6OBA_apo\" : 1003,\n",
    "\"6HLL_apo\" : 850,\n",
    "\"6HLL\" : 851,\n",
    "\"6IIV_apo\" : 1075,\n",
    "\"6IIV\" : 1076,\n",
    "\"6KPC_apo\" : 955,\n",
    "\"6KPC\" : 956,\n",
    "\"5VEW_apo\" : 698,\n",
    "\"5VEW\" : 699,\n",
    "\"6FKC_apo\" : 706,\n",
    "\"6FKC\" : 707,\n",
    "\"5ZKB_apo\" : 1028,\n",
    "\"5ZKB\" : 1029,\n",
    "\"6TOT_apo\" : 874,\n",
    "\"6TOT\" : 875,\n",
    "\"6PS3_apo\" : 926,\n",
    "\"6PS3\" : 927,\n",
    "\"5NM2_apo\" : 783,\n",
    "\"5NM2\" : 784,\n",
    "\"6ME7_apo\" : 993,\n",
    "\"6ME7\" : 994,\n",
    "\"4EJ4_apo\" : 724,\n",
    "\"4EJ4\" : 725,\n",
    "\"5TE3_apo\" : 750,\n",
    "\"5XRA_apo\" : 896,\n",
    "\"5XRA\" : 897,\n",
    "\"6D27_apo\" : 884,\n",
    "\"6D27\" : 885,\n",
    "\"6C1R\" : 1085,\n",
    "\"6C1R\" : 1100,\n",
    "\"5OM4_apo\" : 714,\n",
    "\"5OM4\" : 715,\n",
    "\"6E59_apo\" : 855,\n",
    "\"6E59\" : 856,\n",
    "\"5ZBH_apo\" : 904,\n",
    "\"5ZBH\" : 905,\n",
    "\"5VRA_apo\" : 1011,\n",
    "\"5VRA\" : 1012,\n",
    "\"5GLI_apo\" : 837,\n",
    "\"5WIU_apo\" : 1005,\n",
    "\"5WIU\" : 1006,\n",
    "\"6AKX_apo\" : 779,\n",
    "\"6AKX\" : 1095,\n",
    "\"5OLG_apo\" : 922,\n",
    "\"5OLG\" : 923,\n",
    "\"5JTB_apo\" : 696,\n",
    "\"5JTB\" : 697,\n",
    "\"5K2A_apo\" : 1020,\n",
    "\"5K2A\" : 1021,\n",
    "\"6PS4_apo\" : 829,\n",
    "\"6PS4\" : 830,\n",
    "\"5X7D_apo\" : 769,\n",
    "\"5X7D\" : 770,\n",
    "\"5UIW_apo\" : 898,\n",
    "\"5UIW\" : 899,\n",
    "\"5WF5\" : 687,\n",
    "\"5WF5_apo\" : 1049,\n",
    "\"5VBL\" : 1062,\n",
    "\"5VBL_apo\" : 1063,\n",
    "\"6GT3_apo\" : 702,\n",
    "\"6GT3\" : 703,\n",
    "\"5OLZ_apo\" : 813,\n",
    "\"5OLZ\" : 814,\n",
    "\"6PRZ_apo\" : 979,\n",
    "\"6PRZ\" : 980,\n",
    "\"1GZM_apo\" : 894,\n",
    "\"1GZM\" : 895,\n",
    "\"5X33_apo\" : 957,\n",
    "\"5X33\" : 958,\n",
    "\"6KK7_apo\" : 876,\n",
    "\"6KK7\" : 877,\n",
    "\"6K1Q_apo\" : 787,\n",
    "\"6K1Q\" : 788,\n",
    "\"6TO7_apo\" : 935,\n",
    "\"6TO7\" : 936,\n",
    "\"6ME9_apo\" : 817,\n",
    "\"6ME9\" : 818,\n",
    "\"5O9H_apo\" : 920,\n",
    "\"5O9H\" : 921,\n",
    "\"5UNG_apo\" : 890,\n",
    "\"5UNG\" : 891,\n",
    "\"5TVN_apo\" : 846,\n",
    "\"5TVN\" : 847,\n",
    "\"6ME8_apo\" : 945,\n",
    "\"6ME8\" : 946,\n",
    "\"6KP6_apo\" : 983,\n",
    "\"6ME3_apo\" : 692,\n",
    "\"6ME3\" : 693,\n",
    "\"6AQF_apo\" : 753,\n",
    "\"6AQF\" : 754,\n",
    "\"5DYS_apo\" : 872,\n",
    "\"5DYS\" : 873,\n",
    "\"6TPG_apo\" : 865,\n",
    "\"6TPG\" : 866,\n",
    "\"3C9M_apo\" : 756,\n",
    "\"3C9M\" : 757,\n",
    "\"6TOD_apo\" : 914,\n",
    "\"6TOD\" : 915,\n",
    "\"6PS2_apo\" : 863,\n",
    "\"6PS2\" : 864,\n",
    "\"5T1A_apo\" : 734,\n",
    "\"5T1A\" : 735,\n",
    "\"6D35_apo\" : 888,\n",
    "\"4Z9G_apo\" : 1030,\n",
    "\"4Z9G\" : 1031,\n",
    "\"6AKY_apo\" : 941,\n",
    "\"6AKY\" : 1124,\n",
    "\"6RZ6_apo\" : 825,\n",
    "\"6RZ6\" : 826,\n",
    "\"6ME2_apo\" : 712,\n",
    "\"6ME2\" : 713,\n",
    "\"5KW2_apo\" : 765,\n",
    "\"5KW2\" : 766,\n",
    "\"4XES_apo\" : 767,\n",
    "\"4XES\" : 768,\n",
    "\"6RZ5_apo\" : 861,\n",
    "\"6RZ5\" : 862,\n",
    "\"6FFH_apo\" : 761,\n",
    "\"6FFH\" : 762,\n",
    "\"6DS0_apo\" : 959,\n",
    "\"6DS0\" : 960,\n",
    "\"2YCZ_apo\" : 785,\n",
    "\"2YCZ\" : 786,\n",
    "\"6GPS_apo\" : 708,\n",
    "\"6GPS\" : 709,\n",
    "\"6FK9_apo\" : 777,\n",
    "\"6FK9\" : 778,\n",
    "\"6LI2_apo\" : 986,\n",
    "\"4BUO_apo\" : 880,\n",
    "\"4BUO\" : 881,\n",
    "\"6FKB_apo\" : 720,\n",
    "\"6FKB\" : 721,\n",
    "\"5NDD_apo\" : 977,\n",
    "\"5NDD\" : 978,\n",
    "\"6TOS_apo\" : 821,\n",
    "\"6TOS\" : 822,\n",
    "\"5N2R_apo\" : 906,\n",
    "\"5N2R\" : 1123,\n",
    "\"6RZ4_apo\" : 738,\n",
    "\"6RZ4\" : 739,\n",
    "\"6FJ3_apo\" : 755,\n",
    "\"5TZR_apo\" : 793,\n",
    "\"5TZR\" : 794,\n",
    "\"6FFI_apo\" : 842,\n",
    "\"6FFI\" : 843,\n",
    "\"6D32_apo\" : 892,\n",
    "\"6D32\" : 893,\n",
    "\"5V54_apo\" : 908,\n",
    "\"5V54\" : 909,\n",
    "\"6IIU_apo\" : 1043,\n",
    "\"6IIU\" : 1044,\n",
    "\"6KQI\" : 1002,\n",
    "\"6D26_apo\" : 984,\n",
    "\"6D26\" : 985,\n",
    "\"6PS1_apo\" : 763,\n",
    "\"6PS1\" : 764,\n",
    "\"6FK7_apo\" : 912,\n",
    "\"6FK7\" : 913,\n",
    "\"6GPX_apo\" : 799,\n",
    "\"6GPX\" : 800,\n",
    "\"5OLO_apo\" : 975,\n",
    "\"5OLO\" : 976,\n",
    "\"4O9R_apo\" : 1022,\n",
    "\"4O9R\" : 1023,\n",
    "\"5UVI_apo\" : 835,\n",
    "\"5UVI\" : 836,\n",
    "\"5WQC_apo\" : 844,\n",
    "\"5WQC\" : 845,\n",
    "\"5V56_apo\" : 823,\n",
    "\"5V56\" : 824,\n",
    "\"5TZY_apo\" : 928,\n",
    "\"5TZY\" : 929,\n",
    "\"5D6L_apo\" : 962,\n",
    "\"5D6L\" : 963,\n",
    "\"6FK8_apo\" : 840,\n",
    "\"6FK8\" : 841,\n",
    "\"5UNH_apo\" : 939,\n",
    "\"5UNH\" : 940,\n",
    "\"6A94_apo\" : 801,\n",
    "\"6A94\" : 802,\n",
    "\"4GPO_apo\" : 961,\n",
    "\"5TE5_apo\" : 809,\n",
    "\"5TE5\" : 810,\n",
    "\"5UNF_apo\" : 916,\n",
    "\"5UNF\" : 917,\n",
    "\"6JZH_apo\" : 1000,\n",
    "\"6JZH\" : 1001,\n",
    "\"4JKV_apo\" : 791,\n",
    "\"4JKV\" : 792,\n",
    "\"5ZKQ_apo\" : 694,\n",
    "\"5ZKQ\" : 695,\n",
    "\"6IGK_apo\" : 1015,\n",
    "\"6IGK\" : 1016,\n",
    "\"4GBR_apo\" : 970,\n",
    "\"4GBR\" : 971,\n",
    "\"5NDZ_apo\" : 1018,\n",
    "\"5NDZ\" : 1019,\n",
    "\"4NTJ_apo\" : 716,\n",
    "\"4NTJ\" : 717,\n",
    "\"5ZBQ_apo\" : 700,\n",
    "\"5ZBQ\" : 1040,\n",
    "\"5X93_apo\" : 964,\n",
    "\"5X93\" : 965,\n",
    "\"5OM1_apo\" : 857,\n",
    "\"5OM1\" : 858,\n",
    "\"6C1Q\" : 1084,\n",
    "\"6C1Q\" : 1099,\n",
    "\"6J20_apo\" : 910,\n",
    "\"6J20\" : 911,\n",
    "\"4QIM_apo\" : 932,\n",
    "\"4QIM\" : 933,\n",
    "\"5NLX_apo\" : 991,\n",
    "\"5NLX\" : 992,\n",
    "\"5ZKC_apo\" : 968,\n",
    "\"5ZKC\" : 1126,\n",
    "\"6TP6_apo\" : 869,\n",
    "\"6TP6\" : 870,\n",
    "\"3C9L_apo\" : 900,\n",
    "\"3C9L\" : 901,\n",
    "\"5ZHP_apo\" : 947,\n",
    "\"5ZHP\" : 1125,\n",
    "\"5OLV\" : 1017,\n",
    "\"6HLO_apo\" : 730,\n",
    "\"6HLO\" : 731,\n",
    "\"5WIV_apo\" : 831,\n",
    "\"5WIV\" : 832,\n",
    "\"6ME4_apo\" : 803,\n",
    "\"6ME4\" : 804,\n",
    "\"5EE7_apo\" : 815,\n",
    "\"5EE7\" : 816,\n",
    "\"5YC8_apo\" : 1122,\n",
    "\"5YC8\" : 1127,\n",
    "\"6MH8_apo\" : 981,\n",
    "\"6MH8\" : 982,\n",
    "\"6FK6_apo\" : 943,\n",
    "\"6FK6\" : 944,\n",
    "\"6M9T_apo\" : 748,\n",
    "\"6M9T\" : 749,\n",
    "\"6ZDV_apo\" : 827,\n",
    "\"6ZDV\" : 828,\n",
    "\"6RZ9_apo\" : 690,\n",
    "\"6RZ9\" : 691,\n",
    "\"6TP3_apo\" : 758,\n",
    "\"6TPJ_apo\" : 744,\n",
    "\"6TPJ\" : 745,\n",
    "\"5YQZ_apo\" : 742,\n",
    "\"5YQZ\" : 743,\n",
    "\"6IGL_apo\" : 1059,\n",
    "\"6IGL\" : 1064,\n",
    "\"4NC3_apo\" : 819,\n",
    "\"4NC3\" : 820,\n",
    "\"5K2C_apo\" : 998,\n",
    "\"5K2C\" : 999,\n",
    "\"2Z73_apo\" : 688,\n",
    "\"2Z73\" : 689,\n",
    "\"6LI1_apo\" : 934,\n",
    "\"6PS5_apo\" : 751,\n",
    "\"6PS5\" : 752,\n",
    "\"6DRX_apo\" : 718,\n",
    "\"6DRX\" : 719,\n",
    "\"6A93_apo\" : 848,\n",
    "\"6A93\" : 849,\n",
    "\"6ME5_apo\" : 807,\n",
    "\"6ME5\" : 808,\n",
    "\"6PS8_apo\" : 1007,\n",
    "\"6PS8\" : 1008,\n",
    "\"5K2D_apo\" : 949,\n",
    "\"5K2D\" : 950,\n",
    "\"6TQ6_apo\" : 859,\n",
    "\"6TQ6\" : 860,\n",
    "\"6TQ7_apo\" : 736,\n",
    "\"6TQ7\" : 737,\n",
    "\"5K2B_apo\" : 853,\n",
    "\"5K2B\" : 854,\n",
    "\"6HLP_apo\" : 937,\n",
    "\"6HLP\" : 938,\n",
    "\"6ME6_apo\" : 795,\n",
    "\"6ME6\" : 796,\n",
    "\"6PS6_apo\" : 951,\n",
    "\"6PS6\" : 952,\n",
    "\"6KK1_apo\" : 704,\n",
    "\"6KK1\" : 705,\n",
    "\"4N4W_apo\" : 771,\n",
    "\"4N4W\" : 772,\n",
    "\"5MZP_apo\" : 987,\n",
    "\"5MZP\" : 988,\n",
    "\"5ZK8_apo\" : 797,\n",
    "\"5ZK8\" : 1096,\n",
    "\"6TQ9_apo\" : 833,\n",
    "\"6TQ9\" : 834,\n",
    "\"5OLH_apo\" : 930,\n",
    "\"5OLH\" : 931,\n",
    "\"6LUQ_apo\" : 838,\n",
    "\"6LUQ\" : 839,\n",
    "\"5XPR_apo\" : 886,\n",
    "\"5XPR\" : 887,\n",
    "\"5WF6_apo\" : 1009,\n",
    "\"5WF6\" : 1010,\n",
    "\"5V57_apo\" : 740,\n",
    "\"5V57\" : 741,\n",
    "\"5WS3_apo\" : 726,\n",
    "\"5WS3\" : 727,\n",
    "\"6J21_apo\" : 781,\n",
    "\"6J21\" : 782,\n",
    "\"6CM4_apo\" : 924,\n",
    "\"6CM4\" : 925,\n",
    "\"6OL9_apo\" : 973,\n",
    "\"6OL9\" : 1094,\n",
    "\"6RZ8_apo\" : 789,\n",
    "\"6RZ8\" : 790,\n",
    "\"5XR8_apo\" : 852,\n",
    "\"6TQ4_apo\" : 966,\n",
    "\"6TQ4\" : 967,\n",
    "\"5NX2_apo\" : 1061,\n",
    "\"5NX2\" : 1065,\n",
    "\"6DRZ_apo\" : 811,\n",
    "\"6DRZ\" : 812,\n",
    "\"6AK3_apo\" : 1026,\n",
    "\"6AK3\" : 1027,\n",
    "\"6ZDR_apo\" : 1032,\n",
    "\"6ZDR\" : 1033,\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def resp_to_dict(resp):\n",
    "    # Convert a json reponse into a dictionary\n",
    "    return eval(resp.content.decode('UTF-8').replace('true', 'True').replace('false','False'))\n",
    "\n",
    "def get_headers(s, subm_id):\n",
    "    \"\"\"\n",
    "    Obtain headers, csrftoken and sessionid from current session object\n",
    "    \"\"\"\n",
    "    sessionid = str(s.cookies['sessionid'])\n",
    "    csrftoken = str(s.cookies['csrftoken'])\n",
    "    headers = {\n",
    "        'Referer' : mainurl+'/dynadb/step1/'+subm_id,\n",
    "        'Cookie' : 'csrftoken=%s; sessionid=%s' %(csrftoken,sessionid),\n",
    "        'X-CSRFToken' : csrftoken,\n",
    "    }\n",
    "    return(sessionid, csrftoken, headers)\n",
    "def check_chains(pdbcode, mymol):\n",
    "    \"\"\"\n",
    "    Check how many chains from the original PDB structure remain in mymol structure\n",
    "    And to which Segments of our structure they correspond\n",
    "    \"\"\"\n",
    "    # Load blosum score matrix to align proteins\n",
    "    blosum62 = substitution_matrices.load(\"BLOSUM62\")\n",
    "\n",
    "    # Obtain sequences for original PDB file chains, and classifying them by chainID\n",
    "    # Also check which segment corresponds to what chain\n",
    "    pdbmol = Molecule(pdbcode)\n",
    "    chainseg = {}\n",
    "    chainset = set(pdbmol.get('chain', sel='protein'))\n",
    "    pdbmol_segseqs = pdbmol.sequence()\n",
    "    pdbmol_chainseqs = {}\n",
    "    for chain in chainset:\n",
    "        segid = np.unique(pdbmol.get('segid', sel='chain '+chain))[0]\n",
    "        pdbmol_chainseqs[chain] = pdbmol_segseqs[segid]\n",
    "\n",
    "    # Merging all protein chains in our systems into a single megachain\n",
    "    mymol_megachain = ''\n",
    "    for seg,chain in mymol.sequence().items():\n",
    "        mymol_megachain = mymol_megachain + chain\n",
    "\n",
    "    # Aligning sequences from original PDB to simulated PDB megasequence\n",
    "    # To know which chains from the original PDB are preserved\n",
    "    chain_present = {}\n",
    "    segtochain = {}\n",
    "    # For each chain in the original PDB file\n",
    "    for chain,seq in pdbmol_chainseqs.items():\n",
    "        chain_present[chain] = False\n",
    "        # For each segment in our molecule\n",
    "        for seg,myseq in mymol.sequence().items():\n",
    "            # Align our molecule segments to the pdb chains\n",
    "            aligs = pairwise2.align.localms(seq, myseq, 5,-1, -1, -0.5)\n",
    "            # Check if any of the alignments has more than 4 score by position\n",
    "            is_present = any([ (alig.score/(alig.end-alig.start)) > 3.5 for alig in aligs ]) \n",
    "            if is_present:\n",
    "                chain_present[chain] = is_present\n",
    "                segtochain[seg] = chain\n",
    "\n",
    "    return (chain_present, segtochain)\n",
    "\n",
    "def get_uniprot_alignment(mymol, segpos, uniprotkbac, isoform, entry_segs):\n",
    "    \"\"\"\n",
    "    Find out the exact coordinates of each Uniprot segment in our protein\n",
    "    \"\"\"\n",
    "\n",
    "    # Obtain sequence of uniprotkbac\n",
    "    uniprot_id = \"%s-%s\"%(uniprotkbac,isoform) if isoform else uniprotkbac \n",
    "    response = requests.get(\"https://www.uniprot.org/uniprot/?query=accession:%s&sort=score&columns=sequence&format=tab\"%uniprotkbac)\n",
    "    uniseq = response.text.split('\\n')[1]\n",
    "    \n",
    "    # Match this uniprot to our protein segments, and save matches\n",
    "    for seg,myseq in mymol.sequence().items():\n",
    "        \n",
    "        # Align our molecule segments to the pdb chains\n",
    "        aligs = pairwise2.align.localms(myseq, uniseq, 5,-1, -1, -0.5)\n",
    "        # For every alignment\n",
    "        for alig in aligs:\n",
    "            start = alig.start\n",
    "            end = alig.end\n",
    "            # If the alignment has no significant amount of mismatches, keep it as segment\n",
    "            if (alig.score/(end-start)) > 3.5:\n",
    "                \n",
    "                # Find coordinates of this segment\n",
    "                alig_myseq = alig[0]\n",
    "                \n",
    "                # Find start of uniprot sequence in alginment\n",
    "                if alig_myseq.startswith('-'):\n",
    "                    starting = 0\n",
    "                else:\n",
    "                    starting = start - 1 \n",
    "                # Find ending of uniprot sequence in alginment            \n",
    "                if alig_myseq.endswith('-'):            \n",
    "                    ending = len(myseq) - 1\n",
    "                else:\n",
    "                    ending = end - 1 - alig.seqA.count('-')\n",
    "                \n",
    "                if uniprotkbac not in entry_segs:\n",
    "                    entry_segs[uniprotkbac] = []\n",
    "                    \n",
    "                # Do not add repeated alignments                \n",
    "                beg = segpos[seg][starting]\n",
    "                end = segpos[seg][ending]\n",
    "                repeated_seg = any([ ((seg['beg']==beg) and (seg['end']==end)) for seg in entry_segs[uniprot_id]])\n",
    "                if not len(entry_segs[uniprot_id]) or not repeated_seg:\n",
    "                    entry_segs[uniprot_id].append({\n",
    "                                'chainid' : mymol.get('chain', 'segid '+seg)[0],\n",
    "                                'segid' : seg,\n",
    "                                'beg' : segpos[seg][starting],\n",
    "                                'end' : segpos[seg][ending]})\n",
    "\n",
    "    return(entry_segs) \n",
    "\n",
    "\n",
    "def get_pdb_info(pdbcode, mymol, ligandsdict, blacklist, apo):\n",
    "    \"\"\"\n",
    "    Get informarion of the system specified in the pdbcode from the RCSB-PDB webpage.\n",
    "    Mainly uniprot sequences, chainIDs and uniprot codes for the chains\n",
    "    \"\"\"\n",
    "    \n",
    "    # Make string for selecting lignads in rcsb's api\n",
    "    ligs = '\"'+'\",\"'.join(ligandsdict[pdbcode].keys())+'\"'\n",
    "    gpcr_names = ['ceptor','rhodopsin','smoothened']\n",
    "    \n",
    "    #Check which chains from the original PDB structure are preserved in mymol structure\n",
    "    (chain_present, segtochain) = check_chains(pdbcode, mymol)\n",
    "    \n",
    "    # Get residue names in our system\n",
    "    allresnames = set(mymol.resname)\n",
    "    \n",
    "    # Extract information from pdb webpage using api\n",
    "    ligdict = dict()\n",
    "    protdict = dict()\n",
    "    datadict = dict()\n",
    "    entry_segs = dict()\n",
    "    segpos = dict()\n",
    "\n",
    "    # Obtain dictionary of sorted protein resids in each segment\n",
    "    for seg in set(mymol.get('segid', 'protein')):\n",
    "        segpos[seg] = list(dict.fromkeys(mymol.get('resid', 'segid '+seg)))\n",
    "    \n",
    "    # Get information from PDB api\n",
    "    pdbdict = requests.get('https://data.rcsb.org/graphql?query={\\\n",
    "        entry(entry_id: \"'+pdbcode+'\") {\\\n",
    "            exptl {method}\\\n",
    "            polymer_entities {\\\n",
    "                entity_poly{pdbx_strand_id,rcsb_sample_sequence_length}\\\n",
    "                rcsb_polymer_entity{pdbx_description}\\\n",
    "                rcsb_polymer_entity_align{\\\n",
    "                    aligned_regions {length}\\\n",
    "                    reference_database_accession \\\n",
    "                    reference_database_isoform \\\n",
    "                }\\\n",
    "            }\\\n",
    "        }\\\n",
    "        chem_comps(comp_ids: ['+ligs+']) {\\\n",
    "            rcsb_chem_comp_descriptor {InChIKey}\\\n",
    "            chem_comp{id,name}\\\n",
    "        }\\\n",
    "    }').json()['data']\n",
    "        \n",
    "    # Determine experimental method used, and use the corresponding id in GPCRmd database\n",
    "    method = pdbdict['entry']['exptl'][0]['method'].lower()\n",
    "    if 'x-ray' in method: \n",
    "        method_id = 0\n",
    "        method_segs = 0\n",
    "    elif 'nmr' in method:\n",
    "        method_id = 1\n",
    "        method_segs = 1\n",
    "    elif 'electron microscopy' == method:\n",
    "        method_id = 4\n",
    "        method_segs = 7\n",
    "    else:\n",
    "        method_id = 5 # Other method\n",
    "        method_segs = 6\n",
    "            \n",
    "    # Get ligand information and classify it (in case the system actually has ligands)\n",
    "    if len(pdbdict['chem_comps'])>0:\n",
    "        for lig in pdbdict['chem_comps']:\n",
    "            ligandInchi = lig['rcsb_chem_comp_descriptor']['InChIKey']\n",
    "            ligandResname = lig['chem_comp']['id']\n",
    "            ligandName = lig['chem_comp']['name']\n",
    "    \n",
    "            # Exclude ligands not present in our simulated system\n",
    "            if ligandResname in allresnames:\n",
    "                ligdict[ligandResname] = (ligandName,ligandInchi)\n",
    "\n",
    "            # Beware of covalent-bound retinols\n",
    "            if (ligandResname=='RET') and (\"LYR\" in allresnames):\n",
    "                ligdict[\"LYR\"] = (\"Lysine-Retinol\",\"FSIGCSBKSBDRLV-GGOCYBHBSA-N\")\n",
    "                \n",
    "            # Beware of other covalent-bound ligands\n",
    "            if (\"COV\" in allresnames) and (ligandsdict[pdbcode][ligandResname][1]):\n",
    "                ligdict[\"COV\"] = (ligandName,ligandInchi)\n",
    "                \n",
    "    # Extract protein chains information\n",
    "    recname = \"\"\n",
    "    peplig = (\"\",100000000000000000)\n",
    "    for poly in pdbdict['entry']['polymer_entities']:\n",
    "\n",
    "        # Determine if this polymer (chain) is a GPCR\n",
    "        uniname = poly['rcsb_polymer_entity']['pdbx_description'].lower()\n",
    "        seqlen = poly['entity_poly']['rcsb_sample_sequence_length']\n",
    "        chainIds = poly['entity_poly']['pdbx_strand_id'].split(',')\n",
    "        isgpcr = any([(name in uniname) for name in gpcr_names]) \n",
    "        \n",
    "        # Extract name of the system from pdb info\n",
    "        if isgpcr:\n",
    "            uninames = uniname.split(',')\n",
    "            for unis in uninames:\n",
    "                if any([(name in unis) for name in gpcr_names]):\n",
    "                    recname = unis\n",
    "        else: \n",
    "            # Select shortest non-receptor sequence as potential peptide ligand\n",
    "            if seqlen < peplig[1]:\n",
    "                peplig = (uniname.split(',')[0],seqlen)\n",
    "                    \n",
    "        # Get uniprots present in the system, and their corresponbding segments\n",
    "        for alreg in poly['rcsb_polymer_entity_align']:\n",
    "            uniprotkbac = alreg['reference_database_accession']\n",
    "            isoform = alreg['reference_database_isoform']\n",
    "            uniprot_id = \"%s-%s\"%(uniprotkbac,isoform) if isoform else uniprotkbac \n",
    "            entry_segs = get_uniprot_alignment(mymol, segpos, uniprotkbac, isoform, entry_segs)\n",
    "            \n",
    "            # If this uniprot is present in our system, assign its data to protdict\n",
    "            if uniprot_id in entry_segs:\n",
    "                protdict[uniprotkbac] = {\n",
    "                    'isoform' : isoform if isoform else '1',\n",
    "                    'isgpcr' : isgpcr,\n",
    "                    'segs' : entry_segs[uniprot_id],\n",
    "                    'method' : method_id,\n",
    "                    'method_segs' : method_segs\n",
    "                }\n",
    "    \n",
    "    # Filter ligdict of blacklisted molecules and ions\n",
    "    for lig in ligdict:\n",
    "        if (lig in blacklist) or (len(lig) == 2):\n",
    "            ligdict.pop(lig)\n",
    "\n",
    "    ligname = \"\"\n",
    "    # Find the ligand molecule, if not apo\n",
    "    if not apo:\n",
    "        for lig in ligdict:\n",
    "            molname = ligdict[lig][0]\n",
    "            inchikey = ligdict[lig][1]\n",
    "            if lig != 'CLR': # Take any non-cholesterol, non blacklisted molecule as ligand\n",
    "                # some molecule names are excessivelly large. Take inchikey instead\n",
    "                ligname = molname if len(molname) < 51 else inchikey \n",
    "            # If there is not any other molecule in the system, nor a potential peptide ligand, take cholesterol\n",
    "            elif (len(ligdict.keys()) == 1) and not peplig[0]: \n",
    "                ligname = ligdict[lig][0]\n",
    "        \n",
    "        # If still we have no ligand name, must be peptide\n",
    "        if not ligname:\n",
    "            ligname = peplig[0]\n",
    "\n",
    "    # Make system name\n",
    "    sysname = recname+\" in complex with \"+ligname if not apo else recname+\" (apoform)\"\n",
    "    \n",
    "    return (protdict,ligdict,segtochain,method_id,sysname)\n",
    "\n",
    "def login(s):\n",
    "    # Login into GPCRmd \n",
    "    # Will you go, lassie, go? And we'll all gooo toghetheeeer\n",
    "    headers = {\n",
    "        'Cookie': 'csrftoken=ZYa0n3mNBjwBMlpDh4lnbBJQdi7GdlP4',\n",
    "        'Referer': mainurl+'/accounts/login/',\n",
    "    }\n",
    "    datalogin = {\n",
    "        'username': 'david',\n",
    "        'password': 'Ameboid',\n",
    "        'next' : '/accounts/memberpage/',\n",
    "        'csrfmiddlewaretoken' : 'ZYa0n3mNBjwBMlpDh4lnbBJQdi7GdlP4'\n",
    "    }\n",
    "    print('loging into GPCRmd')\n",
    "    logo = s.post(mainurl+'/accounts/login/', \n",
    "               data=datalogin,\n",
    "               headers=headers)\n",
    "    return s\n",
    "\n",
    "def new_submission(s, mainurl):\n",
    "    \"\"\"\n",
    "    Create a new submission entry in GPCRmd\n",
    "    \"\"\"\n",
    "\n",
    "    response_new = s.get(mainurl + '/dynadb/step0/')\n",
    "    soup = BeautifulSoup(response_new.text, 'html.parser')\n",
    "    step1_link = soup.find('form',attrs={'id' : 'mainform'}).get('action')\n",
    "    subm_id = step1_link.split('/')[-2]\n",
    "    print('new submission %s created'%subm_id)\n",
    "    return subm_id\n",
    "\n",
    "def new_step1(s, subm_id, pdbcode, trajperiod, timestep, prodpath, modelfile, method_id, sysname, apo):\n",
    "    \"\"\"\n",
    "    Fullfill and send the step 1 of the new submissionform\n",
    "    # Looone liiie the fieeeelds of Athenryyyyy...\n",
    "    # Where once, we watched, the small seabirds flyyyyy....\n",
    "    \"\"\"\n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "    \n",
    "    # All the stuff we need to send \n",
    "    data_submit = {\n",
    "        'name' : sysname,\n",
    "        'type' : 0 if apo else 1, # 0 for apo, 1 for complex\n",
    "        'pdbid' : pdbcode,\n",
    "        'description' : 'Classical unbiased (NVT ensemble) complex flexibility assay.',\n",
    "        'source_type' : method_id,\n",
    "        'id_dynamics_methods': '1', #Molecular mechanics,\n",
    "        'software': 'ACEMD3',\n",
    "        'sversion': 'GPUGRID', # TODO: Is that correct??,\n",
    "        'ff': 'CHARMM',\n",
    "        'ffversion': '36m Feb 2016',\n",
    "        'id_assay_types': 1, # Orthosteric (un)/binding,\n",
    "        'id_dynamics_membrane_types': 2, # Homogeneus membrane,\n",
    "        'id_dynamics_solvent_types': 2, # TIP3P solvent,\n",
    "        'timestep': timestep,\n",
    "        'delta': (trajperiod*timestep)/1e6, # time/frames\n",
    "        'add_info': ' Classical unbiased (NVT ensemble) complex flexibility assay.', \n",
    "    }\n",
    "    files_submit = {\n",
    "        'dynamics' : open(prodpath+'structure.pdb', 'rb'),\n",
    "        'model' : open(modelfile, 'rb')\n",
    "    }\n",
    "\n",
    "    # Submit step 1\n",
    "    rep = s.post(mainurl+'/dynadb/step1_submit/'+subm_id+'/',\n",
    "            headers = headers,\n",
    "            files = files_submit,\n",
    "            data = data_submit)\n",
    "    print('step1 finalized ',rep)\n",
    "\n",
    "    # Raise exceptions if something failed\n",
    "    if not rep.ok:\n",
    "        #s.post(mainurl+'/dynadb/delete_submission/'+subm_id)\n",
    "        raise Exception('Step 1 submitted returned %d'%rep.status_code)\n",
    "\n",
    "\n",
    "    \n",
    "def smalmol_getdata(s, molid, inchikey, subm_id, moltype, resname, sdfpath=\"\"):\n",
    "    \"\"\"\n",
    "    Get data for small molecules \n",
    "    \"\"\"\n",
    "    \n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "\n",
    "    # Get info of molecule\n",
    "    rep = s.get(mainurl+'/dynadb/smalmol_info/?inchikey=%s&submission_id=%s'%(inchikey,subm_id),\n",
    "        headers = headers,\n",
    "        )\n",
    "    rep_dict = rep.json()\n",
    "    \n",
    "    # Prepare submission dictionary with the data of this molecule\n",
    "    mol_datasubmit = {\n",
    "        \"smalmol_type\"+molid : moltype,\n",
    "        \"smalmol_inchikey\"+molid : inchikey,\n",
    "        \"smalmol_name\"+molid : rep_dict['name'],\n",
    "        \"smalmol_resname\"+molid : resname,\n",
    "        \"smalmol_iupac\"+molid : rep_dict['iupac'],\n",
    "        \"smalmol_chemblid\"+molid :  rep_dict['chemblid'],\n",
    "        \"smalmol_cid\"+molid :  rep_dict['cid'],\n",
    "        \"smalmol_inchi\"+molid : rep_dict['inchi'],\n",
    "        \"smalmol_sinchi\"+molid :  rep_dict['sinchi'],\n",
    "        \"smalmol_sinchikey\"+molid :  rep_dict['sinchikey'],\n",
    "        \"smalmol_smiles\"+molid :  rep_dict['smiles'],\n",
    "        \"smalmol_netcharge\"+molid :  rep_dict['net_charge'],\n",
    "        \"smalmol_synonims\"+molid :  rep_dict['other_names'],\n",
    "        \"smalmol_description\"+molid :  '',\n",
    "        \"image_path\"+molid :  rep_dict['imagepath']\n",
    "    }\n",
    "    \n",
    "    # Add SDFfile to the submission if required\n",
    "    mol_files = {} if rep_dict['inGPCRmd'] else {\"sdfmol\"+molid : open(sdfpath)} \n",
    "    \n",
    "    return(mol_datasubmit,mol_files)\n",
    "\n",
    "def new_step2(s, subm_id, ligdict, apo):\n",
    "    \"\"\"\n",
    "    Fullfil and send step2 of new submission form\n",
    "    \"\"\"    \n",
    "    print('initiating step 2: small molecule data')\n",
    "\n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "    \n",
    "    # Introduce common small molecules (waters, lipids and ions)\n",
    "    i = 0\n",
    "    num_entries = []\n",
    "    step2_data = {\n",
    "        'csrfmiddlewaretoken' : csrftoken,\n",
    "        'submit' : 'submit'\n",
    "    }\n",
    "    step2_files = {}\n",
    "    common_mols = {\n",
    "        'TIP3' : ('XLYOFNOQVPJJNP-UHFFFAOYSA-N',6),\n",
    "        'POPC' : ('WTJKGGKOPKCXLL-VYOBOKEXSA-N',7),\n",
    "        'SOD' : ('FKNQFGJONOIPTF-UHFFFAOYSA-N',8),\n",
    "        'CLA' : ('VEXZGXHMUGYJMC-UHFFFAOYSA-M',8)\n",
    "    }\n",
    "    for resname,(inchikey,moltype) in common_mols.items():\n",
    "        (mol_datasubmit, mol_files) = smalmol_getdata(s, str(i), inchikey, subm_id, moltype, resname)\n",
    "        step2_data.update(mol_datasubmit)\n",
    "        step2_files.update(mol_files)\n",
    "        num_entries.append(i)\n",
    "        i+=1\n",
    "\n",
    "    # Introduce ligands: all will be defined as orthosteric ligands\n",
    "    lignames = []\n",
    "    for lig,(name,inchikey) in ligdict.items():\n",
    "        # Skip this if molecule is apoform\n",
    "        if apo:\n",
    "            break\n",
    "        # Avoid blacklisted molecules or ions\n",
    "        if (lig in detergent_blacklist) or (lig in glucids_blacklist)  or (len(lig) == 2):\n",
    "            continue\n",
    "        # Download ligand, store it into temporary file\n",
    "        response = requests.get('https://files.rcsb.org/ligands/view/'+lig+'_ideal.sdf')\n",
    "        with open('tmpfile.sdf','wb') as tmpout:\n",
    "            tmpout.write(response.content)\n",
    "\n",
    "        # CLR (cholesterol) is usually a experimental lipid \n",
    "        moltype = '3' if lig=='CLR' else '0'\n",
    "            \n",
    "        # Retrieve molecule information\n",
    "        (mol_datasubmit, mol_files) = smalmol_getdata(s, str(i), inchikey, subm_id, moltype, lig, 'tmpfile.sdf')\n",
    "        step2_data.update(mol_datasubmit)\n",
    "        step2_files.update(mol_files)\n",
    "        num_entries.append(i)\n",
    "        i+=1\n",
    "        os.remove('tmpfile.sdf')\n",
    "    \n",
    "    # Add number of entries\n",
    "    step2_data['num_entries'] = ','.join(map(str, num_entries))\n",
    "    \n",
    "    # Replace empty entries with an empty string\n",
    "    for key,val in step2_data.items():\n",
    "        if not val:\n",
    "            step2_data[key] = ''\n",
    "    \n",
    "    # Submit step2\n",
    "    rep = s.post(mainurl+'/dynadb/step2_submit/'+subm_id+'/',\n",
    "        headers = headers,\n",
    "        data = step2_data,\n",
    "        files = step2_files)\n",
    "    print('step2 finalized ',rep)\n",
    "\n",
    "    # Raise exceptions if something failed\n",
    "    if not rep.ok:\n",
    "        #s.post(mainurl+'/dynadb/delete_submission/'+subm_id)\n",
    "        raise Exception('Step 2 submitted returned %d'%rep.status_code)\n",
    "\n",
    "def new_step3(s, subm_id, pdbcode, protdict):\n",
    "    \"\"\"\n",
    "    Fullfil and sent the step3 (protein chains) of the new submission form\n",
    "    \"\"\"\n",
    "    print('initiating step 3: protein chains')\n",
    "\n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "    \n",
    "    # Start dict with the data to be send into GPCRmd\n",
    "    # But thougth they paved the footways here with goldust, I still would chose... my Isle of Innesfree\n",
    "    step3_data = {\n",
    "        'csrfmiddlewaretoken':csrftoken,\n",
    "        'submission_id' : subm_id,\n",
    "        'submit' : 'Submit',\n",
    "    }\n",
    "\n",
    "    # Iterate over pdb chains. For each one we'll create an entry\n",
    "    i=0\n",
    "    entrynum = []\n",
    "    for uniprotkbac,entry_data in protdict.items():\n",
    "        ii = str(i)\n",
    "        isgpcr = entry_data['isgpcr']\n",
    "        isoform = entry_data['isoform']\n",
    "        segs = entry_data['segs']\n",
    "        method_segs = entry_data['method_segs']\n",
    "\n",
    "        # Retrieve info using uniprot code\n",
    "        unidict = s.get(mainurl+'/dynadb/prot_info/'+subm_id+'/',\n",
    "                          params = {\n",
    "                              'uniprotkbac' : uniprotkbac,\n",
    "                              'isoform' : isoform,\n",
    "                              'submission_id' : subm_id,\n",
    "                          },\n",
    "                          headers = headers\n",
    "                         ).json()\n",
    "\n",
    "        # We'll obtain the alignment of our protein sequence and its UNIPROT sequence using\n",
    "        # GPCRmd's tool for do it\n",
    "        data_alig = {\n",
    "            'unisequence' : unidict['Sequence'],\n",
    "            'submission_id' : subm_id\n",
    "        }\n",
    "\n",
    "        # Get segments for alignment data\n",
    "        j = 0\n",
    "        segnums = []\n",
    "        for seg in segs:\n",
    "            ss = str(j)\n",
    "            data_alig['chain'+ss]= seg['chainid']\n",
    "            data_alig['segid'+ss]= seg['segid']\n",
    "            data_alig['from'+ss]= seg['beg']\n",
    "            data_alig['to'+ss]= seg['end']\n",
    "            segnums.append(ss)\n",
    "\n",
    "            # Store segment information as well in the submit dict\n",
    "            step3_segdata = {\n",
    "                ss+'seg'+ii: j,\n",
    "                ss+'pdbid'+ii: pdbcode,\n",
    "                ss+'sourcetype'+ii: method_segs,\n",
    "                ss+'chain'+ii: seg['chainid'],\n",
    "                ss+'segid'+ii: seg['segid'],\n",
    "                ss+'from_resid'+ii: seg['beg'],\n",
    "                ss+'to_resid'+ii: seg['end'],\n",
    "            }\n",
    "            step3_data.update(step3_segdata)\n",
    "\n",
    "            j += 1\n",
    "        data_alig['segnums']=','.join(segnums)\n",
    "        step3_data['num_segs'+ii]=','.join(segnums)\n",
    "\n",
    "        # Send alignment request\n",
    "        alignment = s.post(mainurl+'/dynadb/get_alignment/',\n",
    "                      data = data_alig,\n",
    "                      headers = headers\n",
    "                     ).text\n",
    "\n",
    "        # Get mutations\n",
    "        mutations_dict = s.post(mainurl+'/dynadb/protein/get_mutations/',\n",
    "                      data = {\n",
    "                          'alignment' : alignment,\n",
    "                          'sequence' : unidict['Sequence'],\n",
    "                      },\n",
    "                      headers = headers\n",
    "                     ).json()['mutations']\n",
    "\n",
    "        # Store mutations in submission dictionary\n",
    "        h = 0\n",
    "        mutnums = []\n",
    "        for mut in mutations_dict:\n",
    "            hh = str(h)\n",
    "            step3_mutdata = {\n",
    "                hh+'from'+ii : mut['from'],\n",
    "                hh+'to'+ii : mut['to'],\n",
    "                hh+'resid'+ii : mut['resid'],\n",
    "            }\n",
    "            step3_data.update(step3_mutdata)\n",
    "            mutnums.append(hh)\n",
    "            h+=1\n",
    "        step3_data['num_muts'+ii]=','.join(mutnums)\n",
    "\n",
    "        # Get chain name according to uniprot\n",
    "        names_array = unidict['Protein names'].split(\" (\")\n",
    "        names_array = list(map(lambda x: re.sub(\"\\)$|\\) \\[.*\\]$\", \"\", x), names_array)) # Convert the name string into name aray\n",
    "        name = names_array.pop(0) # We get the first name as the official one\n",
    "        other_names = ';'.join(names_array)\n",
    "        \n",
    "        # Start storing information to send in step3\n",
    "        step3_entrydata = {\n",
    "            'isoform'+ii : isoform,\n",
    "            'prot_uniprot'+ii : uniprotkbac,\n",
    "            'name'+ii : name,\n",
    "            'aliases'+ii : other_names,\n",
    "            'species_code'+ii : unidict['Organism ID'],\n",
    "            'species_name'+ii : unidict['Organism'],\n",
    "            'unisequence'+ii : unidict['Sequence'],\n",
    "        }\n",
    "        step3_data.update(step3_entrydata)\n",
    "\n",
    "        entrynum.append(ii)    \n",
    "        i+=1\n",
    "\n",
    "    # Store number of entries\n",
    "    step3_data['num_entries'] = ','.join(entrynum)\n",
    "\n",
    "    # Submit step3\n",
    "    rep = s.post(mainurl+'/dynadb/step3_submit/'+subm_id+'/',\n",
    "        headers = headers,\n",
    "        data = step3_data)\n",
    "    \n",
    "    print('step3 finalized ',rep)\n",
    "\n",
    "    # Raise exceptions if something failed\n",
    "    if not rep.ok:\n",
    "        #s.post(mainurl+'/dynadb/delete_submission/'+subm_id)\n",
    "        raise Exception('Step 3 submitted returned %d'%rep.status_code)\n",
    "\n",
    "def other_files_zip(prodpath):\n",
    "    \"\"\"\n",
    "    Compress protocol files in the path submitted \n",
    "    \"\"\"\n",
    "    outfolder = prodpath+'other_files'\n",
    "    outfile = outfolder+'.tar.gz'\n",
    "    if not os.path.exists(outfile):\n",
    "        os.makedirs(outfolder)\n",
    "        # For each replicate, create a folder with symbolic links of the to-compress files\n",
    "        for rep in [1,2,3]:\n",
    "            repath = '%srep_%d/' % (prodpath,rep)\n",
    "            outrep = '%s/rep_%d/' % (outfolder,rep)\n",
    "            os.makedirs(outrep)\n",
    "            list_files = ['input', 'input.coor', 'input.vel','input.xsc','log.txt']\n",
    "            for file in list_files:\n",
    "                shutil.copyfile(repath+file, outrep+file)\n",
    "\n",
    "        # Compress file\n",
    "        shutil.make_archive(outfolder, 'gztar', outfolder)\n",
    "        shutil.rmtree(outfolder)\n",
    "\n",
    "    return(outfile)\n",
    "\n",
    "def new_step4(s, subm_id, prodpath, repath):\n",
    "    \"\"\"\n",
    "    Fullfil and sent the step4 (files) of the new submission form\n",
    "    \"\"\"\n",
    "    print('initiating step 4: simulation files')\n",
    "\n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "\n",
    "    # Prepare trajectories\n",
    "    step4_files = []\n",
    "    for trajid in range(1,4):\n",
    "        trajpath = '%s/rep_%d/output_wrapped.xtc'%(prodpath,trajid)\n",
    "        step4_files.append(('trj', ('traj%d.xtc'%trajid, open(trajpath, 'rb'))))\n",
    "    \n",
    "    # Prepare the rest of files (parameters files will name-changed to avoid problems with filextensions)\n",
    "    step4_files.append(\n",
    "        ('dyn', ('structure.psf', open(repath+'structure.psf','rb')))\n",
    "    )\n",
    "    step4_files.append(\n",
    "        ('prm', ('parameters.prm', open(repath+'parameters','rb')))\n",
    "    )\n",
    "    other_files = other_files_zip(prodpath)\n",
    "    step4_files.append(\n",
    "        ('oth', ('other_files.tar.gz', open(other_files,'rb')))\n",
    "    )\n",
    "    \n",
    "    # Encode data\n",
    "    m=MultipartEncoder(fields=step4_files)\n",
    "    headers['Content-Type'] = m.content_type\n",
    "    \n",
    "    # Submit step4\n",
    "    rep = s.post(mainurl+'/dynadb/step4_submit/'+subm_id+'/',\n",
    "        headers = headers,\n",
    "        data = m)\n",
    "    \n",
    "    print('step4 finalized ',rep)\n",
    "\n",
    "    # Raise exceptions if something failed\n",
    "    if not rep.ok:\n",
    "        s.post(mainurl+'/dynadb/delete_submission/'+subm_id)\n",
    "        raise Exception('Step 4 submitted returned %d'%rep.status_code)\n",
    "    # If everything went ok so far, write a \"submitted\" file to know this system has already been submitted\n",
    "    else:\n",
    "        with open(prodpath+'submitted.txt','w') as out:\n",
    "            out.write(subm_id)\n",
    "\n",
    "def check_completeness(prodpath):\n",
    "    \"\"\"\n",
    "    Check if we actually have the three necessary replicates for submission\n",
    "    \"\"\"\n",
    "\n",
    "    incomplete = False\n",
    "    for trajid in range(1,4):\n",
    "        trajpath = '%s/rep_%d/output_wrapped.xtc'%(prodpath,trajid)\n",
    "        if not os.path.exists(trajpath):\n",
    "            incomplete = True\n",
    "\n",
    "    return(incomplete)            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "loging into GPCRmd\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 16:50:12,249 - moleculekit.readers - INFO - Attempting PDB query for 5OLV\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 5OLV_apo simulation ########################\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 16:50:13,824 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=\"all\" option when reading the file.\n",
      "2022-04-05 16:50:13,827 - moleculekit.molecule - INFO - Removed 78 atoms. 3629 atoms remaining in the molecule.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "new submission 1289 created\n",
      "1289 5OLV True\n",
      "step1 finalized  <Response [200]>\n",
      "initiating step 2: small molecule data\n",
      "step2 finalized  <Response [200]>\n",
      "initiating step 3: protein chains\n",
      "step3 finalized  <Response [200]>\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 16:51:45,592 - moleculekit.readers - WARNING - Non-integer values were read from the PDB \"serial\" field. Dropping PDB values and assigning new ones.\n",
      "2022-04-05 16:51:45,830 - moleculekit.readers - WARNING - Reading PDB file with more than 99999 atoms. Bond information can be wrong.\n",
      "2022-04-05 16:51:47,935 - moleculekit.readers - INFO - Attempting PDB query for 6D35\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 6D35_complex simulation ########################\n",
      "new submission 1290 created\n",
      "1290 6D35 False\n",
      "step1 finalized  <Response [200]>\n",
      "initiating step 2: small molecule data\n",
      "step2 finalized  <Response [200]>\n",
      "initiating step 3: protein chains\n",
      "step3 finalized  <Response [200]>\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 16:55:17,204 - moleculekit.readers - WARNING - Non-integer values were read from the PDB \"serial\" field. Dropping PDB values and assigning new ones.\n",
      "2022-04-05 16:55:17,406 - moleculekit.readers - WARNING - Reading PDB file with more than 99999 atoms. Bond information can be wrong.\n",
      "2022-04-05 16:55:19,143 - moleculekit.readers - INFO - Attempting PDB query for 6FJ3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 6FJ3_complex simulation ########################\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 16:55:20,922 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=\"all\" option when reading the file.\n",
      "2022-04-05 16:55:20,926 - moleculekit.molecule - INFO - Removed 8 atoms. 5344 atoms remaining in the molecule.\n",
      "2022-04-05 16:55:21,155 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue YCM\n",
      "2022-04-05 16:55:21,169 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AC5\n",
      "2022-04-05 16:55:21,171 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AIB\n",
      "2022-04-05 16:55:21,172 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue HRG\n",
      "2022-04-05 16:55:21,173 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue NLE\n",
      "2022-04-05 16:55:24,058 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AC5\n",
      "2022-04-05 16:55:24,059 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AIB\n",
      "2022-04-05 16:55:24,061 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue HRG\n",
      "2022-04-05 16:55:24,063 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue NLE\n",
      "2022-04-05 16:55:26,594 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AC5\n",
      "2022-04-05 16:55:26,595 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AIB\n",
      "2022-04-05 16:55:26,596 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue HRG\n",
      "2022-04-05 16:55:26,597 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue NLE\n",
      "2022-04-05 16:55:29,489 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AC5\n",
      "2022-04-05 16:55:29,490 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AIB\n",
      "2022-04-05 16:55:29,492 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue HRG\n",
      "2022-04-05 16:55:29,493 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue NLE\n",
      "2022-04-05 16:55:39,779 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AC5\n",
      "2022-04-05 16:55:39,781 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AIB\n",
      "2022-04-05 16:55:39,782 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue HRG\n",
      "2022-04-05 16:55:39,784 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue NLE\n",
      "2022-04-05 16:55:45,291 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AC5\n",
      "2022-04-05 16:55:45,293 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AIB\n",
      "2022-04-05 16:55:45,294 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue HRG\n",
      "2022-04-05 16:55:45,295 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue NLE\n",
      "2022-04-05 16:55:48,521 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AC5\n",
      "2022-04-05 16:55:48,522 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue AIB\n",
      "2022-04-05 16:55:48,524 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue HRG\n",
      "2022-04-05 16:55:48,526 - moleculekit.molecule - WARNING - Cannot provide one-letter code for non-standard residue NLE\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "new submission 1291 created\n",
      "1291 6FJ3 False\n",
      "step1 finalized  <Response [200]>\n",
      "initiating step 2: small molecule data\n",
      "step2 finalized  <Response [200]>\n",
      "initiating step 3: protein chains\n",
      "step3 finalized  <Response [200]>\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 16:58:20,663 - moleculekit.readers - INFO - Attempting PDB query for 6KQI\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 6KQI_apo simulation ########################\n",
      "new submission 1292 created\n",
      "1292 6KQI True\n",
      "step1 finalized  <Response [200]>\n",
      "initiating step 2: small molecule data\n",
      "step2 finalized  <Response [200]>\n",
      "initiating step 3: protein chains\n",
      "step3 finalized  <Response [200]>\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 16:59:39,827 - moleculekit.readers - INFO - Attempting PDB query for 6KQI\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 6KQI_complex simulation ########################\n",
      "new submission 1293 created\n",
      "1293 6KQI False\n",
      "step1 finalized  <Response [200]>\n",
      "initiating step 2: small molecule data\n",
      "step2 finalized  <Response [200]>\n",
      "initiating step 3: protein chains\n",
      "step3 finalized  <Response [200]>\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 17:01:06,023 - moleculekit.readers - WARNING - Non-integer values were read from the PDB \"serial\" field. Dropping PDB values and assigning new ones.\n",
      "2022-04-05 17:01:06,184 - moleculekit.readers - WARNING - Reading PDB file with more than 99999 atoms. Bond information can be wrong.\n",
      "2022-04-05 17:01:07,433 - moleculekit.readers - INFO - Attempting PDB query for 6LI0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 6LI0_apo simulation ########################\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 17:01:08,812 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=\"all\" option when reading the file.\n",
      "2022-04-05 17:01:08,815 - moleculekit.molecule - INFO - Removed 10 atoms. 3900 atoms remaining in the molecule.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "new submission 1294 created\n",
      "1294 6LI0 True\n",
      "step1 finalized  <Response [200]>\n",
      "initiating step 2: small molecule data\n",
      "step2 finalized  <Response [200]>\n",
      "initiating step 3: protein chains\n",
      "step3 finalized  <Response [200]>\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 17:03:12,077 - moleculekit.readers - WARNING - Non-integer values were read from the PDB \"serial\" field. Dropping PDB values and assigning new ones.\n",
      "2022-04-05 17:03:12,275 - moleculekit.readers - WARNING - Reading PDB file with more than 99999 atoms. Bond information can be wrong.\n",
      "2022-04-05 17:03:13,571 - moleculekit.readers - INFO - Attempting PDB query for 6LI0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 6LI0_complex simulation ########################\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-04-05 17:03:14,895 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=\"all\" option when reading the file.\n",
      "2022-04-05 17:03:14,898 - moleculekit.molecule - INFO - Removed 10 atoms. 3900 atoms remaining in the molecule.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "new submission 1295 created\n",
      "1295 6LI0 False\n",
      "step1 finalized  <Response [200]>\n",
      "initiating step 2: small molecule data\n",
      "step2 finalized  <Response [200]>\n",
      "initiating step 3: protein chains\n",
      "step3 finalized  <Response [200]>\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    }
   ],
   "source": [
    "#### Actual Simulation submission\n",
    "# mainurl = 'http://localhost:8000' \n",
    "mainurl = 'https://submission.gpcrmd.org'\n",
    "\n",
    "## Step -2: Login into GPCRmd\n",
    "with requests.Session() as s:\n",
    "    login(s)\n",
    "# For each of the currently-working-with systems defined in Part 1\n",
    "# for pdbcode in pdb_set:\n",
    "for pdbcode in [\"5OLV\",\"6D35\",\"6FJ3\",\"6KQI\",\"6LI0\",\"6LI0\",\"6LW5\"]:\n",
    "# bad_list = [('856','4EJ4',False),('1128','6LRY',False)]\n",
    "# for (subm_id,pdbcode, apo) in bad_list:\n",
    "#     for apo in [False]:\n",
    "    for apo in [True, False]:\n",
    "        \n",
    "        try:\n",
    "            # Set the path to the folder containing this simulation's files\n",
    "            sysname = pdbcode+\"_apo\" if apo else pdbcode+\"_complex\" \n",
    "            prodpath = resultspath+'production/'+pdbcode+'_apo/' if apo else resultspath+'production/'+pdbcode+'/' \n",
    "            repath = prodpath+'rep_1/'# For files no changing across representations\n",
    "            modelfolder = \"%sreceptor2curate_output/%s/\"%(basepath, pdbcode) \n",
    "            modelpath = \"%s%s_apo_curated.pdb\"%(modelfolder,pdbcode) if apo else \"%s%s_complex_curated.pdb\"%(modelfolder,pdbcode)\n",
    "            \n",
    "            # Skip if not all trajectories or already submitted or if already submitted\n",
    "            incomplete = check_completeness(prodpath)\n",
    "            submitted = os.path.exists(prodpath+'submitted.txt')\n",
    "            if incomplete or submitted:\n",
    "                continue            \n",
    "            \n",
    "            # Load molecule\n",
    "            if not os.path.exists(repath+'structure.pdb'):\n",
    "                continue\n",
    "            mymol = Molecule(repath+'structure.pdb')\n",
    "            print('\\n###############Submitting '+sysname+' simulation ########################')            \n",
    "\n",
    "            ## Step -1: Get information of protein chains and ligand molecules from PDB web\n",
    "            (protdict,ligdict,segtochain,method_id,sysname) = get_pdb_info(pdbcode, mymol, ligandsdict, blacklist, apo)\n",
    "            \n",
    "            ## Step 0: Create a new submission\n",
    "            subm_id = new_submission(s, mainurl)\n",
    "            print(subm_id, pdbcode, apo)\n",
    "#             subm_id = \"969\"\n",
    "\n",
    "            ## Step 1: General information\n",
    "            new_step1(s, subm_id, pdbcode, trajperiod, timestep, repath, modelpath, method_id, sysname, apo)\n",
    "            \n",
    "            ## Step 2: Small molecules \n",
    "            new_step2(s, subm_id, ligdict, apo)   \n",
    "\n",
    "            ## Step 3: Protein chains\n",
    "            new_step3(s, subm_id, pdbcode, protdict)\n",
    "\n",
    "            ## Step 4: Dynamics information\n",
    "            new_step4(s, subm_id, prodpath, repath)\n",
    "    \n",
    "        except Exception as e:\n",
    "            print(\"Simulation \"+sysname+\" could not be submitted because \",e)            \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name:  4Z9G\n",
      "Dyn_id:  1031\n",
      "Workbench:  <Response [500]>\n",
      "Report:  <Response [500]>\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for name,dyn_id in dynids.items(): \n",
    "    workbench_reponse = requests.get('https://submission.gpcrmd.org/view/'+str(dyn_id))\n",
    "    report_reponse = requests.get('https://submission.gpcrmd.org/dynadb/dynamics/id/'+str(dyn_id))                       \n",
    "    if not (workbench_reponse.ok and report_reponse.ok):\n",
    "        print('Name: ',name)\n",
    "        print(\"Dyn_id: \",dyn_id)\n",
    "        print(\"Workbench: \", workbench_reponse)\n",
    "        print(\"Report: \", report_reponse)\n",
    "        print(\"\\n\")\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
