{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/htmd/versionwarnings.py:30: UserWarning: As of HTMD 1.16 the default number of threads HTMD spawns for calculations is set to 1. You can enable parallelism at your own risk using `config(njobs=-2)` in the beginning of your scripts. To disable this warning run once: `from htmd import _disableWarnings; _disableWarnings('1.16');`\n",
      "  UserWarning,\n",
      "/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/htmd/versionwarnings.py:38: UserWarning: As of HTMD 1.21 support for ACEMD v2 has stopped. Please use ACEMD3 instead as well as the corresponding equilibration and production protocols. To disable this warning run once: `from htmd import _disableWarnings; _disableWarnings('1.21');`\n",
      "  UserWarning,\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Please cite HTMD: Doerr et al.(2016)JCTC,12,1845. https://dx.doi.org/10.1021/acs.jctc.6b00049\n",
      "\n",
      "HTMD Documentation at: https://www.htmd.org/docs/latest/\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-14 16:20:23,389 - binstar - INFO - Using Anaconda API: https://api.anaconda.org\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New stable HTMD version (2.0.3 python[3.8,<3.9.0a0,3.9,<3.10.0a0,3.7,<3.8.0a0]) is available. You are currently on (1.22.4).There are several methods to update:    - Create a new conda env. using `conda create -n htmd2.0.3 htmd=2.0.3 -c acellera -c psi4 -c conda-forge`    - Create a brand new conda installation and run `conda install htmd -c acellera -c psi4 -c conda-forge`    - Run: `conda update htmd -c acellera -c psi4 -c conda-forge` (NOT RECOMMENDED)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-14 16:20:25,535 - moleculekit.readers - WARNING - Element guessing failed for atom with name  DUM as the guessed element \"D\" was not found in the periodic table. Check for incorrect column alignment in the PDB file or report to moleculekit issue tracker.\n",
      "2022-02-14 16:20:26,202 - moleculekit.readers - WARNING - Element guessing failed for atom with name  DUM as the guessed element \"D\" was not found in the periodic table. Check for incorrect column alignment in the PDB file or report to moleculekit issue tracker.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: PATH=/soft/EB_repo/bio/structure/programs/goolf/1.7.20/Chimera/1.11.2/bin:/soft/system/easybuild/software/Miniconda3/4.7.10:/soft/system/easybuild/software/Miniconda3/4.7.10/bin:/soft/EB_repo/bio/structure/programs/foss/2016b/VMD/1.9.4a9/bin:/soft/system/lua/bin:/soft/system/luarocks/bin:/home/daranda/miniconda3/bin:/home/daranda/miniconda3/condabin:/usr/lib64/qt-3.3/bin:/soft/system/lua/bin:/soft/system/luarocks/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:/opt/puppetlabs/bin:/home/daranda/bin:/home/daranda/software/getcontacts-master:/usr/local/dowser/bin:/usr/local/dowser/bin/linux:/home/daranda/.local/bin:/home/daranda/bin:/home/daranda/bin:/home/daranda/software/getcontacts-master:/usr/local/dowser/bin:/usr/local/dowser/bin/linux:/gpcr/users/daranda/doctorat/GPCR_simulations/fake_slurm/\n"
     ]
    }
   ],
   "source": [
    "####################\n",
    "## Initial stuff\n",
    "####################\n",
    "\n",
    "# Load functions, modules and global variables required in our pipeline\n",
    "from simulate_structures_functions import *\n",
    "from importlib import reload  \n",
    "from htmd.builder.charmm import _recoverProtonations\n",
    "\n",
    "# Are we running curated structures??\n",
    "curated = True\n",
    "\n",
    "# PDB codes of the GPCRs to be simulated. \n",
    "# If no codes are provided, all avalible structures in GPCRdb will be used (except the ones already simulated)\n",
    "noncomplex = {\"3C9L\",\"3C9M\",\"6ZDR\",\"6ZDV\",\"6TPK\", \"6S0Q\", \"6V9S\", \"6WJC\", \"6W25\", \"6OBA\", \"6LW5\", \"6KP6\", \"6LUQ\", \"6LI1\", \"6LI0\", \"6LI2\", \"6KPC\", \"6LRY\", \"6KNM\", \"6TOT\", \"6TOS\", \"6TO7\", \"6TP6\", \"6TQ7\", \"6TOD\", \"6TP3\", \"6TQ4\", \"6TQ9\", \"6TP4\", \"6TQ6\", \"6TPN\", \"6TPG\", \"6TPJ\", \"6OL9\", \"6RZ6\", \"6RZ9\", \"6RZ7\", \"6RZ8\", \"6PT2\", \"6PT3\", \"6KUX\", \"6KUY\", \"6KUW\", \"6IQL\", \"6PS7\", \"6PS0\", \"6PS3\", \"6PS5\", \"6PS1\", \"6PS4\", \"6PRZ\", \"6PS2\", \"6PS6\", \"6KK1\", \"6KK7\", \"6KJV\", \"6PS8\", \"6JZH\", \"6RZ4\", \"6RZ5\", \"6KQI\", \"6K1Q\", \"6GT3\", \"6MH8\", \"6ME2\", \"6ME3\", \"6ME4\", \"6ME5\", \"6ME6\", \"6ME7\", \"6ME8\", \"6ME9\", \"6J21\", \"6J20\", \"6A94\", \"6A93\", \"5ZTY\", \"6HLP\", \"6HLO\", \"6HLL\", \"6GPX\", \"6GPS\", \"6IIV\", \"6IIU\", \"6E59\", \"6M9T\", \"6AK3\", \"5ZHP\", \"5ZKC\", \"5ZK3\", \"5YC8\", \"5ZK8\", \"5ZKB\", \"6IGK\", \"6IGL\", \"6FJ3\", \"6AKX\", \"6AKY\", \"6D27\", \"6D26\", \"6DRX\", \"6DS0\", \"6DRY\", \"6DRZ\", \"6BD4\", \"5ZKQ\", \"5ZKP\", \"6C1R\", \"6C1Q\", \"6D32\", \"6D35\", \"5KW2\", \"5ZBH\", \"5ZBQ\", \"6FK7\", \"6FKA\",  \"6FK6\", \"6FK9\", \"6FKC\", \"6FK8\", \"6FKB\", \"6CM4\", \"6FFI\", \"6FFH\", \"5WF5\", \"5WF6\", \"5V54\", \"5OLH\", \"5OLZ\", \"5OLO\", \"5OM1\", \"5OLG\", \"5OLV\", \"5OM4\", \"5YQZ\", \"6AQF\", \"5O9H\", \"5X33\", \"5VRA\", \"5WS3\", \"5WQC\", \"5WIV\", \"5WIU\", \"5NM4\", \"5NM2\", \"5NLX\", \"5X7D\", \"5X93\", \"5XPR\", \"5XSZ\", \"5N2S\", \"5MZP\", \"5MZJ\", \"5N2R\", \"5XRA\", \"5XR8\", \"5UIW\", \"5NX2\", \"5TZY\", \"5TZR\", \"5JTB\", \"5VBL\", \"5UVI\", \"5VEW\", \"5V56\", \"5V57\", \"5VEX\", \"5NDD\", \"5NDZ\", \"5UNH\", \"5UNF\", \"5UNG\", \"5TE3\", \"5TE5\", \"5UEN\", \"5UIG\", \"5TVN\", \"5T04\", \"5T1A\", \"5U09\", \"5TGZ\", \"5K2C\", \"5K2B\", \"5K2A\", \"5K2D\", \"5GLI\", \"5GLH\", \"5D6L\", \"5DYS\", \"5L7D\", \"5L7I\", \"5IU7\", \"5IUB\", \"5IU8\", \"5IU4\", \"5IUA\", \"4Z9G\", \"5EE7\", \"5DSG\", \"5CXV\", \"4ZJ8\", \"4ZJC\", \"5F8U\", \"5DHG\", \"5DHH\", \"4ZUD\", \"5A8E\", \"5CGC\", \"5CGD\", \"4XEE\", \"4XES\", \"4Z35\", \"4Z34\", \"4Z36\", \"4YAY\", \"4UHR\", \"4XNW\", \"4XNV\", \"4XT3\", \"4RWS\", \"4RWD\", \"4S0V\", \"4U15\", \"4U16\", \"4QIM\", \"4QIN\", \"4PHU\", \"4OO9\", \"4PXZ\", \"4PY0\", \"4BVN\", \"4NTJ\", \"4OR2\", \"4O9R\", \"4BUO\", \"4N4W\", \"4N6H\", \"4NC3\", \"4MBS\", \"4L6R\", \"4K5Y\", \"4JKV\", \"3ZPR\", \"3ZPQ\", \"4IAQ\", \"4IAR\", \"4IB4\", \"4GPO\", \"3VW7\", \"4GBR\", \"4GRV\", \"4EIY\", \"4AMJ\", \"4AMI\", \"4EJ4\", \"4EA3\", \"3UZA\", \"3UZC\", \"4DJH\", \"4DKL\", \"3V2Y\", \"3UON\", \"3REY\", \"3RFM\", \"3RZE\", \"2YCZ\", \"2YCW\", \"2YDV\", \"2YDO\", \"3QAK\", \"2Y04\", \"2Y00\", \"2Y03\", \"2Y02\", \"3PDS\", \"3PBL\", \"3ODU\", \"3OE0\", \"3NY9\", \"3NY8\", \"3NYA\", \"3D4S\", \"2Z73\", \"2RH1\", \"1U19\", \"1GZM\"}\n",
    "noncomplex_nonGPCRmd = {\"3C9L\",\"3C9M\",\"6ZDR\",\"6ZDV\",'6RZ8', '6FFI', '6KJV', '6TQ6', '6DRY', '5WIV', '5V57', '6LI2', '6MH8', '5MZJ', '5KW2', '5X33', '5ZHP', '6PS6', '6FJ3', '6D35', '4NTJ', '6A94', '6AK3', '5UVI', '6FFH', '5WIU', '6KQI', '5XR8', '6RZ9', '5VEX', '6LW5', '6FKA', '6PS7', '6M9T', '6KNM', '6ME6', '6GPX', '5XSZ', '5K2A', '5YQZ', '6PS0', '5K2C', '6ME7', '2Z73', '6D32', '6RZ4', '5YC8', '5T1A', '6AKY', '4N4W', '5WS3', '6LRY', '6J21', '5TE5', '4O9R', '6FK6', '1GZM', '6DRX', '6TQ7', '5NDZ', '6IGL', '5VRA', '5ZBQ', '6TPN', '5ZTY', '6DS0', '5NLX', '5X93', '6HLP', '5OLG', '5OM4', '5O9H', '5XPR', '6A93', '5GLI', '6OBA', '6D27', '6HLO', '6RZ5', '5TE3', '6IIV', '6KP6', '6TQ4', '6TPJ', '5V56', '5UNG', '6FK9', '6PS8', '2YCZ', '6JZH', '5XRA', '5ZKB', '4JKV', '5MZP', '5K2D', '5OLV', '5X7D', '5T04', '5OLO', '5DYS', '6FK7', '6J20', '6K1Q', '5EE7', '6D26', '6PS1', '6DRZ', '4GBR', '6FKC', '6TP6', '6LUQ', '5ZKQ', '5UNH', '6PRZ', '5UNF', '5NX2', '6LI0', '5NM2', '5D6L', '6AQF', '6KK1', '6C1Q', '4Z9G', '6LI1', '5WF5', '6FKB',  '5OLH', '4XT3', '5ZK3', '5VEW', '4XES', '6HLL', '6TOT', '5N2R', '5WF6', '5K2B', '6TPG', '6PS4', '5F8U', '6ME4', '6GT3', '6RZ6', '6ME8', '6ME2', '5VBL', '5N2S', '5ZBH', '6PS3', '6IIU', '4NC3', '5JTB', '5ZKP', '6TQ9', '6GPS', '6TP3', '6RZ7', '6ME3', '5ZK8', '6C1R', '5V54', '5UIG', '5TVN', '5OM1', '6AKX', '5ZKC', '5NM4', '5TZY', '6TOD', '4GPO', '6TO7', '6CM4', '6TOS', '6PS5', '6ME9', '5NDD', '6KPC', '6KK7', '4BUO', '6OL9', '6PS2', '6FK8', '6ME5', '6TP4', '4QIM', '6IGK', '5WQC', '5UIW', '5TZR', '5OLZ', '4EJ4', '6E59'}\n",
    "noncomplex_nonGPCRmd_agonist = {'5T04', '6ME9', '4BUO', '6ME6', '5VBL', '5TZR', '6D35', '6LW5', '5YQZ', '6ME3', '5WF6', '6D32', '5TE5', '4XES', '6AK3', '5DYS', '5KW2', '5WF5', '6FK6', '6ME7', '4NC3', '6ME2', '6FKA', '6DRY', '6FK7', '5TVN', '6KPC', '6KQI', '6LRY', '6IGL', '5NX2', '6M9T', '6FKC', '6FKB', '6FJ3', '6IGK', '6KNM', '6FK9', '6ME8', '6ME5', '6ME4', '5XRA', '5N2S', '5TZY', '6FK8', '6LI0', '5XR8'}\n",
    "pdb_set = noncomplex_nonGPCRmd\n",
    "\n",
    "# Our main path\n",
    "basepath = basepath = os.getcwd()+'/'\n",
    "# Other Paths\n",
    "strucpath = basepath + 'data_structures/'\n",
    "resultspath = basepath + 'simulation_output/'\n",
    "membranepdb = basepath + 'membrane/popc36_box_renumbered.pdb'\n",
    "topparpath = basepath + 'toppar/TOP_PARAMS_ACE3/'#toppar= topology+parameters\n",
    "ligandsdict_path = basepath + 'ligands.json'\n",
    "modres_path = basepath + 'modified_residues.json'\n",
    "slurmpath = basepath+'fake_slurm/'\n",
    "# Path to slurm queing system binaries\n",
    "# In our case, Ismael designed a bunch of small bash scripts (fake_slurm) which do ssh to Hydra and execute slurm there\n",
    "path= os.environ['PATH']\n",
    "\n",
    "# Modify path to include fake slurm\n",
    "%env PATH=$path:$slurmpath\n",
    "\n",
    "# Load topology, parameter and stream files with our current basepath\n",
    "topos = [os.path.join(topparpath,file) for file in toposfilenames] \n",
    "params = [os.path.join(topparpath,file) for file in paramsfilenames]\n",
    "streams = [os.path.join(topparpath,file) for file in streamsfilenames]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading 5WS3 structure (1/191)\n",
      "Structure for 5WS3 already present. Skipping...\n",
      "Downloading 6TP3 structure (2/191)\n",
      "Structure for 6TP3 already present. Skipping...\n",
      "Downloading 5T1A structure (3/191)\n",
      "Structure for 5T1A already present. Skipping...\n",
      "Downloading 6J21 structure (4/191)\n",
      "Structure for 6J21 already present. Skipping...\n",
      "Downloading 4O9R structure (5/191)\n",
      "Structure for 4O9R already present. Skipping...\n",
      "Downloading 6RZ7 structure (6/191)\n",
      "Structure for 6RZ7 already present. Skipping...\n",
      "Downloading 6KP6 structure (7/191)\n",
      "Structure for 6KP6 already present. Skipping...\n",
      "Downloading 6D35 structure (8/191)\n",
      "Structure for 6D35 already present. Skipping...\n",
      "Downloading 6RZ5 structure (9/191)\n",
      "Structure for 6RZ5 already present. Skipping...\n",
      "Downloading 6PRZ structure (10/191)\n",
      "Structure for 6PRZ already present. Skipping...\n",
      "Downloading 5K2C structure (11/191)\n",
      "Structure for 5K2C already present. Skipping...\n",
      "Downloading 5UIG structure (12/191)\n",
      "Structure for 5UIG already present. Skipping...\n",
      "Downloading 6TOD structure (13/191)\n",
      "Structure for 6TOD already present. Skipping...\n",
      "Downloading 6LW5 structure (14/191)\n",
      "Structure for 6LW5 already present. Skipping...\n",
      "Downloading 4XES structure (15/191)\n",
      "Structure for 4XES already present. Skipping...\n",
      "Downloading 5X33 structure (16/191)\n",
      "Structure for 5X33 already present. Skipping...\n",
      "Downloading 6D32 structure (17/191)\n",
      "Structure for 6D32 already present. Skipping...\n",
      "Downloading 6FK6 structure (18/191)\n",
      "Structure for 6FK6 already present. Skipping...\n",
      "Downloading 6TP6 structure (19/191)\n",
      "Structure for 6TP6 already present. Skipping...\n",
      "Downloading 6AKY structure (20/191)\n",
      "Structure for 6AKY already present. Skipping...\n",
      "Downloading 5DYS structure (21/191)\n",
      "Structure for 5DYS already present. Skipping...\n",
      "Downloading 6LI2 structure (22/191)\n",
      "Structure for 6LI2 already present. Skipping...\n",
      "Downloading 4GBR structure (23/191)\n",
      "Structure for 4GBR already present. Skipping...\n",
      "Downloading 6TQ9 structure (24/191)\n",
      "Structure for 6TQ9 already present. Skipping...\n",
      "Downloading 6TPJ structure (25/191)\n",
      "Structure for 6TPJ already present. Skipping...\n",
      "Downloading 6KK7 structure (26/191)\n",
      "Structure for 6KK7 already present. Skipping...\n",
      "Downloading 6PS2 structure (27/191)\n",
      "Structure for 6PS2 already present. Skipping...\n",
      "Downloading 5TE5 structure (28/191)\n",
      "Structure for 5TE5 already present. Skipping...\n",
      "Downloading 6D27 structure (29/191)\n",
      "Structure for 6D27 already present. Skipping...\n",
      "Downloading 6FK7 structure (30/191)\n",
      "Structure for 6FK7 already present. Skipping...\n",
      "Downloading 4Z9G structure (31/191)\n",
      "Structure for 4Z9G already present. Skipping...\n",
      "Downloading 4N4W structure (32/191)\n",
      "Structure for 4N4W already present. Skipping...\n",
      "Downloading 6OBA structure (33/191)\n",
      "Structure for 6OBA already present. Skipping...\n",
      "Downloading 6GT3 structure (34/191)\n",
      "Structure for 6GT3 already present. Skipping...\n",
      "Downloading 4BUO structure (35/191)\n",
      "Structure for 4BUO already present. Skipping...\n",
      "Downloading 6CM4 structure (36/191)\n",
      "Structure for 6CM4 already present. Skipping...\n",
      "Downloading 5UNF structure (37/191)\n",
      "Structure for 5UNF already present. Skipping...\n",
      "Downloading 5VRA structure (38/191)\n",
      "Structure for 5VRA already present. Skipping...\n",
      "Downloading 4XT3 structure (39/191)\n",
      "Structure for 4XT3 already present. Skipping...\n",
      "Downloading 6ME8 structure (40/191)\n",
      "Structure for 6ME8 already present. Skipping...\n",
      "Downloading 5WIV structure (41/191)\n",
      "Structure for 5WIV already present. Skipping...\n",
      "Downloading 5X93 structure (42/191)\n",
      "Structure for 5X93 already present. Skipping...\n",
      "Downloading 5UNH structure (43/191)\n",
      "Structure for 5UNH already present. Skipping...\n",
      "Downloading 6KNM structure (44/191)\n",
      "Structure for 6KNM already present. Skipping...\n",
      "Downloading 6TQ7 structure (45/191)\n",
      "Structure for 6TQ7 already present. Skipping...\n",
      "Downloading 6PS4 structure (46/191)\n",
      "Structure for 6PS4 already present. Skipping...\n",
      "Downloading 4NC3 structure (47/191)\n",
      "Structure for 4NC3 already present. Skipping...\n",
      "Downloading 5KW2 structure (48/191)\n",
      "Structure for 5KW2 already present. Skipping...\n",
      "Downloading 6PS6 structure (49/191)\n",
      "Structure for 6PS6 already present. Skipping...\n",
      "Downloading 5F8U structure (50/191)\n",
      "Structure for 5F8U already present. Skipping...\n",
      "Downloading 4NTJ structure (51/191)\n",
      "Structure for 4NTJ already present. Skipping...\n",
      "Downloading 6KQI structure (52/191)\n",
      "Structure for 6KQI already present. Skipping...\n",
      "Downloading 5GLI structure (53/191)\n",
      "Structure for 5GLI already present. Skipping...\n",
      "Downloading 6TPN structure (54/191)\n",
      "Structure for 6TPN already present. Skipping...\n",
      "Downloading 5V56 structure (55/191)\n",
      "Structure for 5V56 already present. Skipping...\n",
      "Downloading 6AKX structure (56/191)\n",
      "Structure for 6AKX already present. Skipping...\n",
      "Downloading 5WF6 structure (57/191)\n",
      "Structure for 5WF6 already present. Skipping...\n",
      "Downloading 5TZY structure (58/191)\n",
      "Structure for 5TZY already present. Skipping...\n",
      "Downloading 1GZM structure (59/191)\n",
      "Structure for 1GZM already present. Skipping...\n",
      "Downloading 5XRA structure (60/191)\n",
      "Structure for 5XRA already present. Skipping...\n",
      "Downloading 5OLV structure (61/191)\n",
      "Structure for 5OLV already present. Skipping...\n",
      "Downloading 6ZDV structure (62/191)\n",
      "Structure for 6ZDV already present. Skipping...\n",
      "Downloading 6K1Q structure (63/191)\n",
      "Structure for 6K1Q already present. Skipping...\n",
      "Downloading 5TVN structure (64/191)\n",
      "Structure for 5TVN already present. Skipping...\n",
      "Downloading 6M9T structure (65/191)\n",
      "Structure for 6M9T already present. Skipping...\n",
      "Downloading 6TOT structure (66/191)\n",
      "Structure for 6TOT already present. Skipping...\n",
      "Downloading 6KPC structure (67/191)\n",
      "Structure for 6KPC already present. Skipping...\n",
      "Downloading 6ME5 structure (68/191)\n",
      "Structure for 6ME5 already present. Skipping...\n",
      "Downloading 6TP4 structure (69/191)\n",
      "Structure for 6TP4 already present. Skipping...\n",
      "Downloading 6RZ8 structure (70/191)\n",
      "Structure for 6RZ8 already present. Skipping...\n",
      "Downloading 3C9L structure (71/191)\n",
      "Structure for 3C9L already present. Skipping...\n",
      "Downloading 6HLO structure (72/191)\n",
      "Structure for 6HLO already present. Skipping...\n",
      "Downloading 6C1R structure (73/191)\n",
      "Structure for 6C1R already present. Skipping...\n",
      "Downloading 4QIM structure (74/191)\n",
      "Structure for 4QIM already present. Skipping...\n",
      "Downloading 6PS7 structure (75/191)\n",
      "Structure for 6PS7 already present. Skipping...\n",
      "Downloading 6DRX structure (76/191)\n",
      "Structure for 6DRX already present. Skipping...\n",
      "Downloading 5N2R structure (77/191)\n",
      "Structure for 5N2R already present. Skipping...\n",
      "Downloading 5UIW structure (78/191)\n",
      "Structure for 5UIW already present. Skipping...\n",
      "Downloading 5ZHP structure (79/191)\n",
      "Structure for 5ZHP already present. Skipping...\n",
      "Downloading 5K2D structure (80/191)\n",
      "Structure for 5K2D already present. Skipping...\n",
      "Downloading 5WQC structure (81/191)\n",
      "Structure for 5WQC already present. Skipping...\n",
      "Downloading 6FKB structure (82/191)\n",
      "Structure for 6FKB already present. Skipping...\n",
      "Downloading 5NM4 structure (83/191)\n",
      "Structure for 5NM4 already present. Skipping...\n",
      "Downloading 6MH8 structure (84/191)\n",
      "Structure for 6MH8 already present. Skipping...\n",
      "Downloading 6TPG structure (85/191)\n",
      "Structure for 6TPG already present. Skipping...\n",
      "Downloading 6FKC structure (86/191)\n",
      "Structure for 6FKC already present. Skipping...\n",
      "Downloading 5ZK8 structure (87/191)\n",
      "Structure for 5ZK8 already present. Skipping...\n",
      "Downloading 5WIU structure (88/191)\n",
      "Structure for 5WIU already present. Skipping...\n",
      "Downloading 6FK9 structure (89/191)\n",
      "Structure for 6FK9 already present. Skipping...\n",
      "Downloading 6TQ6 structure (90/191)\n",
      "Structure for 6TQ6 already present. Skipping...\n",
      "Downloading 6GPS structure (91/191)\n",
      "Structure for 6GPS already present. Skipping...\n",
      "Downloading 6KK1 structure (92/191)\n",
      "Structure for 6KK1 already present. Skipping...\n",
      "Downloading 6RZ9 structure (93/191)\n",
      "Structure for 6RZ9 already present. Skipping...\n",
      "Downloading 6KJV structure (94/191)\n",
      "Structure for 6KJV already present. Skipping...\n",
      "Downloading 5TE3 structure (95/191)\n",
      "Structure for 5TE3 already present. Skipping...\n",
      "Downloading 4GPO structure (96/191)\n",
      "Structure for 4GPO already present. Skipping...\n",
      "Downloading 6ME9 structure (97/191)\n",
      "Structure for 6ME9 already present. Skipping...\n",
      "Downloading 5ZBQ structure (98/191)\n",
      "Structure for 5ZBQ already present. Skipping...\n",
      "Downloading 5XSZ structure (99/191)\n",
      "Structure for 5XSZ already present. Skipping...\n",
      "Downloading 6GPX structure (100/191)\n",
      "Structure for 6GPX already present. Skipping...\n",
      "Downloading 6DRZ structure (101/191)\n",
      "Structure for 6DRZ already present. Skipping...\n",
      "Downloading 5OLH structure (102/191)\n",
      "Structure for 5OLH already present. Skipping...\n",
      "Downloading 5N2S structure (103/191)\n",
      "Structure for 5N2S already present. Skipping...\n",
      "Downloading 5VEW structure (104/191)\n",
      "Structure for 5VEW already present. Skipping...\n",
      "Downloading 5YC8 structure (105/191)\n",
      "Structure for 5YC8 already present. Skipping...\n",
      "Downloading 5O9H structure (106/191)\n",
      "Structure for 5O9H already present. Skipping...\n",
      "Downloading 6LI0 structure (107/191)\n",
      "Structure for 6LI0 already present. Skipping...\n",
      "Downloading 5NM2 structure (108/191)\n",
      "Structure for 5NM2 already present. Skipping...\n",
      "Downloading 5D6L structure (109/191)\n",
      "Structure for 5D6L already present. Skipping...\n",
      "Downloading 6TQ4 structure (110/191)\n",
      "Structure for 6TQ4 already present. Skipping...\n",
      "Downloading 6FFH structure (111/191)\n",
      "Structure for 6FFH already present. Skipping...\n",
      "Downloading 5VEX structure (112/191)\n",
      "Structure for 5VEX already present. Skipping...\n",
      "Downloading 6ME4 structure (113/191)\n",
      "Structure for 6ME4 already present. Skipping...\n",
      "Downloading 5MZJ structure (114/191)\n",
      "Structure for 5MZJ already present. Skipping...\n",
      "Downloading 5OM1 structure (115/191)\n",
      "Structure for 5OM1 already present. Skipping...\n",
      "Downloading 5TZR structure (116/191)\n",
      "Structure for 5TZR already present. Skipping...\n",
      "Downloading 6TO7 structure (117/191)\n",
      "Structure for 6TO7 already present. Skipping...\n",
      "Downloading 5ZKQ structure (118/191)\n",
      "Structure for 5ZKQ already present. Skipping...\n",
      "Downloading 6FKA structure (119/191)\n",
      "Structure for 6FKA already present. Skipping...\n",
      "Downloading 6PS8 structure (120/191)\n",
      "Structure for 6PS8 already present. Skipping...\n",
      "Downloading 6AK3 structure (121/191)\n",
      "Structure for 6AK3 already present. Skipping...\n",
      "Downloading 3C9M structure (122/191)\n",
      "Structure for 3C9M already present. Skipping...\n",
      "Downloading 2YCZ structure (123/191)\n",
      "Structure for 2YCZ already present. Skipping...\n",
      "Downloading 5ZK3 structure (124/191)\n",
      "Structure for 5ZK3 already present. Skipping...\n",
      "Downloading 6RZ6 structure (125/191)\n",
      "Structure for 6RZ6 already present. Skipping...\n",
      "Downloading 6HLL structure (126/191)\n",
      "Structure for 6HLL already present. Skipping...\n",
      "Downloading 6FJ3 structure (127/191)\n",
      "Structure for 6FJ3 already present. Skipping...\n",
      "Downloading 5YQZ structure (128/191)\n",
      "Structure for 5YQZ already present. Skipping...\n",
      "Downloading 6AQF structure (129/191)\n",
      "Structure for 6AQF already present. Skipping...\n",
      "Downloading 5XPR structure (130/191)\n",
      "Structure for 5XPR already present. Skipping...\n",
      "Downloading 6DRY structure (131/191)\n",
      "Structure for 6DRY already present. Skipping...\n",
      "Downloading 5K2A structure (132/191)\n",
      "Structure for 5K2A already present. Skipping...\n",
      "Downloading 6OL9 structure (133/191)\n",
      "Structure for 6OL9 already present. Skipping...\n",
      "Downloading 6LRY structure (134/191)\n",
      "Structure for 6LRY already present. Skipping...\n",
      "Downloading 5MZP structure (135/191)\n",
      "Structure for 5MZP already present. Skipping...\n",
      "Downloading 5ZKP structure (136/191)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "something failed in downloading refined structure of 5ZKP: 'NoneType' object has no attribute 'get'\n",
      "Downloading 6FK8 structure (137/191)\n",
      "Structure for 6FK8 already present. Skipping...\n",
      "Downloading 6A94 structure (138/191)\n",
      "Structure for 6A94 already present. Skipping...\n",
      "Downloading 5ZTY structure (139/191)\n",
      "Structure for 5ZTY already present. Skipping...\n",
      "Downloading 6ME7 structure (140/191)\n",
      "Structure for 6ME7 already present. Skipping...\n",
      "Downloading 5V57 structure (141/191)\n",
      "Structure for 5V57 already present. Skipping...\n",
      "Downloading 4EJ4 structure (142/191)\n",
      "Structure for 4EJ4 already present. Skipping...\n",
      "Downloading 5WF5 structure (143/191)\n",
      "Structure for 5WF5 already present. Skipping...\n",
      "Downloading 5X7D structure (144/191)\n",
      "Structure for 5X7D already present. Skipping...\n",
      "Downloading 6PS3 structure (145/191)\n",
      "Structure for 6PS3 already present. Skipping...\n",
      "Downloading 6ZDR structure (146/191)\n",
      "Structure for 6ZDR already present. Skipping...\n",
      "Downloading 6IIV structure (147/191)\n",
      "Structure for 6IIV already present. Skipping...\n",
      "Downloading 6JZH structure (148/191)\n",
      "Structure for 6JZH already present. Skipping...\n",
      "Downloading 5XR8 structure (149/191)\n",
      "Structure for 5XR8 already present. Skipping...\n",
      "Downloading 5UNG structure (150/191)\n",
      "Structure for 5UNG already present. Skipping...\n",
      "Downloading 5EE7 structure (151/191)\n",
      "Structure for 5EE7 already present. Skipping...\n",
      "Downloading 6HLP structure (152/191)\n",
      "Structure for 6HLP already present. Skipping...\n",
      "Downloading 6LI1 structure (153/191)\n",
      "Structure for 6LI1 already present. Skipping...\n",
      "Downloading 5T04 structure (154/191)\n",
      "Structure for 5T04 already present. Skipping...\n",
      "Downloading 2Z73 structure (155/191)\n",
      "Structure for 2Z73 already present. Skipping...\n",
      "Downloading 5ZKC structure (156/191)\n",
      "Structure for 5ZKC already present. Skipping...\n",
      "Downloading 6TOS structure (157/191)\n",
      "Structure for 6TOS already present. Skipping...\n",
      "Downloading 5NX2 structure (158/191)\n",
      "Structure for 5NX2 already present. Skipping...\n",
      "Downloading 5NDD structure (159/191)\n",
      "Structure for 5NDD already present. Skipping...\n",
      "Downloading 6LUQ structure (160/191)\n",
      "Structure for 6LUQ already present. Skipping...\n",
      "Downloading 6PS0 structure (161/191)\n",
      "Structure for 6PS0 already present. Skipping...\n",
      "Downloading 5UVI structure (162/191)\n",
      "Structure for 5UVI already present. Skipping...\n",
      "Downloading 6C1Q structure (163/191)\n",
      "Structure for 6C1Q already present. Skipping...\n",
      "Downloading 6DS0 structure (164/191)\n",
      "Structure for 6DS0 already present. Skipping...\n",
      "Downloading 5OLO structure (165/191)\n",
      "Structure for 5OLO already present. Skipping...\n",
      "Downloading 6PS5 structure (166/191)\n",
      "Structure for 6PS5 already present. Skipping...\n",
      "Downloading 6E59 structure (167/191)\n",
      "Structure for 6E59 already present. Skipping...\n",
      "Downloading 5ZKB structure (168/191)\n",
      "Structure for 5ZKB already present. Skipping...\n",
      "Downloading 6ME6 structure (169/191)\n",
      "Structure for 6ME6 already present. Skipping...\n",
      "Downloading 6J20 structure (170/191)\n",
      "Structure for 6J20 already present. Skipping...\n",
      "Downloading 6IGL structure (171/191)\n",
      "Structure for 6IGL already present. Skipping...\n",
      "Downloading 5ZBH structure (172/191)\n",
      "Structure for 5ZBH already present. Skipping...\n",
      "Downloading 5NLX structure (173/191)\n",
      "Structure for 5NLX already present. Skipping...\n",
      "Downloading 5JTB structure (174/191)\n",
      "Structure for 5JTB already present. Skipping...\n",
      "Downloading 5V54 structure (175/191)\n",
      "Structure for 5V54 already present. Skipping...\n",
      "Downloading 6FFI structure (176/191)\n",
      "Structure for 6FFI already present. Skipping...\n",
      "Downloading 6IIU structure (177/191)\n",
      "Structure for 6IIU already present. Skipping...\n",
      "Downloading 4JKV structure (178/191)\n",
      "Structure for 4JKV already present. Skipping...\n",
      "Downloading 5NDZ structure (179/191)\n",
      "Structure for 5NDZ already present. Skipping...\n",
      "Downloading 6IGK structure (180/191)\n",
      "Structure for 6IGK already present. Skipping...\n",
      "Downloading 6RZ4 structure (181/191)\n",
      "Structure for 6RZ4 already present. Skipping...\n",
      "Downloading 6A93 structure (182/191)\n",
      "Structure for 6A93 already present. Skipping...\n",
      "Downloading 5OLZ structure (183/191)\n",
      "Structure for 5OLZ already present. Skipping...\n",
      "Downloading 5OLG structure (184/191)\n",
      "Structure for 5OLG already present. Skipping...\n",
      "Downloading 6PS1 structure (185/191)\n",
      "Structure for 6PS1 already present. Skipping...\n",
      "Downloading 5OM4 structure (186/191)\n",
      "Structure for 5OM4 already present. Skipping...\n",
      "Downloading 5VBL structure (187/191)\n",
      "Structure for 5VBL already present. Skipping...\n",
      "Downloading 5K2B structure (188/191)\n",
      "Structure for 5K2B already present. Skipping...\n",
      "Downloading 6ME3 structure (189/191)\n",
      "Structure for 6ME3 already present. Skipping...\n",
      "Downloading 6ME2 structure (190/191)\n",
      "Structure for 6ME2 already present. Skipping...\n",
      "Downloading 6D26 structure (191/191)\n",
      "Structure for 6D26 already present. Skipping...\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6GPX/6GPX_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6KK7/6KK7_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4N4W/4N4W_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TPJ/6TPJ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6IIV/6IIV_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6JZH/6JZH_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6RZ6/6RZ6_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5UIW/5UIW_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6AKY/6AKY_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5KW2/5KW2_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS7/6PS7_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5WS3/5WS3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FK8/6FK8_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TQ6/6TQ6_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6IIU/6IIU_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5XPR/5XPR_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6D26/6D26_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6OL9/6OL9_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZK3/5ZK3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5NLX/5NLX_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5UNF/5UNF_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5NDD/5NDD_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5GLI/5GLI_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FKA/6FKA_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZK8/5ZK8_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FK7/6FK7_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ME5/6ME5_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS2/6PS2_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6KNM/6KNM_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6AK3/6AK3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5K2A/5K2A_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5OLO/5OLO_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ME8/6ME8_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5VEX/5VEX_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6GT3/6GT3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5WIU/5WIU_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6D35/6D35_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5V57/5V57_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZKB/5ZKB_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5OLV/5OLV_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5TE3/5TE3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6RZ5/6RZ5_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ME9/6ME9_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TP6/6TP6_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ME2/6ME2_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4GPO/4GPO_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6LI0/6LI0_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6HLP/6HLP_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5K2B/5K2B_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ME4/6ME4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5F8U/5F8U_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4Z9G/4Z9G_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5V56/5V56_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZBH/5ZBH_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5NM2/5NM2_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5DYS/5DYS_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4O9R/4O9R_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6IGL/6IGL_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5O9H/5O9H_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6DRX/6DRX_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4QIM/4QIM_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS6/6PS6_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4EJ4/4EJ4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6LI1/6LI1_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5N2R/5N2R_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6DS0/6DS0_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PRZ/6PRZ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5WF6/5WF6_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6J21/6J21_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6C1Q/6C1Q_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5VEW/5VEW_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6KPC/6KPC_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5XSZ/5XSZ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5YC8/5YC8_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5OLG/5OLG_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5MZJ/5MZJ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5YQZ/5YQZ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5XRA/5XRA_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS3/6PS3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4BUO/4BUO_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6A94/6A94_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5TZY/5TZY_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6K1Q/6K1Q_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6RZ8/6RZ8_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5JTB/5JTB_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6D32/6D32_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5X7D/5X7D_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TOT/6TOT_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6LRY/6LRY_refined.pdb was not found.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6KQI/6KQI_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ZDV/6ZDV_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TPN/6TPN_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5WIV/5WIV_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4NTJ/4NTJ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6HLO/6HLO_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6C1R/6C1R_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5V54/5V54_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6LI2/6LI2_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FJ3/6FJ3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4NC3/4NC3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5TVN/5TVN_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6RZ7/6RZ7_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FKB/6FKB_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6KJV/6KJV_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5XR8/5XR8_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6M9T/6M9T_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5T1A/5T1A_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6CM4/6CM4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4GBR/4GBR_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6J20/6J20_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5VRA/5VRA_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZBQ/5ZBQ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5UIG/5UIG_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TOS/6TOS_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ZDR/6ZDR_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6KK1/6KK1_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TO7/6TO7_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TQ7/6TQ7_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6OBA/6OBA_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6A93/6A93_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6DRY/6DRY_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5OLZ/5OLZ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5NM4/5NM4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5K2D/5K2D_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ME7/6ME7_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/1GZM/1GZM_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TQ4/6TQ4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TOD/6TOD_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TPG/6TPG_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FK6/6FK6_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6D27/6D27_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FFH/6FFH_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5OLH/5OLH_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6RZ9/6RZ9_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5NDZ/5NDZ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6AKX/6AKX_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS5/6PS5_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZHP/5ZHP_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TP4/6TP4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6HLL/6HLL_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5MZP/5MZP_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6IGK/6IGK_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5UNH/5UNH_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FKC/6FKC_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZKQ/5ZKQ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/3C9L/3C9L_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5NX2/5NX2_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS1/6PS1_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5OM1/5OM1_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5D6L/5D6L_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS4/6PS4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ME6/6ME6_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6AQF/6AQF_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5X93/5X93_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4XT3/4XT3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6LW5/6LW5_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6ME3/6ME3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5TZR/5TZR_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZTY/5ZTY_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6GPS/6GPS_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5UVI/5UVI_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5K2C/5K2C_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TP3/6TP3_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5WQC/5WQC_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5N2S/5N2S_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6DRZ/6DRZ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6TQ9/6TQ9_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5EE7/5EE7_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FFI/6FFI_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/2Z73/2Z73_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5TE5/5TE5_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/2YCZ/2YCZ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS0/6PS0_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4JKV/4JKV_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5T04/5T04_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5WF5/5WF5_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6LUQ/6LUQ_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6FK9/6FK9_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6PS8/6PS8_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/3C9M/3C9M_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6KP6/6KP6_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6RZ4/6RZ4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5VBL/5VBL_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZKC/5ZKC_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6MH8/6MH8_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5UNG/5UNG_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5X33/5X33_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5OM4/5OM4_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/4XES/4XES_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/6E59/6E59_refined.pdb was not found.\n",
      "Error: File /gpcr/users/daranda/doctorat/GPCR_simulations//data_structures/5ZKP/5ZKP_refined.pdb was not found.\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6GPX_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6GPX_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand F7N \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/F7N/6GPX_F7N.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/F7N/6GPX_F7N.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 97Y \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/97Y/6KK7_97Y.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/97Y/6KK7_97Y.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand SNT \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/SNT/4N4W_SNT.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/SNT/4N4W_SNT.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/4N4W_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/4N4W_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand SUV \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/SUV/6TPJ_SUV.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/SUV/6TPJ_SUV.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NH4 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NH4/6TPJ_NH4.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NH4/6TPJ_NH4.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand A90 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A90/6IIV_A90.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A90/6IIV_A90.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6IIV_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6IIV_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6JZH_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6JZH_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/6JZH_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/6JZH_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand KNW \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/KNW/6RZ6_KNW.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/KNW/6RZ6_KNW.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/5UIW_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/5UIW_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6AKY_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6AKY_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand A4X \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A4X/6AKY_A4X.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A4X/6AKY_A4X.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 6XQ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/6XQ/5KW2_6XQ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/6XQ/5KW2_6XQ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/6PS7_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/6PS7_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6PS7_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6PS7_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 7MA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7MA/5WS3_7MA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7MA/5WS3_7MA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand DNZ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DNZ/6FK8_DNZ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DNZ/6FK8_DNZ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NVH \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NVH/6TQ6_NVH.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NVH/6TQ6_NVH.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand A8X \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A8X/6IIU_A8X.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A8X/6IIU_A8X.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6IIU_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6IIU_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand K86 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/K86/5XPR_K86.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/K86/5XPR_K86.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FSY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FSY/6D26_FSY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FSY/6D26_FSY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 0HK \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/0HK/6OL9_0HK.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/0HK/6OL9_0HK.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand QNB \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/QNB/5ZK3_QNB.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/QNB/5ZK3_QNB.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NLX_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NLX_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5NLX_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5NLX_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 8TZ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8TZ/5NDD_8TZ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8TZ/5NDD_8TZ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NDD_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NDD_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand DN5 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DN5/6FKA_DN5.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DN5/6FKA_DN5.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 3C0 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/3C0/5ZK8_3C0.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/3C0/5ZK8_3C0.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand DO5 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DO5/6FK7_DO5.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DO5/6FK7_DO5.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand AWY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/AWY/6ME5_AWY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/AWY/6ME5_AWY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JTZ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JTZ/6PS2_JTZ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JTZ/6PS2_JTZ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6KNM_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6KNM_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand P2E \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/P2E/6AK3_P2E.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/P2E/6AK3_P2E.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5K2A_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5K2A_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5K2A_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5K2A_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9XW \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9XW/5OLO_9XW.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9XW/5OLO_9XW.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLO_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLO_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JEY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6ME8_JEY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6ME8_JEY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6ME8_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6ME8_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6GT3_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6GT3_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand F9Q \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/F9Q/6GT3_F9Q.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/F9Q/6GT3_F9Q.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand AQD \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/AQD/5WIU_AQD.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/AQD/5WIU_AQD.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 836 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/836/5V57_836.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/836/5V57_836.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FMN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5V57_FMN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5V57_FMN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 82F \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/82F/5ZKB_82F.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/82F/5ZKB_82F.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9Y2 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9Y2/5OLV_9Y2.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9Y2/5OLV_9Y2.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLV_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLV_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZLK \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZLK/6RZ5_ZLK.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZLK/6RZ5_ZLK.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6RZ5_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6RZ5_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JEV \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEV/6ME9_JEV.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEV/6ME9_JEV.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6ME9_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6ME9_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NT5 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NT5/6TP6_NT5.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NT5/6TP6_NT5.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CL \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CL/6TP6_CL.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CL/6TP6_CL.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JEV \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEV/6ME2_JEV.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEV/6ME2_JEV.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand EN6 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/EN6/6LI0_EN6.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/EN6/6LI0_EN6.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FMN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/6LI0_FMN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/6LI0_FMN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand GAW \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/GAW/6HLP_GAW.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/GAW/6HLP_GAW.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5K2B_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5K2B_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5K2B_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5K2B_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ML2 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ML2/6ME4_ML2.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ML2/6ME4_ML2.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand P32 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/P32/5F8U_P32.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/P32/5F8U_P32.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 1Q5 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/1Q5/4Z9G_1Q5.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/1Q5/4Z9G_1Q5.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 836 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/836/5V56_836.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/836/5V56_836.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FMN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5V56_FMN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5V56_FMN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9AF \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9AF/5ZBH_9AF.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9AF/5ZBH_9AF.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NM2_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NM2_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5NM2_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5NM2_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CY8 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CY8/4O9R_CY8.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CY8/4O9R_CY8.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9P2 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9P2/5O9H_9P2.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9P2/5O9H_9P2.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand H8G \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/H8G/6DRX_H8G.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/H8G/6DRX_H8G.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand A8T \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A8T/4QIM_A8T.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A8T/4QIM_A8T.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/4QIM_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/4QIM_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand TIM \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/TIM/6PS6_TIM.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/TIM/6PS6_TIM.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand EJ4 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/EJ4/4EJ4_EJ4.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/EJ4/4EJ4_EJ4.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FMN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/6LI1_FMN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/6LI1_FMN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5N2R_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5N2R_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 8JN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8JN/5N2R_8JN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8JN/5N2R_8JN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand H8M \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/H8M/6DS0_H8M.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/H8M/6DS0_H8M.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JTZ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JTZ/6PRZ_JTZ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JTZ/6PRZ_JTZ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand UKA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/UKA/5WF6_UKA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/UKA/5WF6_UKA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9P2 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9P2/6C1Q_9P2.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9P2/6C1Q_9P2.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 97Y \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/97Y/5VEW_97Y.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/97Y/5VEW_97Y.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 8D0 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8D0/6KPC_8D0.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8D0/6KPC_8D0.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 3C0 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/3C0/5YC8_3C0.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/3C0/5YC8_3C0.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand HG \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/HG/5YC8_HG.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/HG/5YC8_HG.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5OLG_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5OLG_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLG_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLG_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5MZJ_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5MZJ_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand TEP \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/TEP/5MZJ_TEP.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/TEP/5MZJ_TEP.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FMN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5XRA_FMN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5XRA_FMN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 8D3 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8D3/5XRA_8D3.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8D3/5XRA_8D3.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CVD \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CVD/6PS3_CVD.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CVD/6PS3_CVD.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand GLY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/GLY/4BUO_GLY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/GLY/4BUO_GLY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZOT \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZOT/6A94_ZOT.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZOT/6A94_ZOT.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6A94_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6A94_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand A6L \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A6L/6A94_A6L.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A6L/6A94_A6L.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand MK6 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/MK6/5TZY_MK6.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/MK6/5TZY_MK6.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 7OS \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7OS/5TZY_7OS.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7OS/5TZY_7OS.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand D2U \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/D2U/6K1Q_D2U.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/D2U/6K1Q_D2U.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand KNZ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/KNZ/6RZ8_KNZ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/KNZ/6RZ8_KNZ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5JTB_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5JTB_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand IOD \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/IOD/5JTB_IOD.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/IOD/5JTB_IOD.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CY8 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CY8/6D32_CY8.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CY8/6D32_CY8.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 8VS \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8VS/5X7D_8VS.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8VS/5X7D_8VS.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CAU \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CAU/5X7D_CAU.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CAU/5X7D_CAU.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NRK \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NRK/6TOT_NRK.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NRK/6TOT_NRK.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9GF \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9GF/6KQI_9GF.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9GF/6KQI_9GF.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9GL \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9GL/6KQI_9GL.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9GL/6KQI_9GL.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand QGW \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/QGW/6ZDV_QGW.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/QGW/6ZDV_QGW.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6ZDV_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6ZDV_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NU8 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NU8/6TPN_NU8.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NU8/6TPN_NU8.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand AQD \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/AQD/5WIV_AQD.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/AQD/5WIV_AQD.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5WIV_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5WIV_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand AZJ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/AZJ/4NTJ_AZJ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/AZJ/4NTJ_AZJ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand EFD \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/EFD/6C1R_EFD.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/EFD/6C1R_EFD.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6C1R_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6C1R_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6LI2_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6LI2_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ACY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ACY/6FJ3_ACY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ACY/6FJ3_ACY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CL \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CL/6FJ3_CL.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CL/6FJ3_CL.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ERM \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ERM/4NC3_ERM.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ERM/4NC3_ERM.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand TRS \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/TRS/4NC3_TRS.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/TRS/4NC3_TRS.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 7LD \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7LD/5TVN_7LD.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7LD/5TVN_7LD.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand KNW \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/KNW/6RZ7_KNW.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/KNW/6RZ7_KNW.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand DLH \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DLH/6FKB_DLH.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DLH/6FKB_DLH.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 97Y \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/97Y/6KJV_97Y.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/97Y/6KJV_97Y.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FMN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5XR8_FMN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5XR8_FMN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 8D0 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8D0/5XR8_8D0.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8D0/5XR8_8D0.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand J9P \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/J9P/6M9T_J9P.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/J9P/6M9T_J9P.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand VT5 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/VT5/5T1A_VT5.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/VT5/5T1A_VT5.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/5T1A_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/5T1A_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CAU \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CAU/4GBR_CAU.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CAU/4GBR_CAU.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5VRA_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5VRA_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5VRA_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5VRA_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9AO \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9AO/5ZBQ_9AO.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9AO/5ZBQ_9AO.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 8D1 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8D1/5UIG_8D1.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8D1/5UIG_8D1.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NRE \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NRE/6TOS_NRE.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NRE/6TOS_NRE.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6TOS_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6TOS_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand QGE \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/QGE/6ZDR_QGE.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/QGE/6ZDR_QGE.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6ZDR_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6ZDR_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 97Y \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/97Y/6KK1_97Y.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/97Y/6KK1_97Y.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand SUV \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/SUV/6TO7_SUV.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/SUV/6TO7_SUV.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6TO7_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6TO7_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NVK \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NVK/6TQ7_NVK.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NVK/6TQ7_NVK.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JTZ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JTZ/6OBA_JTZ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JTZ/6OBA_JTZ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand M3J \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/M3J/6OBA_M3J.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/M3J/6OBA_M3J.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6A93_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6A93_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand H8D \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/H8D/6DRY_H8D.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/H8D/6DRY_H8D.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand T4E \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/T4E/5OLZ_T4E.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/T4E/5OLZ_T4E.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLZ_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLZ_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NM4_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NM4_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5NM4_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5NM4_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5K2D_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5K2D_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5K2D_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5K2D_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JEY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6ME7_JEY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6ME7_JEY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6ME7_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6ME7_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand LDA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/LDA/1GZM_LDA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/LDA/1GZM_LDA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand C8E \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/C8E/1GZM_C8E.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/C8E/1GZM_C8E.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/1GZM_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/1GZM_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NV8 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NV8/6TQ4_NV8.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NV8/6TQ4_NV8.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6TQ4_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6TQ4_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 7MA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7MA/6TOD_7MA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7MA/6TOD_7MA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6TOD_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6TOD_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 7MA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7MA/6TPG_7MA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7MA/6TPG_7MA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand DOK \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DOK/6FK6_DOK.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DOK/6FK6_DOK.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FT4 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FT4/6D27_FT4.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FT4/6D27_FT4.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand D7W \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/D7W/6FFH_D7W.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/D7W/6FFH_D7W.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9XT \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9XT/5OLH_9XT.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9XT/5OLH_9XT.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLH_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OLH_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand KO5 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/KO5/6RZ9_KO5.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/KO5/6RZ9_KO5.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NDZ_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5NDZ_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6AKX_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6AKX_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand A4R \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A4R/6AKX_A4R.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/A4R/6AKX_A4R.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand SNP \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/SNP/6PS5_SNP.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/SNP/6PS5_SNP.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9EC \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9EC/5ZHP_9EC.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9EC/5ZHP_9EC.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NRZ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NRZ/6TP4_NRZ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NRZ/6TP4_NRZ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5MZP_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5MZP_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CFF \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CFF/5MZP_CFF.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CFF/5MZP_CFF.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand DLB \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DLB/6FKC_DLB.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DLB/6FKC_DLB.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/5ZKQ_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/5ZKQ_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/3C9L_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/3C9L_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand LDA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/LDA/3C9L_LDA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/LDA/3C9L_LDA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand C8E \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/C8E/3C9L_C8E.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/C8E/3C9L_C8E.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand TIM \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/TIM/6PS1_TIM.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/TIM/6PS1_TIM.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand T4E \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/T4E/5OM1_T4E.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/T4E/5OM1_T4E.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OM1_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OM1_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CL \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CL/5OM1_CL.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CL/5OM1_CL.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CAU \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CAU/5D6L_CAU.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CAU/5D6L_CAU.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JRZ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JRZ/6PS4_JRZ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JRZ/6PS4_JRZ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JEY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6ME6_JEY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6ME6_JEY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6ME6_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6ME6_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/6AQF_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/6AQF_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand K87 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/K87/5X93_K87.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/K87/5X93_K87.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JEY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6ME3_JEY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6ME3_JEY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5TZR_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5TZR_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand MK6 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/MK6/5TZR_MK6.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/MK6/5TZR_MK6.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9JU \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9JU/5ZTY_9JU.mol2 does not exist. Skipping....\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-14 16:20:27,405 - moleculekit.readers - INFO - Attempting PDB query for 6TPJ\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9JU/5ZTY_9JU.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6GPS_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/6GPS_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand F7N \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/F7N/6GPS_F7N.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/F7N/6GPS_F7N.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5UVI_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5UVI_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5K2C_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/5K2C_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5K2C_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5K2C_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 7MA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7MA/5WQC_7MA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7MA/5WQC_7MA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 8K8 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8K8/5N2S_8K8.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/8K8/5N2S_8K8.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand H8J \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/H8J/6DRZ_H8J.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/H8J/6DRZ_H8J.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NVN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NVN/6TQ9_NVN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NVN/6TQ9_NVN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 5MV \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/5MV/5EE7_5MV.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/5MV/5EE7_5MV.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand D8B \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/D8B/6FFI_D8B.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/D8B/6FFI_D8B.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand PC1 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/PC1/2Z73_PC1.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/PC1/2Z73_PC1.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand I32 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/I32/2YCZ_I32.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/I32/2YCZ_I32.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand CAU \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CAU/6PS0_CAU.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/CAU/6PS0_CAU.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 1KS \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/1KS/4JKV_1KS.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/1KS/4JKV_1KS.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand GMJ \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/GMJ/6LUQ_GMJ.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/GMJ/6LUQ_GMJ.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand DNK \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DNK/6FK9_DNK.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/DNK/6FK9_DNK.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand JEY \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6PS8_JEY.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/JEY/6PS8_JEY.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6RZ4_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/6RZ4_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/5VBL_ZN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZN/5VBL_ZN.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 3C0 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/3C0/5ZKC_3C0.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/3C0/5ZKC_3C0.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand ZMA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/6MH8_ZMA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/ZMA/6MH8_ZMA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 7Y9 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7Y9/5X33_7Y9.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/7Y9/5X33_7Y9.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand T4E \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/T4E/5OM4_T4E.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/T4E/5OM4_T4E.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand NA \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OM4_NA.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/NA/5OM4_NA.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand L76 \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/L76/6E59_L76.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/L76/6E59_L76.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand 9ER \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9ER/5ZKP_9ER.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/9ER/5ZKP_9ER.mol2 does not exist. Skipping....\n",
      "Getting toppar file for ligand FMN \n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5ZKP_FMN.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/Ligands/FMN/5ZKP_FMN.mol2 does not exist. Skipping....\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-14 16:20:30,737 - moleculekit.molecule - INFO - Removed 8607 atoms. 37 atoms remaining in the molecule.\n",
      "2022-02-14 16:20:30,795 - moleculekit.molecule - INFO - Removed 0 atoms. 37 atoms remaining in the molecule.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/6TPJ_YCM.mol2 does not exist. Skipping....\n",
      "file /gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/6TPJ_YCM.mol2 does not exist. Skipping....\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/6TPJ_legacy_toppar.str' -> '/gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/pre_6TPJ_legacy_toppar.str'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-3-b93d33c0e326>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     13\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0;31m# Get topology-parameter files for modified residues\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m \u001b[0mget_modres_toppar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmodresdict\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbasepath\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0musername\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpassword\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/gpcr/users/daranda/doctorat/GPCR_simulations/simulate_structures_functions.py\u001b[0m in \u001b[0;36mget_modres_toppar\u001b[0;34m(modresdict, basepath, username, password, pdbfiles)\u001b[0m\n\u001b[1;32m    755\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mmodres\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mmodresdict\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpdbcode\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    756\u001b[0m             \u001b[0mmodrespath\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbasepath\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'toppar/mod_residues/'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mmodres\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'/'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 757\u001b[0;31m             \u001b[0mmodres_covlig_toppar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpdbcode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmodres\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmodrespath\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpdbfiles\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    758\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    759\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/gpcr/users/daranda/doctorat/GPCR_simulations/simulate_structures_functions.py\u001b[0m in \u001b[0;36mmodres_covlig_toppar\u001b[0;34m(pdbcode, modres, modrespath, covlig, pdbfiles)\u001b[0m\n\u001b[1;32m    893\u001b[0m         \u001b[0;31m#Rename output files from paramchem\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    894\u001b[0m         \u001b[0mpre_file\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmodrespath\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'pre_'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mtopparname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 895\u001b[0;31m         \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmodrespath\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mtopparname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpre_file\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    896\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    897\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/6TPJ_legacy_toppar.str' -> '/gpcr/users/daranda/doctorat/GPCR_simulations/toppar/mod_residues/YCM/pre_6TPJ_legacy_toppar.str'"
     ]
    }
   ],
   "source": [
    "################################################\n",
    "# Part 1: Download data and prepare dictionaries\n",
    "################################################\n",
    "\n",
    "# Download and store refined structures from GPCRdb\n",
    "download_GPCRdb_structures(pdb_set, strucpath)\n",
    "\n",
    "#Create or moidfy the ligands dictionary\n",
    "(ligandsdict, ligandsset, modresdict) = ligand_dictionary(pdb_set, ligandsdict_path, modres_path, basepath, blacklist)\n",
    "\n",
    "# Get topology-parameter files for ligands\n",
    "get_lig_toppar(ligandsdict, basepath, username, password)\n",
    "\n",
    "# Get topology-parameter files for modified residues\n",
    "get_modres_toppar(modresdict, basepath, username, password)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:38:20,763 - moleculekit.molecule - INFO - Removed 0 atoms. 4888 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:20,798 - moleculekit.molecule - INFO - Removed 0 atoms. 4888 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:23,069 - moleculekit.molecule - WARNING - Alternative atom locations detected. Only altloc A was kept. If you prefer to keep all use the keepaltloc=\"all\" option when reading the file.\n",
      "2021-12-30 09:38:23,070 - moleculekit.molecule - INFO - Removed 15 atoms. 3684 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:23,323 - moleculekit.molecule - INFO - Removed 0 atoms. 4888 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:23,354 - moleculekit.molecule - INFO - Removed 2 atoms. 4886 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:23,450 - moleculekit.molecule - INFO - Removed 0 atoms. 4886 atoms remaining in the molecule.\n",
      "2021-12-30 09:38:26,818 - moleculekit.molecule - INFO - Removed 0 atoms. 4886 atoms remaining in the molecule.\n",
      "/home/david/miniconda3/lib/python3.6/site-packages/Bio/SubsMat/__init__.py:131: BiopythonDeprecationWarning: Bio.SubsMat has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.substitution_matrices as a replacement, and contact the Biopython developers if you still need the Bio.SubsMat module.\n",
      "  BiopythonDeprecationWarning,\n",
      "2021-12-30 09:38:29,716 - moleculekit.tools.sequencestructuralalignment - INFO - No segment was specified by the user for `mol`. Alignment will be done on all protein segments.\n",
      "2021-12-30 09:38:29,756 - moleculekit.tools.sequencestructuralalignment - INFO - No segment was specified by the user for `ref`. Alignment will be done on all protein segments.\n",
      "2021-12-30 09:38:29,933 - moleculekit.tools.sequencestructuralalignment - WARNING - 12 alignments found. Limiting to 1 as specified in the `maxalignments` argument.\n",
      "2021-12-30 09:38:29,935 - moleculekit.tools.sequencestructuralalignment - INFO - Alignment #0 was done on 69 residues: mol segid P resid 154-222\n",
      "/home/david/miniconda3/lib/python3.6/site-packages/moleculekit/align.py:16: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, A), array(float32, 2d, A))\n",
      "  covariance = np.dot(P.T, Q)\n",
      "/home/david/miniconda3/lib/python3.6/site-packages/moleculekit/align.py:54: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, C), array(float32, 2d, A))\n",
      "  all1 = np.dot(all1, rot.T)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adding membrane...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:38:33,738 - htmd.builder.builder - INFO - Replicating Membrane 3x3\n",
      "Replicating Membrane: 100%|██████████| 9/9 [00:02<00:00,  4.08it/s]\n",
      "2021-12-30 09:39:57,781 - moleculekit.molecule - INFO - Removed 2066 atoms. 42862 atoms remaining in the molecule.\n",
      "2021-12-30 09:39:58,915 - moleculekit.molecule - INFO - Removed 139 residues from appended Molecule due to collisions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Solvating...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:39:59,620 - htmd.builder.solvate - INFO - Using water pdb file at: /home/david/miniconda3/lib/python3.6/site-packages/htmd/share/solvate/wat.pdb\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wataerbox Max and min:  [51.999847 52.89102  28.060892] [-47.308155 -46.37798  -27.734108]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:40:00,394 - htmd.builder.solvate - INFO - Replicating 8 water segments, 2 by 2 by 2\n",
      "Solvating: 100%|██████████| 8/8 [00:08<00:00,  1.05s/it]\n",
      "2021-12-30 09:40:10,433 - htmd.builder.solvate - INFO - 14165 water molecules were added to the system.\n",
      "2021-12-30 09:40:16,562 - moleculekit.molecule - INFO - Removed 300 atoms. 84548 atoms remaining in the molecule.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pre-build...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:40:22,384 - htmd.builder.charmm - INFO - Writing out segments.\n",
      "2021-12-30 09:40:28,612 - htmd.builder.builder - INFO - One disulfide bond was added\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Disulfide Bond between: UniqueResidueID<resname: 'CYX', chain: 'P', resid: 174, insertion: '', segid: 'P'>\n",
      "                   and: UniqueResidueID<resname: 'CYX', chain: 'P', resid: 179, insertion: '', segid: 'P'>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:40:30,554 - htmd.builder.charmm - INFO - Starting the build.\n",
      "2021-12-30 09:40:37,826 - htmd.builder.charmm - WARNING - Failed to set coordinates for 2 atoms.\n",
      "2021-12-30 09:40:37,828 - htmd.builder.charmm - WARNING - Failed to guess coordinates for 1 atoms due to bad angles.\n",
      "2021-12-30 09:40:37,828 - htmd.builder.charmm - WARNING - Poorly guessed coordinates for 225 atoms.\n",
      "2021-12-30 09:40:37,828 - htmd.builder.charmm - WARNING - 2 undefined warnings were produced during building.\n",
      "2021-12-30 09:40:37,828 - htmd.builder.charmm - WARNING - Please check /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/pre-build/5ZTY/log.txt for further information.\n",
      "2021-12-30 09:40:37,829 - htmd.builder.charmm - INFO - Finished building.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking aromatic insertions...\n",
      "WARNING: removed 0 atoms within 1.5 of a protein aromatic ring\n",
      "Renumbering...\n",
      "Ionizing...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:42:11,896 - htmd.builder.charmm - INFO - Writing out segments.\n",
      "2021-12-30 09:42:18,160 - htmd.builder.builder - INFO - One disulfide bond was added\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Disulfide Bond between: UniqueResidueID<resname: 'CYS', chain: 'P', resid: 174, insertion: '', segid: 'P'>\n",
      "                   and: UniqueResidueID<resname: 'CYS', chain: 'P', resid: 179, insertion: '', segid: 'P'>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:42:19,361 - htmd.builder.charmm - INFO - Starting the build.\n",
      "2021-12-30 09:42:26,693 - htmd.builder.charmm - WARNING - Failed to set coordinates for 1 atoms.\n",
      "2021-12-30 09:42:26,696 - htmd.builder.charmm - WARNING - Poorly guessed coordinates for 1 atoms.\n",
      "2021-12-30 09:42:26,697 - htmd.builder.charmm - WARNING - 2 undefined warnings were produced during building.\n",
      "2021-12-30 09:42:26,699 - htmd.builder.charmm - WARNING - Please check /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/ionize/5ZTY/log.txt for further information.\n",
      "2021-12-30 09:42:26,702 - htmd.builder.charmm - INFO - Finished building.\n",
      "2021-12-30 09:42:32,556 - htmd.builder.ionize - INFO - Adding 17 anions + 0 cations for neutralizing and 102 ions for the given salt concentration.\n",
      "2021-12-30 09:43:07,077 - htmd.builder.charmm - INFO - Writing out segments.\n",
      "2021-12-30 09:43:14,896 - htmd.builder.charmm - INFO - Starting the build.\n",
      "2021-12-30 09:43:21,976 - htmd.builder.charmm - WARNING - Failed to set coordinates for 1 atoms.\n",
      "2021-12-30 09:43:21,979 - htmd.builder.charmm - WARNING - Poorly guessed coordinates for 1 atoms.\n",
      "2021-12-30 09:43:21,982 - htmd.builder.charmm - WARNING - 2 undefined warnings were produced during building.\n",
      "2021-12-30 09:43:21,984 - htmd.builder.charmm - WARNING - Please check /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/ionize/5ZTY/log.txt for further information.\n",
      "2021-12-30 09:43:21,986 - htmd.builder.charmm - INFO - Finished building.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:44:13,843 - htmd.builder.charmm - INFO - Writing out segments.\n",
      "2021-12-30 09:44:20,582 - htmd.builder.builder - INFO - One disulfide bond was added\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Disulfide Bond between: UniqueResidueID<resname: 'CYS', chain: 'P', resid: 174, insertion: '', segid: 'P'>\n",
      "                   and: UniqueResidueID<resname: 'CYS', chain: 'P', resid: 179, insertion: '', segid: 'P'>\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2021-12-30 09:44:21,674 - htmd.builder.charmm - INFO - Starting the build.\n",
      "2021-12-30 09:44:31,029 - htmd.builder.charmm - WARNING - Failed to set coordinates for 1 atoms.\n",
      "2021-12-30 09:44:31,030 - htmd.builder.charmm - WARNING - Poorly guessed coordinates for 1 atoms.\n",
      "2021-12-30 09:44:31,031 - htmd.builder.charmm - WARNING - 2 undefined warnings were produced during building.\n",
      "2021-12-30 09:44:31,032 - htmd.builder.charmm - WARNING - Please check /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/5ZTY/log.txt for further information.\n",
      "2021-12-30 09:44:31,033 - htmd.builder.charmm - INFO - Finished building.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "End of 5ZTY after 394.39238953590393 seconds\n",
      "\n",
      "Build model for 5ZTY_apo already exists. Skipping...\n"
     ]
    }
   ],
   "source": [
    "###########################\n",
    "## Part 2: Build the models \n",
    "###########################\n",
    "# Iterate by GPCRdb structures to simulate\n",
    "pdbs_number = len(pdb_set)\n",
    "i = 0\n",
    "for pdbcode in {\"5ZTY\"}:\n",
    "# for pdbcode in bad_set:\n",
    "    try:\n",
    "        for apo in [False, True]:\n",
    "            \n",
    "            #Starting simulation\n",
    "            start_time = time.time()        \n",
    "            i += 1\n",
    "            sysname = pdbcode+'_apo' if apo else pdbcode\n",
    "            mystrucpath = strucpath+pdbcode+'/'\n",
    "\n",
    "            # Skip if there is already a model build for this\n",
    "            if os.path.exists(resultspath+'build/'+sysname+'/structure.pdb'):\n",
    "                print('Build model for '+sysname+' already exists. Skipping...')\n",
    "                continue\n",
    "\n",
    "            # Check if simulation is aminergic\n",
    "            aminergic = gpcrdb_dict[pdbcode]['family'].startswith('001_001')\n",
    "            adenosine = gpcrdb_dict[pdbcode]['family'].startswith('001_006_001')\n",
    "\n",
    "            ## Load curated structures\n",
    "            if curated:\n",
    "                apo_name = \"apo\" if apo else \"complex\"\n",
    "                curated_filename = str(\"%sreceptor2curate_output/%s/%s_%s_curated.pdb\"%(basepath, pdbcode, pdbcode, apo_name))\n",
    "                gpcrdb_mol = Molecule(curated_filename)\n",
    "                # Remove ligand hidrogens in case curators put any\n",
    "                gpcrdb_mol.remove(\"not (protein or water or lipid) and element H\")\n",
    "            ## Load non-curated structure\n",
    "            else:\n",
    "                (sod2x50, watered_filename) = internal_waters(mystrucpath, pdbcode, gpcrdb_dict, apo)\n",
    "                gpcrdb_mol = Molecule(watered_filename)\n",
    "            \n",
    "            # Add peptide ligand to structure if necessary\n",
    "            if any([ True for lig in gpcrdb_dict[pdbcode]['ligands'] if lig['type'] in {'protein', 'peptide'} ]) and not (apo):\n",
    "                add_peplig(watered_filename, pdbcode)\n",
    "            \n",
    "            # If the pipeline is running in 'apoform mode', remove any non-protein, non-ion, non-water thing on the system      \n",
    "            # Delete also sod2x50 (we don't want them here)\n",
    "            if apo:\n",
    "                gpcrdb_mol.remove('not (protein or water or ion) or element Na')\n",
    "                \n",
    "            # Remove unnecessary ligand molecules: mostly crystalization detergents, quelants, buffers,\n",
    "            # or post-traductional glicosilations\n",
    "            gpcrdb_mol.remove('resname '+' '.join(blacklist))\n",
    "            \n",
    "            # Remove 2x50Sodium from non-A-class GPCRs\n",
    "            if not gpcrdb_dict[pdbcode]['family'].startswith('001'):\n",
    "                gpcrdb_mol.remove('element NA')\n",
    "\n",
    "            # Get aligned OPM structure\n",
    "            thickness,opm_mol = get_opm(pdbcode)\n",
    "\n",
    "            # Ismael's function to add labels (segid) for 'ligand' and 'protein' parts of the system\n",
    "            gpcrdb_mol_fixed,prot_segids = fix_and_prepare_input(gpcrdb_mol,pdbcode,modresdict,new_pdb_chain)\n",
    "            \n",
    "            # If there's any, parameterize and rename covalent-bound ligands\n",
    "            if not apo:\n",
    "                (gpcrdb_mol_fixed, covligs) = covalent_ligands(gpcrdb_mol_fixed, pdbcode, ligandsdict)\n",
    "            else:\n",
    "                covligs = []\n",
    "            \n",
    "            # write file to remember to which chain in the original structure belongs each segment\n",
    "            segchain_json(gpcrdb_mol_fixed, pdbcode, basepath, prot_segids)\n",
    "            \n",
    "            # Align structrues using sequences, and take first one\n",
    "            alignment_results = sequenceStructureAlignment(gpcrdb_mol_fixed, opm_mol, maxalignments = 1)\n",
    "            mol_aligned = alignment_results[0] \n",
    "            \n",
    "            #Center to receptor XY\n",
    "            center = np.mean(mol_aligned.get('coords',sel='chain P'),axis=0)\n",
    "            mol_aligned.moveBy([-center[0],-center[1],0])\n",
    "            \n",
    "            # Prepare protein: asign titration states, flipping side chains of HIS, ASN and GLN; rotate some sidechains, optimize waters, etc.\n",
    "            # Most of this is done with a HTMD function called proteinPrepare()\n",
    "            # Skip step if we are working with curators structures\n",
    "            prepared_mol = mol_aligned if curated else prepare_system(mol_aligned, pdbcode, thickness, sod2x50, aminergic, adenosine)\n",
    "            \n",
    "            #Add membrane\n",
    "            print('Adding membrane...')\n",
    "            membranemol = Molecule(membranepdb)\n",
    "            mol_membraned, membrane_resnames, membrane_segids, xreps, yreps = add_membrane(prepared_mol, membranemol,prot_segids,membrane_distance)\n",
    "            \n",
    "            # Needed later for equilibration\n",
    "            with open(mystrucpath+\"const_sel.txt\",'w') as out: \n",
    "                const_sel = 'segid '+' '.join(prot_segids)+' and name C CA N O or not (segid ' + \\\n",
    "                  ' '.join(prot_segids)+' or lipid or water or ions ) and noh or segid ION WAT and noh'\n",
    "                out.write(const_sel)\n",
    "\n",
    "            #Solvate\n",
    "            print('Solvating...')\n",
    "            mol_solvated = solvate_pdbmol(mol_membraned,membrane_segids,water_thickness,water_margin,buffer=buffer,coldist=coldist,prefix='WT')\n",
    "\n",
    "            # Check if system has lone-pair hallogen atoms. If it does, use legacy CGenFF parameters\n",
    "            has_halo = bool(len(gpcrdb_mol_fixed.get('name', 'element Br Cl I and not ion')))\n",
    "            if has_halo:\n",
    "                cgenff_par = [topparpath+'David_top_params/parameters/legacy_par_all36_cgenff.prm']\n",
    "                cgenff_top = [topparpath+'David_top_params/topologies/legacy_top_all36_cgenff.rtf']\n",
    "            else: \n",
    "                cgenff_par = [topparpath+'General_top_params/parameters/par_all36_cgenff.prm']\n",
    "                cgenff_top = [topparpath+'General_top_params/topologies/top_all36_cgenff.rtf']\n",
    "            \n",
    "            #Obtain extra parameters for ligands and modified residues \n",
    "            ligstreams=extra_parameters(pdbcode, ligandsdict, modresdict, blacklist, covligs, basepath, has_halo)\n",
    "            \n",
    "            # Assignign terminology for cap atoms of protein chain, depending if it is the receptor protein or not\n",
    "            caps = get_caps(prot_segids, mol_solvated)\n",
    "            #{'P0': ['first ACE', 'last CT3'], 'P1': ['first ACE', 'last CT3']}\n",
    "\n",
    "            #Pre-build model\n",
    "            print('Pre-build...')\n",
    "            prebuildmol = charmm.build(mol_solvated, \n",
    "                                       topo=topos+cgenff_top, \n",
    "                                       param=params+cgenff_par,\n",
    "                                       stream=streams+ligstreams,\n",
    "                                       caps=caps,\n",
    "                                       outdir=resultspath+'/pre-build/'+sysname,\n",
    "                                       ionize=False)\n",
    "\n",
    "            # Save prebuild model topologies in files, and  store prebuild model in molecule object\n",
    "            prebuild_psffile = prebuildmol.topoloc\n",
    "            prebuild_pdbfile = os.path.splitext(prebuildmol.topoloc)[0]+'.pdb'\n",
    "            prebuildmol = Molecule(prebuild_pdbfile)\n",
    "            _recoverProtonations(prebuildmol)\n",
    "\n",
    "            # Checking of aromatic insertions (takes quite a lot fo time)\n",
    "            print('Checking aromatic insertions...')\n",
    "            mol_removed,removed_indexes = remove_aromatic_insertions(prebuildmol,prot_segids, outpdb=resultspath+'/pre-build/'+sysname+'/aromatic_check.pdb')\n",
    "\n",
    "            # Checking of water/lipid ratio\n",
    "            lipid_num = len(set(prebuildmol.get('resid',sel='segid '+membrane_lipid_segid)))\n",
    "            solv_num = len(mol_removed.get('index',sel='resname TIP3 and name OH2'))\n",
    "            if float(solv_num) / lipid_num < 35:\n",
    "                raise ValueError('Water/lipid ratio lower than 35.')\n",
    "\n",
    "            #Renumber residues\n",
    "            print('Renumbering...')\n",
    "            mol_renumbered = renumber_resid_vmd(mol_removed,'segid '+' '.join(membrane_segids),by=2)\n",
    "\n",
    "            # Ionizing system\n",
    "            print('Ionizing...')\n",
    "            molbuilt = charmm.build(mol_removed,\n",
    "                                    topo=topos+cgenff_top, \n",
    "                                    param=params+cgenff_par,\n",
    "                                    stream=streams+ligstreams,                        \n",
    "                                    outdir=resultspath+'/ionize/'+sysname,\n",
    "                                    saltconc=0.15,\n",
    "                                    caps=caps)\n",
    "            build_psffile = molbuilt.topoloc\n",
    "            build_pdbfile = os.path.splitext(molbuilt.topoloc)[0]+'.pdb'\n",
    "            molbuilt = Molecule(build_pdbfile)\n",
    "            _recoverProtonations(molbuilt)\n",
    "\n",
    "            # Remove clashing rebuild waters\n",
    "            molbuilt.remove(\"same residue as (water and within 1 of protein)\")\n",
    "            \n",
    "            #Building system\n",
    "            print('Building...')\n",
    "            molbuilt = renumber_resid_vmd(molbuilt,'segid \"WT.*\" or segid I',by=2)\n",
    "            molbuilt = charmm.build(molbuilt, \n",
    "                                    topo=topos+cgenff_top, \n",
    "                                    param=params+cgenff_par,\n",
    "                                    stream=streams+ligstreams,                        \n",
    "                                    outdir=resultspath+'/build/'+sysname,\n",
    "                                    caps=caps,ionize=False)\n",
    "\n",
    "            print('End of %s after %s seconds\\n' % (sysname, time.time() - start_time))\n",
    "\n",
    "    except Exception as e:\n",
    "        print(\"model \"+pdbcode+\" could not be build because \",e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "model 6LI1 could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/6LI1. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 6LI1 already has been equilibrated\n",
      " structure 6ME6 already has been equilibrated\n",
      " structure 6ME6 already has been equilibrated\n",
      " structure 5OM1 already has been equilibrated\n",
      " structure 5OM1 already has been equilibrated\n",
      " structure 6LUQ already has been equilibrated\n",
      " structure 6LUQ already has been equilibrated\n",
      " structure 3C9M already has been equilibrated\n",
      " structure 3C9M already has been equilibrated\n",
      " structure 6D35 already has been equilibrated\n",
      " structure 6D35 already has been equilibrated\n",
      " structure 6HLO already has been equilibrated\n",
      " structure 6HLO already has been equilibrated\n",
      " structure 6ME2 already has been equilibrated\n",
      " structure 6ME2 already has been equilibrated\n",
      " structure 5V54 already has been equilibrated\n",
      " structure 5V54 already has been equilibrated\n",
      " structure 3C9L already has been equilibrated\n",
      " structure 3C9L already has been equilibrated\n",
      " structure 6TPJ already has been equilibrated\n",
      " structure 6TPJ already has been equilibrated\n",
      " structure 6TO7 already has been equilibrated\n",
      " structure 6TO7 already has been equilibrated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-18 15:57:09,453 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 15:57:09,454 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 15:58:21,365 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/6TP3/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " structure 6TP3 already has been equilibrated\n",
      " structure 6D26 already has been equilibrated\n",
      " structure 6D26 already has been equilibrated\n",
      " structure 5TZR already has been equilibrated\n",
      " structure 5TZR already has been equilibrated\n",
      " structure 6LW5 already has been equilibrated\n",
      " structure 6LW5 already has been equilibrated\n",
      " structure 6TP6 already has been equilibrated\n",
      " structure 6TP6 already has been equilibrated\n",
      " structure 5DYS already has been equilibrated\n",
      " structure 5DYS already has been equilibrated\n",
      " structure 4BUO already has been equilibrated\n",
      " structure 4BUO already has been equilibrated\n",
      " structure 5OLG already has been equilibrated\n",
      " structure 5OLG already has been equilibrated\n",
      " structure 4XT3 already has been equilibrated\n",
      " structure 4XT3 already has been equilibrated\n",
      " structure 6MH8 already has been equilibrated\n",
      " structure 6MH8 already has been equilibrated\n",
      " structure 4NC3 already has been equilibrated\n",
      " structure 4NC3 already has been equilibrated\n",
      "model 4GPO could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/4GPO. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 4GPO already has been equilibrated\n",
      " structure 5D6L already has been equilibrated\n",
      " structure 5D6L already has been equilibrated\n",
      " structure 5NM2 already has been equilibrated\n",
      " structure 5NM2 already has been equilibrated\n",
      " structure 5ZK3 already has been equilibrated\n",
      " structure 5ZK3 already has been equilibrated\n",
      " structure 4NTJ already has been equilibrated\n",
      " structure 4NTJ already has been equilibrated\n",
      " structure 6FJ3 already has been equilibrated\n",
      " structure 6FJ3 already has been equilibrated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-18 15:58:52,430 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 15:58:52,431 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 16:00:08,888 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/5XR8/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " structure 5XR8 already has been equilibrated\n",
      " structure 6RZ4 already has been equilibrated\n",
      " structure 6RZ4 already has been equilibrated\n",
      " structure 6ME8 already has been equilibrated\n",
      " structure 6ME8 already has been equilibrated\n",
      " structure 6PS6 already has been equilibrated\n",
      " structure 6PS6 already has been equilibrated\n",
      " structure 5UNG already has been equilibrated\n",
      " structure 5UNG already has been equilibrated\n",
      " structure 5V56 already has been equilibrated\n",
      " structure 5V56 already has been equilibrated\n",
      " structure 5F8U already has been equilibrated\n",
      " structure 5F8U already has been equilibrated\n",
      " structure 5UIG already has been equilibrated\n",
      " structure 5UIG already has been equilibrated\n",
      " structure 5OLZ already has been equilibrated\n",
      " structure 5OLZ already has been equilibrated\n",
      " structure 6E59 already has been equilibrated\n",
      " structure 6E59 already has been equilibrated\n",
      " structure 5MZJ already has been equilibrated\n",
      " structure 5MZJ already has been equilibrated\n",
      " structure 5NDZ already has been equilibrated\n",
      " structure 5NDZ already has been equilibrated\n",
      " structure 6FFI already has been equilibrated\n",
      " structure 6FFI already has been equilibrated\n",
      " structure 5YC8 already has been equilibrated\n",
      " structure 5YC8 already has been equilibrated\n",
      " structure 6J20 already has been equilibrated\n",
      " structure 6J20 already has been equilibrated\n",
      " structure 6FKB already has been equilibrated\n",
      " structure 6FKB already has been equilibrated\n",
      " structure 5EE7 already has been equilibrated\n",
      " structure 5EE7 already has been equilibrated\n",
      " structure 6PS1 already has been equilibrated\n",
      " structure 6PS1 already has been equilibrated\n",
      " structure 2Z73 already has been equilibrated\n",
      " structure 2Z73 already has been equilibrated\n",
      " structure 4N4W already has been equilibrated\n",
      " structure 4N4W already has been equilibrated\n",
      " structure 5ZKQ already has been equilibrated\n",
      " structure 5ZKQ already has been equilibrated\n",
      " structure 6TQ4 already has been equilibrated\n",
      " structure 6TQ4 already has been equilibrated\n",
      " structure 5VRA already has been equilibrated\n",
      " structure 5VRA already has been equilibrated\n",
      " structure 6FKA already has been equilibrated\n",
      " structure 6FKA already has been equilibrated\n",
      " structure 5NX2 already has been equilibrated\n",
      " structure 5NX2 already has been equilibrated\n",
      " structure 1GZM already has been equilibrated\n",
      " structure 1GZM already has been equilibrated\n",
      " structure 5WS3 already has been equilibrated\n",
      " structure 5WS3 already has been equilibrated\n",
      " structure 5ZBH already has been equilibrated\n",
      " structure 5ZBH already has been equilibrated\n",
      " structure 6KPC already has been equilibrated\n",
      " structure 6KPC already has been equilibrated\n",
      " structure 5YQZ already has been equilibrated\n",
      " structure 5YQZ already has been equilibrated\n",
      " structure 6C1R already has been equilibrated\n",
      " structure 6C1R already has been equilibrated\n",
      " structure 6FKC already has been equilibrated\n",
      " structure 6FKC already has been equilibrated\n",
      " structure 2YCZ already has been equilibrated\n",
      " structure 2YCZ already has been equilibrated\n",
      " structure 6OBA already has been equilibrated\n",
      " structure 6OBA already has been equilibrated\n",
      " structure 5OM4 already has been equilibrated\n",
      " structure 5OM4 already has been equilibrated\n",
      " structure 5UVI already has been equilibrated\n",
      " structure 5UVI already has been equilibrated\n",
      " structure 6PS0 already has been equilibrated\n",
      " structure 6PS0 already has been equilibrated\n",
      " structure 6GPX already has been equilibrated\n",
      " structure 6GPX already has been equilibrated\n",
      " structure 5XRA already has been equilibrated\n",
      " structure 5XRA already has been equilibrated\n",
      " structure 5TVN already has been equilibrated\n",
      " structure 5TVN already has been equilibrated\n",
      " structure 6RZ7 already has been equilibrated\n",
      " structure 6RZ7 already has been equilibrated\n",
      " structure 4XES already has been equilibrated\n",
      " structure 4XES already has been equilibrated\n",
      " structure 6LRY already has been equilibrated\n",
      " structure 6LRY already has been equilibrated\n",
      " structure 6DRY already has been equilibrated\n",
      " structure 6DRY already has been equilibrated\n",
      " structure 5TE5 already has been equilibrated\n",
      " structure 5TE5 already has been equilibrated\n",
      " structure 5WIU already has been equilibrated\n",
      " structure 5WIU already has been equilibrated\n",
      " structure 6IIV already has been equilibrated\n",
      " structure 6IIV already has been equilibrated\n",
      " structure 5V57 already has been equilibrated\n",
      " structure 5V57 already has been equilibrated\n",
      " structure 5X93 already has been equilibrated\n",
      " structure 5X93 already has been equilibrated\n",
      " structure 5OLV already has been equilibrated\n",
      " structure 5OLV already has been equilibrated\n",
      " structure 6PS2 already has been equilibrated\n",
      " structure 6PS2 already has been equilibrated\n",
      " structure 5VEW already has been equilibrated\n",
      " structure 5VEW already has been equilibrated\n",
      " structure 4QIM already has been equilibrated\n",
      " structure 4QIM already has been equilibrated\n",
      " structure 6AKY already has been equilibrated\n",
      " structure 6AKY already has been equilibrated\n",
      "model 5TE3 could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/5TE3. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 5TE3 already has been equilibrated\n",
      " structure 5ZK8 already has been equilibrated\n",
      " structure 5ZK8 already has been equilibrated\n",
      " structure 6D27 already has been equilibrated\n",
      " structure 6D27 already has been equilibrated\n",
      " structure 4JKV already has been equilibrated\n",
      " structure 4JKV already has been equilibrated\n",
      " structure 6TPG already has been equilibrated\n",
      " structure 6TPG already has been equilibrated\n",
      " structure 6RZ5 already has been equilibrated\n",
      " structure 6RZ5 already has been equilibrated\n",
      " structure 6TQ6 already has been equilibrated\n",
      " structure 6TQ6 already has been equilibrated\n",
      " structure 6TOT already has been equilibrated\n",
      " structure 6TOT already has been equilibrated\n",
      " structure 6PS8 already has been equilibrated\n",
      " structure 6PS8 already has been equilibrated\n",
      " structure 6K1Q already has been equilibrated\n",
      " structure 6K1Q already has been equilibrated\n",
      " structure 5WQC already has been equilibrated\n",
      " structure 5WQC already has been equilibrated\n",
      " structure 6ME9 already has been equilibrated\n",
      " structure 6ME9 already has been equilibrated\n",
      " structure 6CM4 already has been equilibrated\n",
      " structure 6CM4 already has been equilibrated\n",
      " structure 5T1A already has been equilibrated\n",
      " structure 5T1A already has been equilibrated\n",
      " structure 5OLO already has been equilibrated\n",
      " structure 5OLO already has been equilibrated\n",
      " structure 6AQF already has been equilibrated\n",
      " structure 6AQF already has been equilibrated\n",
      " structure 6OL9 already has been equilibrated\n",
      " structure 6OL9 already has been equilibrated\n",
      " structure 4EJ4 already has been equilibrated\n",
      " structure 4EJ4 already has been equilibrated\n",
      " structure 6AKX already has been equilibrated\n",
      " structure 6AKX already has been equilibrated\n",
      " structure 6FK8 already has been equilibrated\n",
      " structure 6FK8 already has been equilibrated\n",
      " structure 5N2R already has been equilibrated\n",
      " structure 5N2R already has been equilibrated\n",
      " structure 6C1Q already has been equilibrated\n",
      " structure 6C1Q already has been equilibrated\n",
      " structure 5O9H already has been equilibrated\n",
      " structure 5O9H already has been equilibrated\n",
      " structure 5K2A already has been equilibrated\n",
      " structure 5K2A already has been equilibrated\n",
      " structure 4GBR already has been equilibrated\n",
      " structure 4GBR already has been equilibrated\n",
      " structure 6KQI already has been equilibrated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-18 16:01:28,280 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 16:01:28,280 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 16:02:22,617 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/6KQI_apo/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " structure 6TOS already has been equilibrated\n",
      " structure 6TOS already has been equilibrated\n",
      " structure 6J21 already has been equilibrated\n",
      " structure 6J21 already has been equilibrated\n",
      " structure 5OLH already has been equilibrated\n",
      " structure 5OLH already has been equilibrated\n",
      " structure 6DRZ already has been equilibrated\n",
      " structure 6DRZ already has been equilibrated\n",
      " structure 5XPR already has been equilibrated\n",
      " structure 5XPR already has been equilibrated\n",
      " structure 6LI0 already has been equilibrated\n",
      " structure 6LI0 already has been equilibrated\n",
      " structure 6PS7 already has been equilibrated\n",
      " structure 6PS7 already has been equilibrated\n",
      " structure 5VEX already has been equilibrated\n",
      " structure 5VEX already has been equilibrated\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-18 16:02:56,171 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 16:02:56,171 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 16:04:25,677 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/5ZTY/\n",
      "2022-01-18 16:04:55,847 - htmd.protocols.equilibration_v2 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-01-18 16:04:55,848 - htmd.protocols.equilibration_v2 - INFO - Using user-provided restraints and ignoring constraints and fb_potential\n",
      "2022-01-18 16:06:02,014 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/equil/5ZTY_apo/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " structure 6GT3 already has been equilibrated\n",
      " structure 6GT3 already has been equilibrated\n",
      " structure 6RZ6 already has been equilibrated\n",
      " structure 6RZ6 already has been equilibrated\n",
      " structure 6PS3 already has been equilibrated\n",
      " structure 6PS3 already has been equilibrated\n",
      " structure 6A93 already has been equilibrated\n",
      " structure 6A93 already has been equilibrated\n",
      " structure 6ZDV already has been equilibrated\n",
      " structure 6ZDV already has been equilibrated\n",
      " structure 6ME4 already has been equilibrated\n",
      " structure 6ME4 already has been equilibrated\n",
      " structure 5WIV already has been equilibrated\n",
      " structure 5WIV already has been equilibrated\n",
      " structure 6TQ7 already has been equilibrated\n",
      " structure 6TQ7 already has been equilibrated\n",
      " structure 6AK3 already has been equilibrated\n",
      " structure 6AK3 already has been equilibrated\n",
      " structure 5UNF already has been equilibrated\n",
      " structure 5UNF already has been equilibrated\n",
      " structure 5N2S already has been equilibrated\n",
      " structure 5N2S already has been equilibrated\n",
      " structure 5JTB already has been equilibrated\n",
      " structure 5JTB already has been equilibrated\n",
      " structure 6RZ8 already has been equilibrated\n",
      " structure 6RZ8 already has been equilibrated\n",
      " structure 6TPN already has been equilibrated\n",
      " structure 6TPN already has been equilibrated\n",
      " structure 6M9T already has been equilibrated\n",
      " structure 6M9T already has been equilibrated\n",
      " structure 5VBL already has been equilibrated\n",
      " structure 5VBL already has been equilibrated\n",
      " structure 6FK6 already has been equilibrated\n",
      " structure 6FK6 already has been equilibrated\n",
      " structure 5ZKP already has been equilibrated\n",
      " structure 5ZKP already has been equilibrated\n",
      " structure 6PS4 already has been equilibrated\n",
      " structure 6PS4 already has been equilibrated\n",
      " structure 5TZY already has been equilibrated\n",
      " structure 5TZY already has been equilibrated\n",
      " structure 6FK9 already has been equilibrated\n",
      " structure 6FK9 already has been equilibrated\n",
      " structure 5T04 already has been equilibrated\n",
      " structure 5T04 already has been equilibrated\n",
      " structure 6TOD already has been equilibrated\n",
      " structure 6TOD already has been equilibrated\n",
      " structure 6A94 already has been equilibrated\n",
      " structure 6A94 already has been equilibrated\n",
      " structure 5NLX already has been equilibrated\n",
      " structure 5NLX already has been equilibrated\n",
      " structure 6KJV already has been equilibrated\n",
      " structure 6KJV already has been equilibrated\n",
      " structure 6D32 already has been equilibrated\n",
      " structure 6D32 already has been equilibrated\n",
      " structure 5WF6 already has been equilibrated\n",
      " structure 5WF6 already has been equilibrated\n",
      " structure 6ME5 already has been equilibrated\n",
      " structure 6ME5 already has been equilibrated\n",
      " structure 6TQ9 already has been equilibrated\n",
      " structure 6TQ9 already has been equilibrated\n",
      " structure 6IGK already has been equilibrated\n",
      " structure 6IGK already has been equilibrated\n",
      " structure 5X33 already has been equilibrated\n",
      " structure 5X33 already has been equilibrated\n",
      " structure 4Z9G already has been equilibrated\n",
      " structure 4Z9G already has been equilibrated\n",
      " structure 6JZH already has been equilibrated\n",
      " structure 6JZH already has been equilibrated\n",
      " structure 6IGL already has been equilibrated\n",
      " structure 6IGL already has been equilibrated\n",
      " structure 6FK7 already has been equilibrated\n",
      " structure 6FK7 already has been equilibrated\n",
      " structure 6KNM already has been equilibrated\n",
      " structure 6KNM already has been equilibrated\n",
      " structure 5MZP already has been equilibrated\n",
      " structure 5MZP already has been equilibrated\n",
      " structure 5K2B already has been equilibrated\n",
      " structure 5K2B already has been equilibrated\n",
      " structure 6IIU already has been equilibrated\n",
      " structure 6IIU already has been equilibrated\n",
      " structure 6DS0 already has been equilibrated\n",
      " structure 6DS0 already has been equilibrated\n",
      " structure 6PS5 already has been equilibrated\n",
      " structure 6PS5 already has been equilibrated\n",
      " structure 5NDD already has been equilibrated\n",
      " structure 5NDD already has been equilibrated\n",
      " structure 5K2C already has been equilibrated\n",
      " structure 5K2C already has been equilibrated\n",
      " structure 6KK7 already has been equilibrated\n",
      " structure 6KK7 already has been equilibrated\n",
      " structure 5UIW already has been equilibrated\n",
      " structure 5UIW already has been equilibrated\n",
      " structure 5UNH already has been equilibrated\n",
      " structure 5UNH already has been equilibrated\n",
      " structure 6ME7 already has been equilibrated\n",
      " structure 6ME7 already has been equilibrated\n",
      " structure 6KK1 already has been equilibrated\n",
      " structure 6KK1 already has been equilibrated\n",
      " structure 6ZDR already has been equilibrated\n",
      " structure 6ZDR already has been equilibrated\n",
      " structure 6DRX already has been equilibrated\n",
      " structure 6DRX already has been equilibrated\n",
      " structure 6RZ9 already has been equilibrated\n",
      " structure 6RZ9 already has been equilibrated\n",
      "model 6LI2 could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/6LI2. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 6LI2 already has been equilibrated\n",
      " structure 5ZBQ already has been equilibrated\n",
      " structure 5ZBQ already has been equilibrated\n",
      " structure 6HLP already has been equilibrated\n",
      " structure 6HLP already has been equilibrated\n",
      " structure 6HLL already has been equilibrated\n",
      " structure 6HLL already has been equilibrated\n",
      " structure 6TP4 already has been equilibrated\n",
      " structure 6TP4 already has been equilibrated\n",
      "model 5XSZ could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/5XSZ. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 5XSZ already has been equilibrated\n",
      " structure 4O9R already has been equilibrated\n",
      " structure 4O9R already has been equilibrated\n",
      " structure 5NM4 already has been equilibrated\n",
      " structure 5NM4 already has been equilibrated\n",
      "model 5GLI could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/5GLI. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 5GLI already has been equilibrated\n",
      " structure 6PRZ already has been equilibrated\n",
      " structure 6PRZ already has been equilibrated\n",
      " structure 5WF5 already has been equilibrated\n",
      " structure 5WF5 already has been equilibrated\n",
      " structure 5ZKC already has been equilibrated\n",
      " structure 5ZKC already has been equilibrated\n",
      " structure 5ZKB already has been equilibrated\n",
      " structure 5ZKB already has been equilibrated\n",
      " structure 5X7D already has been equilibrated\n",
      " structure 5X7D already has been equilibrated\n",
      " structure 5K2D already has been equilibrated\n",
      " structure 5K2D already has been equilibrated\n",
      " structure 6ME3 already has been equilibrated\n",
      " structure 6ME3 already has been equilibrated\n",
      "model 6KP6 could not be send to equilibrate because of  Could not locate any coordinates file in /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/build/6KP6. Please set the Equilibration.acemd.coordinates property to point to the coordinates file\n",
      " structure 6KP6 already has been equilibrated\n",
      " structure 5ZHP already has been equilibrated\n",
      " structure 5ZHP already has been equilibrated\n",
      " structure 6FFH already has been equilibrated\n",
      " structure 6FFH already has been equilibrated\n",
      " structure 5KW2 already has been equilibrated\n",
      " structure 5KW2 already has been equilibrated\n",
      " structure 6GPS already has been equilibrated\n",
      " structure 6GPS already has been equilibrated\n"
     ]
    }
   ],
   "source": [
    "#########################\n",
    "## Part 3: Equillibration\n",
    "#########################\n",
    "\n",
    "for pdbcode in pdb_set:\n",
    "# for pdbcode in ['5XR8']:\n",
    "    for apo in [False, True]:\n",
    "        try:\n",
    "            modelname = pdbcode+'_apo' if apo else pdbcode\n",
    "            equildir = resultspath+'equil/'+modelname+'/'\n",
    "            if os.path.exists(equildir+'output.xtc') or os.path.exists(equildir+'simrunning'):\n",
    "                print(\" structure %s already has been equilibrated\" % pdbcode)\n",
    "                continue\n",
    "\n",
    "            if not os.path.exists(equildir):\n",
    "                os.makedirs(equildir)\n",
    "\n",
    "            # Taking vmd selection line\n",
    "            const_sel = []\n",
    "            with open(strucpath+pdbcode+'/const_sel.txt', 'r') as outfile:\n",
    "                const_sel += outfile.readlines()\n",
    "\n",
    "            md = define_equilibration(const_sel)\n",
    "            md.write(resultspath+'build/'+modelname,equildir)\n",
    "\n",
    "            #Substitute run.sh generated by HTMD by a different one, adapted to the specified path of ACEMD\n",
    "            with open(equildir + 'run.sh', 'w') as f:\n",
    "                f.write('#!/bin/bash\\n%s > %slog.txt 2>&1' % (acemd_path, equildir))\n",
    "        \n",
    "            #Prepare slurm job  \n",
    "            sq = SlurmQueue()\n",
    "            sq.envvars = acemd_license\n",
    "            sq.jobname = 'eql_'+pdbcode\n",
    "            sq.datadir = None\n",
    "            sq.partition = 'gpcr_gpu'\n",
    "            sq.ngpu = 1\n",
    "            sq.ncpu = 1\n",
    "            sq.prerun = job_commands(equildir, '/home/daranda/'+modelname+'/')\n",
    "            sq.memory = 2000\n",
    "#             sq.exclude = ['arwen','aragorn','bifur']\n",
    "            sq.nodelist = ['arwen']\n",
    "    \n",
    "            # Submit\n",
    "            sq.submit(equildir)\n",
    "            \n",
    "        except Exception as e:\n",
    "            print(\"model \"+modelname+\" could not be send to equilibrate because of \",e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "replicate 1 of structure 4EJ4 already has been simulated\n",
      "replicate 2 of structure 4EJ4 already has been simulated\n",
      "replicate 3 of structure 4EJ4 already has been simulated\n",
      "submitting replicate 1 of 4EJ4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:11:09,787 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (SER 204 CA P P) (SER 204 C P P) (PRO 205 N P P) (PRO 205 CA P P) \" with indexes [2602, 2609, 2611, 2615]\n",
      "2022-02-10 14:11:10,254 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:12:01,055 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/4EJ4/rep_1/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "submitting replicate 2 of 4EJ4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:12:19,914 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (SER 204 CA P P) (SER 204 C P P) (PRO 205 N P P) (PRO 205 CA P P) \" with indexes [2602, 2609, 2611, 2615]\n",
      "2022-02-10 14:12:20,372 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:13:09,531 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/4EJ4/rep_2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "submitting replicate 3 of 4EJ4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:13:27,607 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (SER 204 CA P P) (SER 204 C P P) (PRO 205 N P P) (PRO 205 CA P P) \" with indexes [2602, 2609, 2611, 2615]\n",
      "2022-02-10 14:13:28,053 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:14:16,931 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/4EJ4/rep_3/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "replicate 1 of structure 6LRY already has been simulated\n",
      "replicate 2 of structure 6LRY already has been simulated\n",
      "replicate 3 of structure 6LRY already has been simulated\n",
      "submitting replicate 1 of 6LRY\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:14:48,405 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (ILE 212 CA P P) (ILE 212 C P P) (GLY 213 N P P) (GLY 213 CA P P) \" with indexes [2020, 2035, 2037, 2039]\n",
      "2022-02-10 14:14:48,901 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:15:40,890 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/6LRY/rep_1/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "submitting replicate 2 of 6LRY\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:16:00,232 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (ILE 212 CA P P) (ILE 212 C P P) (GLY 213 N P P) (GLY 213 CA P P) \" with indexes [2020, 2035, 2037, 2039]\n",
      "2022-02-10 14:16:00,697 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:16:44,572 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/6LRY/rep_2/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "submitting replicate 3 of 6LRY\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-10 14:17:03,929 - htmd.builder.builder - WARNING - Found cis peptide bond in 1 frames: [0] in the omega diheral \"Angle of (ILE 212 CA P P) (ILE 212 C P P) (GLY 213 N P P) (GLY 213 CA P P) \" with indexes [2020, 2035, 2037, 2039]\n",
      "2022-02-10 14:17:04,387 - htmd.protocols.production_v6 - WARNING - Lipids detected in input structure. We highly recommend setting useconstantratio=True for membrane simulations.\n",
      "2022-02-10 14:17:48,813 - jobqueues.slurmqueue - INFO - Queueing /gpcr/users/daranda/doctorat/GPCR_simulations/simulation_output/production/6LRY/rep_3/\n"
     ]
    }
   ],
   "source": [
    "#####################\n",
    "## Part 4: Production\n",
    "#####################\n",
    "\n",
    "# Production protocol\n",
    "md = define_production(timestep, trajperiod)\n",
    "\n",
    "# If some model should be skipped, put its name here\n",
    "modelname_skip = {}\n",
    "\n",
    "# For each PDB \n",
    "for pdbcode in ['4EJ4','6LRY']:\n",
    "# for pdbcode in pdb_set:\n",
    "#     for apo in [False]:\n",
    "    for apo in [True, False]:\n",
    "        # must match with equildir in equilibration launcher code and contain input and output of equilibration.\n",
    "        modelname = pdbcode+'_apo' if apo else pdbcode\n",
    "        equildir = '%s/equil/%s/' % (resultspath, modelname)\n",
    "        # IF equillibration is not avalible\n",
    "        if not os.path.exists(equildir+'output.xtc'):\n",
    "            print(\"replicate %d of structure %s has no avalible equillibration\" %(rep, pdbcode))\n",
    "            continue\n",
    "        for rep in range(1,repnum+1):\n",
    "            \n",
    "            try: \n",
    "                    \n",
    "                # If simulation for this PDB has already been run\n",
    "                proddir='%sproduction/%s/rep_%d/' % (resultspath, modelname, rep)\n",
    "                if os.path.exists(proddir+'/output.xtc') or os.path.exists(proddir+'simrunning'):\n",
    "                    print(\"replicate %d of structure %s already has been simulated\" %(rep, pdbcode))\n",
    "                    continue\n",
    "\n",
    "                print('submitting replicate %d of %s' % (rep, pdbcode))\n",
    "                # directory copy output of equilibration to production input (initial working directory for run_prod.sh).\n",
    "                md.write(equildir,proddir)\n",
    "\n",
    "                sq = SlurmQueue()\n",
    "                sq.envvars = acemd_license\n",
    "                sq.jobname = modelname+'_pr'+str(rep)\n",
    "                sq.datadir = None\n",
    "                sq.partition = 'gpcr_gpu'\n",
    "                sq.prerun = job_commands(proddir, '/home/daranda/%s_pr_%d/'%(modelname,rep))\n",
    "                sq.ngpu = 1\n",
    "                sq.ncpu = 1\n",
    "                sq.exclude = ['aragorn','arwen','bifur']\n",
    "\n",
    "                #Substitute run.sh generated by HTMD by a different one, adapted to the specified path of ACEMD\n",
    "                with open(proddir + 'run.sh', 'w') as f:\n",
    "                    f.write('#!/bin/bash\\n%s >%slog.txt 2>&1' % (acemd_path,proddir))\n",
    "\n",
    "                sq.submit(proddir)\n",
    "                \n",
    "            except Exception as e:\n",
    "                print(\"model \"+modelname+\" could not be send to equilibrate because of \",e)            \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#####################\n",
    "## EXTRA: ligand RMSD\n",
    "#####################\n",
    "### Assess quality of systems by calculating the RMSD of the ligand/s molecule/s\n",
    "\n",
    "# Set paths and files\n",
    "outpath = resultspath+'rmsd_lig.tsv'\n",
    "if os.path(outpath):\n",
    "    print('RMSDs already computed. SKipping...')\n",
    "else:\n",
    "    # Iterate over pdbcodes \n",
    "    results = []\n",
    "    for pdbcode in pdbcodes: \n",
    "\n",
    "        # Iterate over production replicates\n",
    "        for mytrajid in [\"1\",\"2\",\"3\"]:\n",
    "\n",
    "            try:\n",
    "                sysname = \"%s_rep%s\"%(pdbcode,mytrajid)\n",
    "\n",
    "                # Input files of simulation\n",
    "                files_path = resultspath+'production/%s/rep_%s/'%(name, mytrajid)\n",
    "                mypdbpath = files_path+'structure.pdb'\n",
    "                mypsfpath = files_path+'structure.psf'\n",
    "                mytrajpath = files_path+'output_wrapped.xtc'\n",
    "\n",
    "                # Skip if no trajectory\n",
    "                if not os.path.exists(mytrajpath):\n",
    "                    print(\"no trajectory replicate %s for system %s avalible. Skipping...\"%(mytrajid, sysname))\n",
    "                    continue\n",
    "\n",
    "                print('computing ligand RMSD for trajectory %s of system %s' % (mytrajid, sysname))\n",
    "\n",
    "                # Load trajectory and topology into MDA universe, and select protein atoms\n",
    "                u = mda.Universe(mypsfpath, mytrajpath)\n",
    "                ligsel = u.select_atoms(\"segid LIG and not resname CLR\")\n",
    "\n",
    "                # Compute rmsd, extract its values and put them in corresponding lists and dicts\n",
    "                R = rms.RMSD(ligsel)\n",
    "                R.run()\n",
    "                rmsd = np.mean([ a[2] for a in R.rmsd ])\n",
    "                results.append((sysname,rmsd)) \n",
    "\n",
    "            except Exception as e:\n",
    "                print(\"error: system %s failed becasue %s\"%(sysname,e))\n",
    "\n",
    "    # Once everything is done, sort and write RMSD results\n",
    "    out = open(outpath,'w')\n",
    "    out.write(\"Simulated_system\\tRMSD_ligand\\n\")\n",
    "    results_sorted = sorted(results, key=lambda tup: tup[0])\n",
    "    for line in results_sorted:\n",
    "        out.write(\"%s\\t%f\\n\"%(line[0],line[1])) \n",
    "    out.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wrapping replicate 1 of 4EJ4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/moleculekit/align.py:16: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, A), array(float32, 2d, A))\n",
      "  covariance = np.dot(P.T, Q)\n",
      "/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/moleculekit/align.py:54: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 2d, C), array(float32, 2d, A))\n",
      "  all1 = np.dot(all1, rot.T)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "wrapping replicate 2 of 4EJ4\n",
      "wrapping replicate 3 of 4EJ4\n",
      "End of 4EJ4 after 193.5178680419922 seconds\n",
      "\n",
      "wrapping replicate 1 of 6LRY\n",
      "wrapping replicate 2 of 6LRY\n",
      "wrapping replicate 3 of 6LRY\n",
      "End of 6LRY after 198.5682544708252 seconds\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "##########################\n",
    "## Part 5: Wrap Structures\n",
    "##########################\n",
    "\n",
    "# Wrap trajectories obtained during production with an htmd command\n",
    "gpcr_sel = \"protein and chain \"+new_pdb_chain\n",
    "# for pdbcode in pdb_set:\n",
    "for pdbcode in ['4EJ4','6LRY']:\n",
    "# for pdbcode in testset:\n",
    "    for apo in [False]:\n",
    "#     for apo in [True, False]:\n",
    "        try:\n",
    "            modelname = pdbcode+'_apo' if apo else pdbcode\n",
    "            mymol_pdb = Molecule('%sproduction/%s/rep_1/structure.pdb' % (resultspath, modelname))\n",
    "            for rep in range(1,repnum+1):\n",
    "            #for rep in [3]:\n",
    "                start_time = time.time()        \n",
    "                print('wrapping replicate %d of %s' % (rep, modelname))\n",
    "                proddir='%sproduction/%s/rep_%d/' % (resultspath, modelname, rep)\n",
    "                rep = str(rep)\n",
    "\n",
    "                # To avoid repeating wrapping in Trajectories already wrapped, check the existance of this file\n",
    "                outname = proddir+'output_wrapped.xtc'\n",
    "                if os.path.exists(outname):\n",
    "                    print('replicate already wrapped. Skipping...')\n",
    "                    continue\n",
    "                \n",
    "                # Wrap system\n",
    "                mymol = Molecule('%sproduction/%s/rep_1/structure.psf' % (resultspath, modelname))\n",
    "                mymol.read(proddir+'output.xtc')\n",
    "                mymol.wrap(gpcr_sel)\n",
    "\n",
    "                # Align frames and write\n",
    "                mymol.align('all', refmol=mymol_pdb)\n",
    "                mymol.write(outname)\n",
    "\n",
    "            print('End of %s after %s seconds\\n' % (modelname, time.time() - start_time))\n",
    "        except Exception as e:\n",
    "            print(\"model \"+modelname+\" could not be wrapped because of \",e)            \n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "sublist = [('1191','5WF5',True),\n",
    "('818','5WF5',False),\n",
    "('819','2Z73',True),\n",
    "('820','2Z73',False),\n",
    "('821','6RZ9',True),\n",
    "('822','6RZ9',False),\n",
    "('823','6ME3',True),\n",
    "('824','6ME3',False),\n",
    "('825','5ZKQ',True),\n",
    "('826','5ZKQ',False),\n",
    "('827','5JTB',True),\n",
    "('828','5JTB',False),\n",
    "('829','5VEW',True),\n",
    "('830','5VEW',False),\n",
    "('831','5ZBQ',True),\n",
    "('833','6GT3',True),\n",
    "('834','6GT3',False),\n",
    "('835','6KK1',True),\n",
    "('836','6KK1',False),\n",
    "('837','6FKC',True),\n",
    "('838','6FKC',False),\n",
    "('839','6GPS',True),\n",
    "('840','6GPS',False),\n",
    "('841','5VEX',True),\n",
    "('842','5VEX',False),\n",
    "('843','6ME2',True),\n",
    "('844','6ME2',False),\n",
    "('845','5OM4',True),\n",
    "('846','5OM4',False),\n",
    "('847','4NTJ',True),\n",
    "('848','4NTJ',False),\n",
    "('849','6DRX',True),\n",
    "('850','6DRX',False),\n",
    "('851','6FKB',True),\n",
    "('852','6FKB',False),\n",
    "('853','5T04',True),\n",
    "('854','5T04',False),\n",
    "('855','4EJ4',True),\n",
    "('856','4EJ4',False),\n",
    "('857','5WS3',True),\n",
    "('858','5WS3',False),\n",
    "('859','5F8U',True),\n",
    "('860','5F8U',False),\n",
    "('861','6HLO',True),\n",
    "('862','6HLO',False),\n",
    "('863','6FKA',True),\n",
    "('864','6FKA',False),\n",
    "('865','5T1A',True),\n",
    "('866','5T1A',False),\n",
    "('867','6TQ7',True),\n",
    "('868','6TQ7',False),\n",
    "('869','6RZ4',True),\n",
    "('870','6RZ4',False),\n",
    "('871','5V57',True),\n",
    "('872','5V57',False),\n",
    "('873','5YQZ',True),\n",
    "('874','5YQZ',False),\n",
    "('875','6TPJ',True),\n",
    "('876','6TPJ',False),\n",
    "('877','6KJV',True),\n",
    "('879','6M9T',True),\n",
    "('880','6M9T',False),\n",
    "('881','5TE3',True),\n",
    "('882','6PS5',True),\n",
    "('883','6PS5',False),\n",
    "('884','6AQF',True),\n",
    "('885','6AQF',False),\n",
    "('886','6FJ3',True),\n",
    "('887','3C9M',True),\n",
    "('888','3C9M',False),\n",
    "('889','6TP3',True),\n",
    "('890','6TP4',True),\n",
    "('891','6TP4',False),\n",
    "('892','6FFH',True),\n",
    "('893','6FFH',False),\n",
    "('894','6PS1',True),\n",
    "('895','6PS1',False),\n",
    "('896','5KW2',True),\n",
    "('897','5KW2',False),\n",
    "('898','4XES',True),\n",
    "('899','4XES',False),\n",
    "('900','5X7D',True),\n",
    "('901','5X7D',False),\n",
    "('902','4N4W',True),\n",
    "('903','4N4W',False),\n",
    "('904','5UIG',True),\n",
    "('905','5UIG',False),\n",
    "('906','5N2S',True),\n",
    "('908','6FK9',True),\n",
    "('909','6FK9',False),\n",
    "('910','6AKX',True),\n",
    "('912','6J21',True),\n",
    "('913','6J21',False),\n",
    "('914','5NM2',True),\n",
    "('915','5NM2',False),\n",
    "('916','2YCZ',True),\n",
    "('917','2YCZ',False),\n",
    "('918','6K1Q',True),\n",
    "('919','6K1Q',False),\n",
    "('920','6RZ8',True),\n",
    "('921','6RZ8',False),\n",
    "('922','4JKV',True),\n",
    "('923','4JKV',False),\n",
    "('924','5TZR',True),\n",
    "('925','5TZR',False),\n",
    "('926','6ME6',True),\n",
    "('927','6ME6',False),\n",
    "('928','5ZK8',True),\n",
    "('930','6GPX',True),\n",
    "('931','6GPX',False),\n",
    "('932','6A94',True),\n",
    "('933','6A94',False),\n",
    "('934','6ME4',True),\n",
    "('935','6ME4',False),\n",
    "('936','6RZ7',True),\n",
    "('937','6RZ7',False),\n",
    "('938','6ME5',True),\n",
    "('939','6ME5',False),\n",
    "('940','5TE5',True),\n",
    "('941','5TE5',False),\n",
    "('942','6DRZ',True),\n",
    "('943','6DRZ',False),\n",
    "('944','5OLZ',True),\n",
    "('945','5OLZ',False),\n",
    "('946','5EE7',True),\n",
    "('947','5EE7',False),\n",
    "('948','6ME9',True),\n",
    "('949','6ME9',False),\n",
    "('950','4NC3',True),\n",
    "('951','4NC3',False),\n",
    "('952','6TOS',True),\n",
    "('953','6TOS',False),\n",
    "('954','5V56',True),\n",
    "('955','5V56',False),\n",
    "('956','6RZ6',True),\n",
    "('957','6RZ6',False),\n",
    "('958','6ZDV',True),\n",
    "('959','6ZDV',False),\n",
    "('960','6PS4',True),\n",
    "('961','6PS4',False),\n",
    "('962','5WIV',True),\n",
    "('963','5WIV',False),\n",
    "('964','6TQ9',True),\n",
    "('965','6TQ9',False),\n",
    "('966','5UVI',True),\n",
    "('967','5UVI',False),\n",
    "('968','5GLI',True),\n",
    "('969','6LUQ',True),\n",
    "('970','6LUQ',False),\n",
    "('971','6FK8',True),\n",
    "('972','6FK8',False),\n",
    "('973','6FFI',True),\n",
    "('974','6FFI',False),\n",
    "('975','5WQC',True),\n",
    "('976','5WQC',False),\n",
    "('977','5TVN',True),\n",
    "('978','5TVN',False),\n",
    "('979','6A93',True),\n",
    "('980','6A93',False),\n",
    "('981','6HLL',True),\n",
    "('982','6HLL',False),\n",
    "('983','5XR8',True),\n",
    "('984','5K2B',True),\n",
    "('985','5K2B',False),\n",
    "('986','6E59',True),\n",
    "('987','6E59',False),\n",
    "('988','5OM1',True),\n",
    "('989','5OM1',False),\n",
    "('990','6TQ6',True),\n",
    "('991','6TQ6',False),\n",
    "('992','6RZ5',True),\n",
    "('993','6RZ5',False),\n",
    "('994','6PS2',True),\n",
    "('995','6PS2',False),\n",
    "('996','6TPG',True),\n",
    "('997','6TPG',False),\n",
    "('1000','6TP6',True),\n",
    "('1001','6TP6',False),\n",
    "('1002','5XSZ',True),\n",
    "('1003','5DYS',True),\n",
    "('1004','5DYS',False),\n",
    "('1005','6TOT',True),\n",
    "('1006','6TOT',False),\n",
    "('1007','6KK7',True),\n",
    "('1008','6KK7',False),\n",
    "('1009','4XT3',True),\n",
    "('1010','4XT3',False),\n",
    "('1011','4BUO',True),\n",
    "('1012','4BUO',False),\n",
    "('1015','6D27',True),\n",
    "('1016','6D27',False),\n",
    "('1017','5XPR',True),\n",
    "('1018','5XPR',False),\n",
    "('1019','6D35',True),\n",
    "('1020','6D35',False),\n",
    "('1021','5UNG',True),\n",
    "('1022','5UNG',False),\n",
    "('1023','6D32',True),\n",
    "('1024','6D32',False),\n",
    "('1025','1GZM',True),\n",
    "('1026','1GZM',False),\n",
    "('1027','5XRA',True),\n",
    "('1028','5XRA',False),\n",
    "('1029','5UIW',True),\n",
    "('1030','5UIW',False),\n",
    "('1031','3C9L',True),\n",
    "('1032','3C9L',False),\n",
    "('1033','6TPN',True),\n",
    "('1034','6TPN',False),\n",
    "('1035','5ZBH',True),\n",
    "('1036','5ZBH',False),\n",
    "('1037','5N2R',True),\n",
    "('1039','5V54',True),\n",
    "('1040','5V54',False),\n",
    "('1041','6J20',True),\n",
    "('1042','6J20',False),\n",
    "('1043','6FK7',True),\n",
    "('1044','6FK7',False),\n",
    "('1045','6TOD',True),\n",
    "('1046','6TOD',False),\n",
    "('1047','5UNF',True),\n",
    "('1048','5UNF',False),\n",
    "('1049','6DRY',True),\n",
    "('1050','6DRY',False),\n",
    "('1051','5O9H',True),\n",
    "('1052','5O9H',False),\n",
    "('1053','5OLG',True),\n",
    "('1054','5OLG',False),\n",
    "('1055','6CM4',True),\n",
    "('1056','6CM4',False),\n",
    "('1057','6PS3',True),\n",
    "('1058','6PS3',False),\n",
    "('1059','5TZY',True),\n",
    "('1060','5TZY',False),\n",
    "('1061','5OLH',True),\n",
    "('1062','5OLH',False),\n",
    "('1063','4QIM',True),\n",
    "('1064','4QIM',False),\n",
    "('1065','6LI1',True),\n",
    "('1066','6TO7',True),\n",
    "('1067','6TO7',False),\n",
    "('1068','6HLP',True),\n",
    "('1069','6HLP',False),\n",
    "('1070','5UNH',True),\n",
    "('1071','5UNH',False),\n",
    "('1072','6AKY',True),\n",
    "('1074','6FK6',True),\n",
    "('1075','6FK6',False),\n",
    "('1076','6ME8',True),\n",
    "('1077','6ME8',False),\n",
    "('1078','5ZHP',True),\n",
    "('1080','5K2D',True),\n",
    "('1081','5K2D',False),\n",
    "('1082','6PS6',True),\n",
    "('1083','6PS6',False),\n",
    "('1084','5NM4',True),\n",
    "('1085','5NM4',False),\n",
    "('1086','6KPC',True),\n",
    "('1087','6KPC',False),\n",
    "('1088','5X33',True),\n",
    "('1089','5X33',False),\n",
    "('1090','6DS0',True),\n",
    "('1091','6DS0',False),\n",
    "('1092','4GPO',True),\n",
    "('1093','5D6L',True),\n",
    "('1094','5D6L',False),\n",
    "('1095','5X93',True),\n",
    "('1096','5X93',False),\n",
    "('1097','6TQ4',True),\n",
    "('1098','6TQ4',False),\n",
    "('1099','5ZKC',True),\n",
    "('1101','4GBR',True),\n",
    "('1102','4GBR',False),\n",
    "('1104','6OL9',True),\n",
    "('1106','5OLO',True),\n",
    "('1107','5OLO',False),\n",
    "('1108','5NDD',True),\n",
    "('1109','5NDD',False),\n",
    "('1110','6PRZ',True),\n",
    "('1111','6PRZ',False),\n",
    "('1112','6MH8',True),\n",
    "('1113','6MH8',False),\n",
    "('1114','6KP6',True),\n",
    "('1115','6D26',True),\n",
    "('1116','6D26',False),\n",
    "('1117','6LI2',True),\n",
    "('1118','5MZP',True),\n",
    "('1119','5MZP',False),\n",
    "('1120','5ZK3',True),\n",
    "('1121','5ZK3',False),\n",
    "('1122','5NLX',True),\n",
    "('1123','5NLX',False),\n",
    "('1124','6ME7',True),\n",
    "('1125','6ME7',False),\n",
    "('1126','5MZJ',True),\n",
    "('1127','6LRY',True),\n",
    "('1128','6LRY',False),\n",
    "('1129','5K2C',True),\n",
    "('1130','5K2C',False),\n",
    "('1131','6JZH',True),\n",
    "('1132','6JZH',False),\n",
    "('1133','6KQI',False),\n",
    "('1134','6OBA',True),\n",
    "('1135','6OBA',False),\n",
    "('1136','5WIU',True),\n",
    "('1137','5WIU',False),\n",
    "('1138','6PS8',True),\n",
    "('1139','6PS8',False),\n",
    "('1140','5WF6',True),\n",
    "('1141','5WF6',False),\n",
    "('1142','5VRA',True),\n",
    "('1143','5VRA',False),\n",
    "('1144','6PS7',True),\n",
    "('1145','6PS7',False),\n",
    "('1146','6IGK',True),\n",
    "('1147','6IGK',False),\n",
    "('1148','5OLV',False),\n",
    "('1149','5NDZ',True),\n",
    "('1150','5NDZ',False),\n",
    "('1151','5K2A',True),\n",
    "('1152','5K2A',False),\n",
    "('1153','4O9R',True),\n",
    "('1154','4O9R',False),\n",
    "('1155','5ZKP',True),\n",
    "('1156','5ZKP',False),\n",
    "('1157','6AK3',True),\n",
    "('1158','6AK3',False),\n",
    "('1159','5ZKB',True),\n",
    "('1160','5ZKB',False),\n",
    "('1161','4Z9G',True),\n",
    "('1162','4Z9G',False),\n",
    "('1163','6ZDR',True),\n",
    "('1164','6ZDR',False),\n",
    "('1174','6KJV',False),\n",
    "('1175','5ZBQ',False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def resp_to_dict(resp):\n",
    "    # Convert a json reponse into a dictionary\n",
    "    return eval(resp.content.decode('UTF-8').replace('true', 'True').replace('false','False'))\n",
    "\n",
    "def get_headers(s, subm_id):\n",
    "    \"\"\"\n",
    "    Obtain headers, csrftoken and sessionid from current session object\n",
    "    \"\"\"\n",
    "    sessionid = str(s.cookies['sessionid'])\n",
    "    csrftoken = str(s.cookies['csrftoken'])\n",
    "    headers = {\n",
    "        'Referer' : mainurl+'/dynadb/step1/'+subm_id,\n",
    "        'Cookie' : 'csrftoken=%s; sessionid=%s' %(csrftoken,sessionid),\n",
    "        'X-CSRFToken' : csrftoken,\n",
    "    }\n",
    "    return(sessionid, csrftoken, headers)\n",
    "def check_chains(pdbcode, mymol):\n",
    "    \"\"\"\n",
    "    Check how many chains from the original PDB structure remain in mymol structure\n",
    "    And to which Segments of our structure they correspond\n",
    "    \"\"\"\n",
    "    # Load blosum score matrix to align proteins\n",
    "    blosum62 = substitution_matrices.load(\"BLOSUM62\")\n",
    "\n",
    "    # Obtain sequences for original PDB file chains, and classifying them by chainID\n",
    "    # Also check which segment corresponds to what chain\n",
    "    pdbmol = Molecule(pdbcode)\n",
    "    chainseg = {}\n",
    "    chainset = set(pdbmol.get('chain', sel='protein'))\n",
    "    pdbmol_segseqs = pdbmol.sequence()\n",
    "    pdbmol_chainseqs = {}\n",
    "    for chain in chainset:\n",
    "        segid = np.unique(pdbmol.get('segid', sel='chain '+chain))[0]\n",
    "        pdbmol_chainseqs[chain] = pdbmol_segseqs[segid]\n",
    "\n",
    "    # Merging all protein chains in our systems into a single megachain\n",
    "    mymol_megachain = ''\n",
    "    for seg,chain in mymol.sequence().items():\n",
    "        mymol_megachain = mymol_megachain + chain\n",
    "\n",
    "    # Aligning sequences from original PDB to simulated PDB megasequence\n",
    "    # To know which chains from the original PDB are preserved\n",
    "    chain_present = {}\n",
    "    segtochain = {}\n",
    "    # For each chain in the original PDB file\n",
    "    for chain,seq in pdbmol_chainseqs.items():\n",
    "        chain_present[chain] = False\n",
    "        # For each segment in our molecule\n",
    "        for seg,myseq in mymol.sequence().items():\n",
    "            # Align our molecule segments to the pdb chains\n",
    "            aligs = pairwise2.align.localms(seq, myseq, 5,-1, -1, -0.5)\n",
    "            # Check if any of the alignments has more than 4 score by position\n",
    "            is_present = any([ (alig.score/(alig.end-alig.start)) > 3.5 for alig in aligs ]) \n",
    "            if is_present:\n",
    "                chain_present[chain] = is_present\n",
    "                segtochain[seg] = chain\n",
    "\n",
    "    return (chain_present, segtochain)\n",
    "\n",
    "def get_uniprot_alignment(mymol, segpos, uniprotkbac, isoform, entry_segs):\n",
    "    \"\"\"\n",
    "    Find out the exact coordinates of each Uniprot segment in our protein\n",
    "    \"\"\"\n",
    "\n",
    "    # Obtain sequence of uniprotkbac\n",
    "    uniprot_id = \"%s-%s\"%(uniprotkbac,isoform) if isoform else uniprotkbac \n",
    "    response = requests.get(\"https://www.uniprot.org/uniprot/?query=accession:%s&sort=score&columns=sequence&format=tab\"%uniprotkbac)\n",
    "    uniseq = response.text.split('\\n')[1]\n",
    "    \n",
    "    # Match this uniprot to our protein segments, and save matches\n",
    "    for seg,myseq in mymol.sequence().items():\n",
    "        \n",
    "        # Align our molecule segments to the pdb chains\n",
    "        aligs = pairwise2.align.localms(myseq, uniseq, 5,-1, -1, -0.5)\n",
    "        # For every alignment\n",
    "        for alig in aligs:\n",
    "            start = alig.start\n",
    "            end = alig.end\n",
    "            # If the alignment has no significant amount of mismatches, keep it as segment\n",
    "            if (alig.score/(end-start)) > 3.5:\n",
    "                \n",
    "                # Find coordinates of this segment\n",
    "                alig_myseq = alig[0]\n",
    "                \n",
    "                # Find start of uniprot sequence in alginment\n",
    "                if alig_myseq.startswith('-'):\n",
    "                    starting = 0\n",
    "                else:\n",
    "                    starting = start - 1 \n",
    "                # Find ending of uniprot sequence in alginment            \n",
    "                if alig_myseq.endswith('-'):            \n",
    "                    ending = len(myseq) - 1\n",
    "                else:\n",
    "                    ending = end - 1 - alig.seqA.count('-')\n",
    "                \n",
    "                if uniprotkbac not in entry_segs:\n",
    "                    entry_segs[uniprotkbac] = []\n",
    "                    \n",
    "                # Do not add repeated alignments                \n",
    "                beg = segpos[seg][starting]\n",
    "                end = segpos[seg][ending]\n",
    "                repeated_seg = any([ ((seg['beg']==beg) and (seg['end']==end)) for seg in entry_segs[uniprot_id]])\n",
    "                if not len(entry_segs[uniprot_id]) or not repeated_seg:\n",
    "                    entry_segs[uniprot_id].append({\n",
    "                                'chainid' : mymol.get('chain', 'segid '+seg)[0],\n",
    "                                'segid' : seg,\n",
    "                                'beg' : segpos[seg][starting],\n",
    "                                'end' : segpos[seg][ending]})\n",
    "\n",
    "    return(entry_segs) \n",
    "\n",
    "\n",
    "def get_pdb_info(pdbcode, mymol, ligandsdict, blacklist, apo):\n",
    "    \"\"\"\n",
    "    Get informarion of the system specified in the pdbcode from the RCSB-PDB webpage.\n",
    "    Mainly uniprot sequences, chainIDs and uniprot codes for the chains\n",
    "    \"\"\"\n",
    "    \n",
    "    # Make string for selecting lignads in rcsb's api\n",
    "    ligs = '\"'+'\",\"'.join(ligandsdict[pdbcode].keys())+'\"'\n",
    "    gpcr_names = ['ceptor','rhodopsin','smoothened']\n",
    "    \n",
    "    #Check which chains from the original PDB structure are preserved in mymol structure\n",
    "    (chain_present, segtochain) = check_chains(pdbcode, mymol)\n",
    "    \n",
    "    # Get residue names in our system\n",
    "    allresnames = set(mymol.resname)\n",
    "    \n",
    "    # Extract information from pdb webpage using api\n",
    "    ligdict = dict()\n",
    "    protdict = dict()\n",
    "    datadict = dict()\n",
    "    entry_segs = dict()\n",
    "    segpos = dict()\n",
    "\n",
    "    # Obtain dictionary of sorted protein resids in each segment\n",
    "    for seg in set(mymol.get('segid', 'protein')):\n",
    "        segpos[seg] = list(dict.fromkeys(mymol.get('resid', 'segid '+seg)))\n",
    "    \n",
    "    # Get information from PDB api\n",
    "    pdbdict = requests.get('https://data.rcsb.org/graphql?query={\\\n",
    "        entry(entry_id: \"'+pdbcode+'\") {\\\n",
    "            exptl {method}\\\n",
    "            polymer_entities {\\\n",
    "                entity_poly{pdbx_strand_id,rcsb_sample_sequence_length}\\\n",
    "                rcsb_polymer_entity{pdbx_description}\\\n",
    "                rcsb_polymer_entity_align{\\\n",
    "                    aligned_regions {length}\\\n",
    "                    reference_database_accession \\\n",
    "                    reference_database_isoform \\\n",
    "                }\\\n",
    "            }\\\n",
    "        }\\\n",
    "        chem_comps(comp_ids: ['+ligs+']) {\\\n",
    "            rcsb_chem_comp_descriptor {InChIKey}\\\n",
    "            chem_comp{id,name}\\\n",
    "        }\\\n",
    "    }').json()['data']\n",
    "        \n",
    "    # Determine experimental method used, and use the corresponding id in GPCRmd database\n",
    "    method = pdbdict['entry']['exptl'][0]['method'].lower()\n",
    "    if 'x-ray' in method: \n",
    "        method_id = 0\n",
    "        method_segs = 0\n",
    "    elif 'nmr' in method:\n",
    "        method_id = 1\n",
    "        method_segs = 1\n",
    "    elif 'electron microscopy' == method:\n",
    "        method_id = 4\n",
    "        method_segs = 7\n",
    "    else:\n",
    "        method_id = 5 # Other method\n",
    "        method_segs = 6\n",
    "            \n",
    "    # Get ligand information and classify it (in case the system actually has ligands)\n",
    "    if len(pdbdict['chem_comps'])>0:\n",
    "        for lig in pdbdict['chem_comps']:\n",
    "            ligandInchi = lig['rcsb_chem_comp_descriptor']['InChIKey']\n",
    "            ligandResname = lig['chem_comp']['id']\n",
    "            ligandName = lig['chem_comp']['name']\n",
    "    \n",
    "            # Exclude ligands not present in our simulated system\n",
    "            if ligandResname in allresnames:\n",
    "                ligdict[ligandResname] = (ligandName,ligandInchi)\n",
    "\n",
    "            # Beware of covalent-bound retinols\n",
    "            if (ligandResname=='RET') and (\"LYR\" in allresnames):\n",
    "                ligdict[\"LYR\"] = (\"Lysine-Retinol\",\"FSIGCSBKSBDRLV-GGOCYBHBSA-N\")\n",
    "                \n",
    "            # Beware of other covalent-bound ligands\n",
    "            if (\"COV\" in allresnames) and (ligandsdict[pdbcode][ligandResname][1]):\n",
    "                ligdict[\"COV\"] = (ligandName,ligandInchi)\n",
    "                \n",
    "    # Extract protein chains information\n",
    "    recname = \"\"\n",
    "    peplig = (\"\",100000000000000000)\n",
    "    for poly in pdbdict['entry']['polymer_entities']:\n",
    "\n",
    "        # Determine if this polymer (chain) is a GPCR\n",
    "        uniname = poly['rcsb_polymer_entity']['pdbx_description'].lower()\n",
    "        seqlen = poly['entity_poly']['rcsb_sample_sequence_length']\n",
    "        chainIds = poly['entity_poly']['pdbx_strand_id'].split(',')\n",
    "        isgpcr = any([(name in uniname) for name in gpcr_names]) \n",
    "        \n",
    "        # Extract name of the system from pdb info\n",
    "        if isgpcr:\n",
    "            uninames = uniname.split(',')\n",
    "            for unis in uninames:\n",
    "                if any([(name in unis) for name in gpcr_names]):\n",
    "                    recname = unis\n",
    "        else: \n",
    "            # Select shortest non-receptor sequence as potential peptide ligand\n",
    "            if seqlen < peplig[1]:\n",
    "                peplig = (uniname.split(',')[0],seqlen)\n",
    "                    \n",
    "        # Get uniprots present in the system, and their corresponbding segments\n",
    "        for alreg in poly['rcsb_polymer_entity_align']:\n",
    "            uniprotkbac = alreg['reference_database_accession']\n",
    "            isoform = alreg['reference_database_isoform']\n",
    "            uniprot_id = \"%s-%s\"%(uniprotkbac,isoform) if isoform else uniprotkbac \n",
    "            entry_segs = get_uniprot_alignment(mymol, segpos, uniprotkbac, isoform, entry_segs)\n",
    "            \n",
    "            # If this uniprot is present in our system, assign its data to protdict\n",
    "            if uniprot_id in entry_segs:\n",
    "                protdict[uniprotkbac] = {\n",
    "                    'isoform' : isoform if isoform else '1',\n",
    "                    'isgpcr' : isgpcr,\n",
    "                    'segs' : entry_segs[uniprot_id],\n",
    "                    'method' : method_id,\n",
    "                    'method_segs' : method_segs\n",
    "                }\n",
    "    \n",
    "    # Filter ligdict of blacklisted molecules and ions\n",
    "    for lig in ligdict:\n",
    "        if (lig in blacklist) or (len(lig) == 2):\n",
    "            ligdict.pop(lig)\n",
    "\n",
    "    ligname = \"\"\n",
    "    # Find the ligand molecule, if not apo\n",
    "    if not apo:\n",
    "        for lig in ligdict:\n",
    "            molname = ligdict[lig][0]\n",
    "            inchikey = ligdict[lig][1]\n",
    "            if lig != 'CLR': # Take any non-cholesterol, non blacklisted molecule as ligand\n",
    "                # some molecule names are excessivelly large. Take inchikey instead\n",
    "                ligname = molname if len(molname) < 51 else inchikey \n",
    "            # If there is not any other molecule in the system, nor a potential peptide ligand, take cholesterol\n",
    "            elif (len(ligdict.keys()) == 1) and not peplig[0]: \n",
    "                ligname = ligdict[lig][0]\n",
    "        \n",
    "        # If still we have no ligand name, must be peptide\n",
    "        if not ligname:\n",
    "            ligname = peplig[0]\n",
    "\n",
    "    # Make system name\n",
    "    sysname = recname+\" in complex with \"+ligname if not apo else recname+\" (apoform)\"\n",
    "    \n",
    "    return (protdict,ligdict,segtochain,method_id,sysname)\n",
    "\n",
    "def login(s):\n",
    "    # Login into GPCRmd \n",
    "    # Will you go, lassie, go? And we'll all gooo toghetheeeer\n",
    "    headers = {\n",
    "        'Cookie': 'csrftoken=ZYa0n3mNBjwBMlpDh4lnbBJQdi7GdlP4',\n",
    "        'Referer': mainurl+'/accounts/login/',\n",
    "    }\n",
    "    datalogin = {\n",
    "        'username': 'david',\n",
    "        'password': 'Ameboid',\n",
    "        'next' : '/accounts/memberpage/',\n",
    "        'csrfmiddlewaretoken' : 'ZYa0n3mNBjwBMlpDh4lnbBJQdi7GdlP4'\n",
    "    }\n",
    "    print('loging into GPCRmd')\n",
    "    logo = s.post(mainurl+'/accounts/login/', \n",
    "               data=datalogin,\n",
    "               headers=headers)\n",
    "    return s\n",
    "\n",
    "def new_submission(s, mainurl):\n",
    "    \"\"\"\n",
    "    Create a new submission entry in GPCRmd\n",
    "    \"\"\"\n",
    "\n",
    "    response_new = s.get(mainurl + '/dynadb/step0/')\n",
    "    soup = BeautifulSoup(response_new.text, 'html.parser')\n",
    "    step1_link = soup.find('form',attrs={'id' : 'mainform'}).get('action')\n",
    "    subm_id = step1_link.split('/')[-2]\n",
    "    print('new submission %s created'%subm_id)\n",
    "    return subm_id\n",
    "\n",
    "def new_step1(s, subm_id, pdbcode, trajperiod, timestep, prodpath, modelfile, method_id, sysname, apo):\n",
    "    \"\"\"\n",
    "    Fullfill and send the step 1 of the new submissionform\n",
    "    # Looone liiie the fieeeelds of Athenryyyyy...\n",
    "    # Where once, we watched, the small seabirds flyyyyy....\n",
    "    \"\"\"\n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "    \n",
    "    # All the stuff we need to send \n",
    "    data_submit = {\n",
    "        'name' : sysname,\n",
    "        'type' : 0 if apo else 1, # 0 for apo, 1 for complex\n",
    "        'pdbid' : pdbcode,\n",
    "        'description' : 'Classical unbiased (NVT ensemble) complex flexibility assay.',\n",
    "        'source_type' : method_id,\n",
    "        'id_dynamics_methods': '1', #Molecular mechanics,\n",
    "        'software': 'ACEMD3',\n",
    "        'sversion': 'GPUGRID', # TODO: Is that correct??,\n",
    "        'ff': 'CHARMM',\n",
    "        'ffversion': '36m Feb 2016',\n",
    "        'id_assay_types': 1, # Orthosteric (un)/binding,\n",
    "        'id_dynamics_membrane_types': 2, # Homogeneus membrane,\n",
    "        'id_dynamics_solvent_types': 2, # TIP3P solvent,\n",
    "        'timestep': timestep,\n",
    "        'delta': (trajperiod*timestep)/1e6, # time/frames\n",
    "        'add_info': ' Classical unbiased (NVT ensemble) complex flexibility assay.', \n",
    "    }\n",
    "    files_submit = {\n",
    "        'dynamics' : open(prodpath+'structure.pdb', 'rb'),\n",
    "        'model' : open(modelfile, 'rb')\n",
    "    }\n",
    "\n",
    "    # Submit step 1\n",
    "    rep = s.post(mainurl+'/dynadb/step1_submit/'+subm_id+'/',\n",
    "            headers = headers,\n",
    "            files = files_submit,\n",
    "            data = data_submit)\n",
    "    print('step1 finalized ',rep)\n",
    "\n",
    "    # Raise exceptions if something failed\n",
    "    if not rep.ok:\n",
    "        #s.post(mainurl+'/dynadb/delete_submission/'+subm_id)\n",
    "        raise Exception('Step 1 submitted returned %d'%rep.status_code)\n",
    "\n",
    "\n",
    "    \n",
    "def smalmol_getdata(s, molid, inchikey, subm_id, moltype, resname, sdfpath=\"\"):\n",
    "    \"\"\"\n",
    "    Get data for small molecules \n",
    "    \"\"\"\n",
    "    \n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "\n",
    "    # Get info of molecule\n",
    "    rep = s.get(mainurl+'/dynadb/smalmol_info/?inchikey=%s&submission_id=%s'%(inchikey,subm_id),\n",
    "        headers = headers,\n",
    "        )\n",
    "    rep_dict = rep.json()\n",
    "    \n",
    "    # Prepare submission dictionary with the data of this molecule\n",
    "    mol_datasubmit = {\n",
    "        \"smalmol_type\"+molid : moltype,\n",
    "        \"smalmol_inchikey\"+molid : inchikey,\n",
    "        \"smalmol_name\"+molid : rep_dict['name'],\n",
    "        \"smalmol_resname\"+molid : resname,\n",
    "        \"smalmol_iupac\"+molid : rep_dict['iupac'],\n",
    "        \"smalmol_chemblid\"+molid :  rep_dict['chemblid'],\n",
    "        \"smalmol_cid\"+molid :  rep_dict['cid'],\n",
    "        \"smalmol_inchi\"+molid : rep_dict['inchi'],\n",
    "        \"smalmol_sinchi\"+molid :  rep_dict['sinchi'],\n",
    "        \"smalmol_sinchikey\"+molid :  rep_dict['sinchikey'],\n",
    "        \"smalmol_smiles\"+molid :  rep_dict['smiles'],\n",
    "        \"smalmol_netcharge\"+molid :  rep_dict['net_charge'],\n",
    "        \"smalmol_synonims\"+molid :  rep_dict['other_names'],\n",
    "        \"smalmol_description\"+molid :  '',\n",
    "        \"image_path\"+molid :  rep_dict['imagepath']\n",
    "    }\n",
    "    \n",
    "    # Add SDFfile to the submission if required\n",
    "    mol_files = {} if rep_dict['inGPCRmd'] else {\"sdfmol\"+molid : open(sdfpath)} \n",
    "    \n",
    "    return(mol_datasubmit,mol_files)\n",
    "\n",
    "def new_step2(s, subm_id, ligdict, apo):\n",
    "    \"\"\"\n",
    "    Fullfil and send step2 of new submission form\n",
    "    \"\"\"    \n",
    "    print('initiating step 2: small molecule data')\n",
    "\n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "    \n",
    "    # Introduce common small molecules (waters, lipids and ions)\n",
    "    i = 0\n",
    "    num_entries = []\n",
    "    step2_data = {\n",
    "        'csrfmiddlewaretoken' : csrftoken,\n",
    "        'submit' : 'submit'\n",
    "    }\n",
    "    step2_files = {}\n",
    "    common_mols = {\n",
    "        'TIP3' : ('XLYOFNOQVPJJNP-UHFFFAOYSA-N',6),\n",
    "        'POPC' : ('WTJKGGKOPKCXLL-VYOBOKEXSA-N',7),\n",
    "        'SOD' : ('FKNQFGJONOIPTF-UHFFFAOYSA-N',8),\n",
    "        'CLA' : ('VEXZGXHMUGYJMC-UHFFFAOYSA-M',8)\n",
    "    }\n",
    "    for resname,(inchikey,moltype) in common_mols.items():\n",
    "        (mol_datasubmit, mol_files) = smalmol_getdata(s, str(i), inchikey, subm_id, moltype, resname)\n",
    "        step2_data.update(mol_datasubmit)\n",
    "        step2_files.update(mol_files)\n",
    "        num_entries.append(i)\n",
    "        i+=1\n",
    "\n",
    "    # Introduce ligands: all will be defined as orthosteric ligands\n",
    "    lignames = []\n",
    "    for lig,(name,inchikey) in ligdict.items():\n",
    "        # Skip this if molecule is apoform\n",
    "        if apo:\n",
    "            break\n",
    "        # Avoid blacklisted molecules or ions\n",
    "        if (lig in detergent_blacklist) or (lig in glucids_blacklist)  or (len(lig) == 2):\n",
    "            continue\n",
    "        # Download ligand, store it into temporary file\n",
    "        response = requests.get('https://files.rcsb.org/ligands/view/'+lig+'_ideal.sdf')\n",
    "        with open('tmpfile.sdf','wb') as tmpout:\n",
    "            tmpout.write(response.content)\n",
    "\n",
    "        # CLR (cholesterol) is usually a experimental lipid \n",
    "        moltype = '3' if lig=='CLR' else '0'\n",
    "            \n",
    "        # Retrieve molecule information\n",
    "        (mol_datasubmit, mol_files) = smalmol_getdata(s, str(i), inchikey, subm_id, moltype, lig, 'tmpfile.sdf')\n",
    "        step2_data.update(mol_datasubmit)\n",
    "        step2_files.update(mol_files)\n",
    "        num_entries.append(i)\n",
    "        i+=1\n",
    "        os.remove('tmpfile.sdf')\n",
    "    \n",
    "    # Add number of entries\n",
    "    step2_data['num_entries'] = ','.join(map(str, num_entries))\n",
    "    \n",
    "    # Replace empty entries with an empty string\n",
    "    for key,val in step2_data.items():\n",
    "        if not val:\n",
    "            step2_data[key] = ''\n",
    "    \n",
    "    # Submit step2\n",
    "    rep = s.post(mainurl+'/dynadb/step2_submit/'+subm_id+'/',\n",
    "        headers = headers,\n",
    "        data = step2_data,\n",
    "        files = step2_files)\n",
    "    print('step2 finalized ',rep)\n",
    "\n",
    "    # Raise exceptions if something failed\n",
    "    if not rep.ok:\n",
    "        #s.post(mainurl+'/dynadb/delete_submission/'+subm_id)\n",
    "        raise Exception('Step 2 submitted returned %d'%rep.status_code)\n",
    "\n",
    "def new_step3(s, subm_id, pdbcode, protdict):\n",
    "    \"\"\"\n",
    "    Fullfil and sent the step3 (protein chains) of the new submission form\n",
    "    \"\"\"\n",
    "    print('initiating step 3: protein chains')\n",
    "\n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "    \n",
    "    # Start dict with the data to be send into GPCRmd\n",
    "    # But thougth they paved the footways here with goldust, I still would chose... my Isle of Innesfree\n",
    "    step3_data = {\n",
    "        'csrfmiddlewaretoken':csrftoken,\n",
    "        'submission_id' : subm_id,\n",
    "        'submit' : 'Submit',\n",
    "    }\n",
    "\n",
    "    # Iterate over pdb chains. For each one we'll create an entry\n",
    "    i=0\n",
    "    entrynum = []\n",
    "    for uniprotkbac,entry_data in protdict.items():\n",
    "        ii = str(i)\n",
    "        isgpcr = entry_data['isgpcr']\n",
    "        isoform = entry_data['isoform']\n",
    "        segs = entry_data['segs']\n",
    "        method_segs = entry_data['method_segs']\n",
    "\n",
    "        # Retrieve info using uniprot code\n",
    "        unidict = s.get(mainurl+'/dynadb/prot_info/'+subm_id+'/',\n",
    "                          params = {\n",
    "                              'uniprotkbac' : uniprotkbac,\n",
    "                              'isoform' : isoform,\n",
    "                              'submission_id' : subm_id,\n",
    "                          },\n",
    "                          headers = headers\n",
    "                         ).json()\n",
    "\n",
    "        # We'll obtain the alignment of our protein sequence and its UNIPROT sequence using\n",
    "        # GPCRmd's tool for do it\n",
    "        data_alig = {\n",
    "            'unisequence' : unidict['Sequence'],\n",
    "            'submission_id' : subm_id\n",
    "        }\n",
    "\n",
    "        # Get segments for alignment data\n",
    "        j = 0\n",
    "        segnums = []\n",
    "        for seg in segs:\n",
    "            ss = str(j)\n",
    "            data_alig['chain'+ss]= seg['chainid']\n",
    "            data_alig['segid'+ss]= seg['segid']\n",
    "            data_alig['from'+ss]= seg['beg']\n",
    "            data_alig['to'+ss]= seg['end']\n",
    "            segnums.append(ss)\n",
    "\n",
    "            # Store segment information as well in the submit dict\n",
    "            step3_segdata = {\n",
    "                ss+'seg'+ii: j,\n",
    "                ss+'pdbid'+ii: pdbcode,\n",
    "                ss+'sourcetype'+ii: method_segs,\n",
    "                ss+'chain'+ii: seg['chainid'],\n",
    "                ss+'segid'+ii: seg['segid'],\n",
    "                ss+'from_resid'+ii: seg['beg'],\n",
    "                ss+'to_resid'+ii: seg['end'],\n",
    "            }\n",
    "            step3_data.update(step3_segdata)\n",
    "\n",
    "            j += 1\n",
    "        data_alig['segnums']=','.join(segnums)\n",
    "        step3_data['num_segs'+ii]=','.join(segnums)\n",
    "\n",
    "        # Send alignment request\n",
    "        alignment = s.post(mainurl+'/dynadb/get_alignment/',\n",
    "                      data = data_alig,\n",
    "                      headers = headers\n",
    "                     ).text\n",
    "\n",
    "        # Get mutations\n",
    "        mutations_dict = s.post(mainurl+'/dynadb/protein/get_mutations/',\n",
    "                      data = {\n",
    "                          'alignment' : alignment,\n",
    "                          'sequence' : unidict['Sequence'],\n",
    "                      },\n",
    "                      headers = headers\n",
    "                     ).json()['mutations']\n",
    "\n",
    "        # Store mutations in submission dictionary\n",
    "        h = 0\n",
    "        mutnums = []\n",
    "        for mut in mutations_dict:\n",
    "            hh = str(h)\n",
    "            step3_mutdata = {\n",
    "                hh+'from'+ii : mut['from'],\n",
    "                hh+'to'+ii : mut['to'],\n",
    "                hh+'resid'+ii : mut['resid'],\n",
    "            }\n",
    "            step3_data.update(step3_mutdata)\n",
    "            mutnums.append(hh)\n",
    "            h+=1\n",
    "        step3_data['num_muts'+ii]=','.join(mutnums)\n",
    "\n",
    "        # Get chain name according to uniprot\n",
    "        names_array = unidict['Protein names'].split(\" (\")\n",
    "        names_array = list(map(lambda x: re.sub(\"\\)$|\\) \\[.*\\]$\", \"\", x), names_array)) # Convert the name string into name aray\n",
    "        name = names_array.pop(0) # We get the first name as the official one\n",
    "        other_names = ';'.join(names_array)\n",
    "        \n",
    "        # Start storing information to send in step3\n",
    "        step3_entrydata = {\n",
    "            'isoform'+ii : isoform,\n",
    "            'prot_uniprot'+ii : uniprotkbac,\n",
    "            'name'+ii : name,\n",
    "            'aliases'+ii : other_names,\n",
    "            'species_code'+ii : unidict['Organism ID'],\n",
    "            'species_name'+ii : unidict['Organism'],\n",
    "            'unisequence'+ii : unidict['Sequence'],\n",
    "        }\n",
    "        step3_data.update(step3_entrydata)\n",
    "\n",
    "        entrynum.append(ii)    \n",
    "        i+=1\n",
    "\n",
    "    # Store number of entries\n",
    "    step3_data['num_entries'] = ','.join(entrynum)\n",
    "\n",
    "    # Submit step3\n",
    "    rep = s.post(mainurl+'/dynadb/step3_submit/'+subm_id+'/',\n",
    "        headers = headers,\n",
    "        data = step3_data)\n",
    "    \n",
    "    print('step3 finalized ',rep)\n",
    "\n",
    "    # Raise exceptions if something failed\n",
    "    if not rep.ok:\n",
    "        #s.post(mainurl+'/dynadb/delete_submission/'+subm_id)\n",
    "        raise Exception('Step 3 submitted returned %d'%rep.status_code)\n",
    "\n",
    "def other_files_zip(prodpath):\n",
    "    \"\"\"\n",
    "    Compress protocol files in the path submitted \n",
    "    \"\"\"\n",
    "    outfolder = prodpath+'other_files'\n",
    "    outfile = outfolder+'.tar.gz'\n",
    "    if not os.path.exists(outfile):\n",
    "        os.makedirs(outfolder)\n",
    "        # For each replicate, create a folder with symbolic links of the to-compress files\n",
    "        for rep in [1,2,3]:\n",
    "            repath = '%srep_%d/' % (prodpath,rep)\n",
    "            outrep = '%s/rep_%d/' % (outfolder,rep)\n",
    "            os.makedirs(outrep)\n",
    "            list_files = ['input', 'input.coor', 'input.vel','input.xsc','log.txt']\n",
    "            for file in list_files:\n",
    "                shutil.copyfile(repath+file, outrep+file)\n",
    "\n",
    "        # Compress file\n",
    "        shutil.make_archive(outfolder, 'gztar', outfolder)\n",
    "        shutil.rmtree(outfolder)\n",
    "\n",
    "    return(outfile)\n",
    "\n",
    "def new_step4(s, subm_id, prodpath, repath):\n",
    "    \"\"\"\n",
    "    Fullfil and sent the step4 (files) of the new submission form\n",
    "    \"\"\"\n",
    "    print('initiating step 4: simulation files')\n",
    "\n",
    "    # Get headers and stuff\n",
    "    (sessionid, csrftoken, headers)=get_headers(s, subm_id)\n",
    "\n",
    "    # Prepare trajectories\n",
    "    step4_files = []\n",
    "    for trajid in range(1,4):\n",
    "        trajpath = '%s/rep_%d/output_wrapped.xtc'%(prodpath,trajid)\n",
    "        step4_files.append(('trj', ('traj%d.xtc'%trajid, open(trajpath, 'rb'))))\n",
    "    \n",
    "    # Prepare the rest of files (parameters files will name-changed to avoid problems with filextensions)\n",
    "    step4_files.append(\n",
    "        ('dyn', ('structure.psf', open(repath+'structure.psf','rb')))\n",
    "    )\n",
    "    step4_files.append(\n",
    "        ('prm', ('parameters.prm', open(repath+'parameters','rb')))\n",
    "    )\n",
    "    other_files = other_files_zip(prodpath)\n",
    "    step4_files.append(\n",
    "        ('oth', ('other_files.tar.gz', open(other_files,'rb')))\n",
    "    )\n",
    "    \n",
    "    # Encode data\n",
    "    m=MultipartEncoder(fields=step4_files)\n",
    "    headers['Content-Type'] = m.content_type\n",
    "    \n",
    "    # Submit step4\n",
    "    rep = s.post(mainurl+'/dynadb/step4_submit/'+subm_id+'/',\n",
    "        headers = headers,\n",
    "        data = m)\n",
    "    \n",
    "    print('step4 finalized ',rep)\n",
    "\n",
    "    # Raise exceptions if something failed\n",
    "    if not rep.ok:\n",
    "        s.post(mainurl+'/dynadb/delete_submission/'+subm_id)\n",
    "        raise Exception('Step 4 submitted returned %d'%rep.status_code)\n",
    "    # If everything went ok so far, write a \"submitted\" file to know this system has already been submitted\n",
    "    else:\n",
    "        with open(prodpath+'submitted.txt','w') as out:\n",
    "            out.write(subm_id)\n",
    "\n",
    "def check_completeness(prodpath):\n",
    "    \"\"\"\n",
    "    Check if we actually have the three necessary replicates for submission\n",
    "    \"\"\"\n",
    "\n",
    "    incomplete = False\n",
    "    for trajid in range(1,4):\n",
    "        trajpath = '%s/rep_%d/output_wrapped.xtc'%(prodpath,trajid)\n",
    "        if not os.path.exists(trajpath):\n",
    "            incomplete = True\n",
    "\n",
    "    return(incomplete)            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "loging into GPCRmd\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-16 10:53:26,630 - moleculekit.readers - INFO - Attempting PDB query for 4EJ4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 4EJ4_complex simulation ########################\n",
      "856 4EJ4 False\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-02-16 10:55:12,159 - moleculekit.readers - INFO - Attempting PDB query for 6LRY\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "###############Submitting 6LRY_complex simulation ########################\n",
      "1128 6LRY False\n",
      "initiating step 4: simulation files\n",
      "step4 finalized  <Response [200]>\n"
     ]
    }
   ],
   "source": [
    "#### Actual Simulation submission\n",
    "# mainurl = 'http://localhost:8000' \n",
    "mainurl = 'https://submission.gpcrmd.org'\n",
    "\n",
    "## Step -2: Login into GPCRmd\n",
    "with requests.Session() as s:\n",
    "    login(s)\n",
    "# For each of the currently-working-with systems defined in Part 1\n",
    "# for pdbcode in pdb_set:\n",
    "# for pdbcode in ['6D32']:\n",
    "bad_list = [('856','4EJ4',False),('1128','6LRY',False)]\n",
    "for (subm_id,pdbcode, apo) in bad_list:\n",
    "#     for apo in [False]:\n",
    "#     for apo in [True, False]:\n",
    "        \n",
    "        try:\n",
    "            # Set the path to the folder containing this simulation's files\n",
    "            sysname = pdbcode+\"_apo\" if apo else pdbcode+\"_complex\" \n",
    "            prodpath = resultspath+'production/'+pdbcode+'_apo/' if apo else resultspath+'production/'+pdbcode+'/' \n",
    "            repath = prodpath+'rep_1/'# For files no changing across representations\n",
    "            modelfolder = \"%sreceptor2curate_output/%s/\"%(basepath, pdbcode) \n",
    "            modelpath = \"%s%s_apo_curated.pdb\"%(modelfolder,pdbcode) if apo else \"%s%s_complex_curated.pdb\"%(modelfolder,pdbcode)\n",
    "            \n",
    "            # Skip if not all trajectories or already submitted or if already submitted\n",
    "            incomplete = check_completeness(prodpath)\n",
    "            submitted = os.path.exists(prodpath+'submitted.txt')\n",
    "            #if incomplete or submitted:\n",
    "            #    continue            \n",
    "            \n",
    "            # Load molecule\n",
    "            if not os.path.exists(repath+'structure.pdb'):\n",
    "                continue\n",
    "            mymol = Molecule(repath+'structure.pdb')\n",
    "            print('\\n###############Submitting '+sysname+' simulation ########################')            \n",
    "\n",
    "            ## Step -1: Get information of protein chains and ligand molecules from PDB web\n",
    "            (protdict,ligdict,segtochain,method_id,sysname) = get_pdb_info(pdbcode, mymol, ligandsdict, blacklist, apo)\n",
    "            \n",
    "            ## Step 0: Create a new submission\n",
    "#             subm_id = new_submission(s, mainurl)\n",
    "            print(subm_id, pdbcode, apo)\n",
    "#             subm_id = \"969\"\n",
    "\n",
    "            ## Step 1: General information\n",
    "#             new_step1(s, subm_id, pdbcode, trajperiod, timestep, repath, modelpath, method_id, sysname, apo)\n",
    "            \n",
    "            ## Step 2: Small molecules \n",
    "#             new_step2(s, subm_id, ligdict, apo)   \n",
    "\n",
    "            ## Step 3: Protein chains\n",
    "#             new_step3(s, subm_id, pdbcode, protdict)\n",
    "\n",
    "            ## Step 4: Dynamics information\n",
    "            new_step4(s, subm_id, prodpath, repath)\n",
    "    \n",
    "        except Exception as e:\n",
    "            print(\"Simulation \"+sysname+\" could not be submitted because \",e)            \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dyn_id:  701\n",
      "Workbench:  <Response [500]>\n",
      "Report:  <Response [500]>\n",
      "\n",
      "\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-5-952d4e826019>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mdyn_id\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m686\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1034\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mworkbench_reponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'https://submission.gpcrmd.org/view/'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdyn_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m     \u001b[0mreport_reponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'https://submission.gpcrmd.org/dynadb/dynamics/id/'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdyn_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mworkbench_reponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mok\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mreport_reponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mok\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Dyn_id: \"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mdyn_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/requests/api.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     75\u001b[0m     \u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetdefault\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'allow_redirects'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 76\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'get'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     77\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     78\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/requests/api.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     59\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    528\u001b[0m         }\n\u001b[1;32m    529\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 530\u001b[0;31m         \u001b[0mresp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    531\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    663\u001b[0m             \u001b[0;31m# Redirect resolving generator.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    664\u001b[0m             \u001b[0mgen\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mresolve_redirects\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 665\u001b[0;31m             \u001b[0mhistory\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mresp\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mresp\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgen\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    666\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    667\u001b[0m             \u001b[0mhistory\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m    663\u001b[0m             \u001b[0;31m# Redirect resolving generator.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    664\u001b[0m             \u001b[0mgen\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mresolve_redirects\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 665\u001b[0;31m             \u001b[0mhistory\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mresp\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mresp\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgen\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    666\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    667\u001b[0m             \u001b[0mhistory\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mresolve_redirects\u001b[0;34m(self, resp, req, stream, timeout, verify, cert, proxies, yield_requests, **adapter_kwargs)\u001b[0m\n\u001b[1;32m    243\u001b[0m                     \u001b[0mproxies\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mproxies\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m                     \u001b[0mallow_redirects\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m                     \u001b[0;34m**\u001b[0m\u001b[0madapter_kwargs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m                 )\n\u001b[1;32m    247\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    641\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    642\u001b[0m         \u001b[0;31m# Send the request\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 643\u001b[0;31m         \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0madapter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    644\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    645\u001b[0m         \u001b[0;31m# Total elapsed time of the request (approximately)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/requests/adapters.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[1;32m    447\u001b[0m                     \u001b[0mdecode_content\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    448\u001b[0m                     \u001b[0mretries\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmax_retries\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 449\u001b[0;31m                     \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    450\u001b[0m                 )\n\u001b[1;32m    451\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[1;32m    670\u001b[0m                 \u001b[0mbody\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    671\u001b[0m                 \u001b[0mheaders\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 672\u001b[0;31m                 \u001b[0mchunked\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mchunked\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    673\u001b[0m             )\n\u001b[1;32m    674\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    419\u001b[0m                     \u001b[0;31m# Python 3 (including for exceptions like SystemExit).\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    420\u001b[0m                     \u001b[0;31m# Otherwise it looks like a bug in the code.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 421\u001b[0;31m                     \u001b[0msix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_from\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    422\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mSocketTimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mBaseSSLError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mSocketError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    423\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_raise_timeout\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0me\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout_value\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mread_timeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/urllib3/packages/six.py\u001b[0m in \u001b[0;36mraise_from\u001b[0;34m(value, from_value)\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/site-packages/urllib3/connectionpool.py\u001b[0m in \u001b[0;36m_make_request\u001b[0;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[1;32m    414\u001b[0m                 \u001b[0;31m# Python 3\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    415\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 416\u001b[0;31m                     \u001b[0mhttplib_response\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgetresponse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    417\u001b[0m                 \u001b[0;32mexcept\u001b[0m \u001b[0mBaseException\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    418\u001b[0m                     \u001b[0;31m# Remove the TypeError from the exception chain in\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/http/client.py\u001b[0m in \u001b[0;36mgetresponse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1352\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1353\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1354\u001b[0;31m                 \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbegin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1355\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mConnectionError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1356\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/http/client.py\u001b[0m in \u001b[0;36mbegin\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    305\u001b[0m         \u001b[0;31m# read until we get a non-100 response\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    306\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 307\u001b[0;31m             \u001b[0mversion\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstatus\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreason\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_read_status\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    308\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mstatus\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0mCONTINUE\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    309\u001b[0m                 \u001b[0;32mbreak\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/http/client.py\u001b[0m in \u001b[0;36m_read_status\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    266\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    267\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_read_status\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 268\u001b[0;31m         \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreadline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_MAXLINE\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"iso-8859-1\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    269\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0m_MAXLINE\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    270\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mLineTooLong\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"status line\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/socket.py\u001b[0m in \u001b[0;36mreadinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    584\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    585\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 586\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    587\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_timeout_occurred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/ssl.py\u001b[0m in \u001b[0;36mrecv_into\u001b[0;34m(self, buffer, nbytes, flags)\u001b[0m\n\u001b[1;32m   1010\u001b[0m                   \u001b[0;34m\"non-zero flags not allowed in calls to recv_into() on %s\"\u001b[0m \u001b[0;34m%\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1011\u001b[0m                   self.__class__)\n\u001b[0;32m-> 1012\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnbytes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1013\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1014\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0msocket\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnbytes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mflags\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/ssl.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m    872\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Read on closed or unwrapped SSL socket.\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    873\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 874\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    875\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mSSLError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    876\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mSSL_ERROR_EOF\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msuppress_ragged_eofs\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/soft/system/easybuild/software/Miniconda3/4.7.10/lib/python3.6/ssl.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, len, buffer)\u001b[0m\n\u001b[1;32m    629\u001b[0m         \"\"\"\n\u001b[1;32m    630\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mbuffer\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 631\u001b[0;31m             \u001b[0mv\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    632\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    633\u001b[0m             \u001b[0mv\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sslobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "for dyn_id in range(686, 1034): \n",
    "    workbench_reponse = requests.get('https://submission.gpcrmd.org/view/'+str(dyn_id))\n",
    "    report_reponse = requests.get('https://submission.gpcrmd.org/dynadb/dynamics/id/'+str(dyn_id))                       \n",
    "    if not (workbench_reponse.ok and report_reponse.ok):\n",
    "        print(\"Dyn_id: \",dyn_id)\n",
    "        print(\"Workbench: \", workbench_reponse)\n",
    "        print(\"Report: \", report_reponse)\n",
    "        print(\"\\n\")\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
